<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Simon Reider">

<title>MBVagTrans Results</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="MbVagTrans_Results_files/libs/clipboard/clipboard.min.js"></script>
<script src="MbVagTrans_Results_files/libs/quarto-html/quarto.js"></script>
<script src="MbVagTrans_Results_files/libs/quarto-html/popper.min.js"></script>
<script src="MbVagTrans_Results_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="MbVagTrans_Results_files/libs/quarto-html/anchor.min.js"></script>
<link href="MbVagTrans_Results_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="MbVagTrans_Results_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="MbVagTrans_Results_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="MbVagTrans_Results_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="MbVagTrans_Results_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#pbs-empty-control" id="toc-pbs-empty-control" class="nav-link" data-scroll-target="#pbs-empty-control">PBS empty control</a></li>
  <li><a href="#diversity-analysis" id="toc-diversity-analysis" class="nav-link" data-scroll-target="#diversity-analysis">Diversity Analysis</a>
  <ul class="collapse">
  <li><a href="#alpha-diversity" id="toc-alpha-diversity" class="nav-link" data-scroll-target="#alpha-diversity">Alpha-diversity</a>
  <ul class="collapse">
  <li><a href="#effect-of-covariates" id="toc-effect-of-covariates" class="nav-link" data-scroll-target="#effect-of-covariates">Effect of covariates</a></li>
  </ul></li>
  <li><a href="#beta-diversity" id="toc-beta-diversity" class="nav-link" data-scroll-target="#beta-diversity">Beta-diversity</a></li>
  </ul></li>
  <li><a href="#differential-abundance" id="toc-differential-abundance" class="nav-link" data-scroll-target="#differential-abundance">Differential abundance</a></li>
  <li><a href="#taxonomic-composition" id="toc-taxonomic-composition" class="nav-link" data-scroll-target="#taxonomic-composition">Taxonomic composition</a>
  <ul class="collapse">
  <li><a href="#phylum-level" id="toc-phylum-level" class="nav-link" data-scroll-target="#phylum-level">Phylum level</a></li>
  <li><a href="#class-level" id="toc-class-level" class="nav-link" data-scroll-target="#class-level">Class level</a></li>
  <li><a href="#order-level" id="toc-order-level" class="nav-link" data-scroll-target="#order-level">Order level</a></li>
  <li><a href="#family-level" id="toc-family-level" class="nav-link" data-scroll-target="#family-level">Family level</a></li>
  <li><a href="#genus-level" id="toc-genus-level" class="nav-link" data-scroll-target="#genus-level">Genus level</a></li>
  </ul></li>
  <li><a href="#secondary-analyses" id="toc-secondary-analyses" class="nav-link" data-scroll-target="#secondary-analyses">Secondary analyses</a>
  <ul class="collapse">
  <li><a href="#duration-of-menopause" id="toc-duration-of-menopause" class="nav-link" data-scroll-target="#duration-of-menopause">Duration of menopause</a></li>
  <li><a href="#duration-of-gender-affirming-hormone-therapy-gaht" id="toc-duration-of-gender-affirming-hormone-therapy-gaht" class="nav-link" data-scroll-target="#duration-of-gender-affirming-hormone-therapy-gaht">Duration of gender-affirming hormone therapy (GAHT)</a></li>
  <li><a href="#absolute-height-of-testosterone-in-the-trans-men-group" id="toc-absolute-height-of-testosterone-in-the-trans-men-group" class="nav-link" data-scroll-target="#absolute-height-of-testosterone-in-the-trans-men-group">Absolute Height of Testosterone in the trans-men group</a></li>
  <li><a href="#duration-of-gnrh-therapy-in-trans-men-group" id="toc-duration-of-gnrh-therapy-in-trans-men-group" class="nav-link" data-scroll-target="#duration-of-gnrh-therapy-in-trans-men-group">Duration of GnRH therapy in trans-men group</a></li>
  <li><a href="#cycle-dependency-in-pre-menopausal-group" id="toc-cycle-dependency-in-pre-menopausal-group" class="nav-link" data-scroll-target="#cycle-dependency-in-pre-menopausal-group">Cycle dependency in pre-menopausal group</a></li>
  <li><a href="#influence-of-sexual-activity-on-microbiome-composition" id="toc-influence-of-sexual-activity-on-microbiome-composition" class="nav-link" data-scroll-target="#influence-of-sexual-activity-on-microbiome-composition">Influence of sexual activity on microbiome composition</a></li>
  <li><a href="#influence-of-the-duration-of-amenorrhea-on-microbiome-composition" id="toc-influence-of-the-duration-of-amenorrhea-on-microbiome-composition" class="nav-link" data-scroll-target="#influence-of-the-duration-of-amenorrhea-on-microbiome-composition">Influence of the duration of amenorrhea on microbiome composition</a></li>
  <li><a href="#influence-of-sex-of-partner" id="toc-influence-of-sex-of-partner" class="nav-link" data-scroll-target="#influence-of-sex-of-partner">Influence of sex of partner</a></li>
  </ul></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="MbVagTrans_Results.pdf"><i class="bi bi-file-pdf"></i>PDF</a></li></ul></div></nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">MBVagTrans Results</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Simon Reider </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>Include here metadata summaries and summary of phyloseq object</p>
</section>
<section id="pbs-empty-control" class="level1">
<h1>PBS empty control</h1>
<p>The PBS empty control seems to be contaminated with normal vaginal flora.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>taxdistri <span class="ot">=</span> <span class="fu">read_csv2</span>( <span class="st">"results/microbiome/pbs_taxonomic_composition.csv"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">mutate</span>(taxdistri, <span class="at">Abundance =</span> <span class="fu">round</span>(Abundance, <span class="at">digits =</span> <span class="dv">2</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Taxonomy</th>
<th style="text-align: right;">Abundance</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">p_Firmicutes;f_Lactobacillaceae;g_Lactobacillus</td>
<td style="text-align: right;">30.57</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Actinobacteriota;f_Bifidobacteriaceae;g_Bifidobacterium</td>
<td style="text-align: right;">16.69</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Actinobacteriota;f_Bifidobacteriaceae;g_Gardnerella</td>
<td style="text-align: right;">12.40</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Proteobacteria;f_Moraxellaceae;g_Enhydrobacter</td>
<td style="text-align: right;">5.80</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Firmicutes;f_Family XI;g_Anaerococcus</td>
<td style="text-align: right;">4.72</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Actinobacteriota;f_Bifidobacteriaceae;g_Alloscardovia</td>
<td style="text-align: right;">3.61</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Firmicutes;f_Streptococcaceae;g_Lactococcus</td>
<td style="text-align: right;">3.41</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Bacteroidota;f_Prevotellaceae;g_Prevotella</td>
<td style="text-align: right;">3.37</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Firmicutes;f_Streptococcaceae;g_Streptococcus</td>
<td style="text-align: right;">3.05</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Proteobacteria;f_Moraxellaceae;g_Acinetobacter</td>
<td style="text-align: right;">2.45</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Actinobacteriota;f_Atopobiaceae;g_Atopobium</td>
<td style="text-align: right;">2.16</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Firmicutes;f_Family XI;g_Finegoldia</td>
<td style="text-align: right;">1.55</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Firmicutes;f_Staphylococcaceae;g_Staphylococcus</td>
<td style="text-align: right;">1.31</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Actinobacteriota;f_Propionibacteriaceae;g_Cutibacterium</td>
<td style="text-align: right;">1.17</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Firmicutes;f_Lachnospiraceae;g_Anaerostipes</td>
<td style="text-align: right;">1.17</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Firmicutes;f_Ruminococcaceae;g_Faecalibacterium</td>
<td style="text-align: right;">1.08</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Actinobacteriota;f_Corynebacteriaceae;g_Corynebacterium</td>
<td style="text-align: right;">0.92</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Actinobacteriota;f_Actinomycetaceae;g_NA</td>
<td style="text-align: right;">0.69</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Proteobacteria;f_Xanthobacteraceae;g_Bradyrhizobium</td>
<td style="text-align: right;">0.66</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Actinobacteriota;f_Actinomycetaceae;g_Varibaculum</td>
<td style="text-align: right;">0.59</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Firmicutes;f_Family XI;g_Ezakiella</td>
<td style="text-align: right;">0.59</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Firmicutes;f_Enterococcaceae;g_Enterococcus</td>
<td style="text-align: right;">0.58</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Actinobacteriota;f_Corynebacteriaceae;g_NA</td>
<td style="text-align: right;">0.54</td>
</tr>
<tr class="even">
<td style="text-align: left;">p_Actinobacteriota;f_Pseudonocardiaceae;g_Prauserella</td>
<td style="text-align: right;">0.54</td>
</tr>
<tr class="odd">
<td style="text-align: left;">p_Firmicutes;f_Marinococcaceae;g_Alteribacillus</td>
<td style="text-align: right;">0.39</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="results/microbiome/pbs_taxonomic_composition_piechart.png" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Taxonomic composition of PBS sample</figcaption>
</figure>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ps_samples <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"intermediate/ps_samples.rds"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(ps_samples)<span class="sc">$</span>grp <span class="ot">=</span> <span class="fu">factor</span>(<span class="fu">sample_data</span>(ps_samples)<span class="sc">$</span>grp, </span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"01_postmenopausal"</span>,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">"02_premenopausal"</span>,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">"03_trans-men"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="diversity-analysis" class="level1">
<h1>Diversity Analysis</h1>
<section id="alpha-diversity" class="level2">
<h2 class="anchored" data-anchor-id="alpha-diversity">Alpha-diversity</h2>
<p>Alpha diversity describes the number and distribution of taxa <em>within</em> a sample or group of samples. Different indices exist that capture and describe different aspects of this ecologic measure.</p>
<p>In this dataset, alpha diversity is markedly different between the group of interest (i.e.&nbsp;<em>03_trans</em>) and the two control groups. However, the intervention group is more similar to the first control group (i.e.&nbsp;<em>postmenopausal women</em>).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>adiv <span class="ot">=</span> <span class="fu">plot_richness</span>(ps_samples, <span class="at">x =</span> <span class="st">"grp"</span>, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">"Observed"</span>, <span class="st">"InvSimpson"</span>, <span class="st">"Shannon"</span>))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>aov <span class="ot">=</span> adiv<span class="sc">$</span>data <span class="sc">%&gt;%</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">anova_test</span>(value<span class="sc">~</span>grp)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>tukey <span class="ot">=</span> adiv<span class="sc">$</span>data <span class="sc">%&gt;%</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tukey_hsd</span>(value<span class="sc">~</span>grp)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(adiv<span class="sc">$</span>data, <span class="fu">aes</span>(<span class="at">x =</span> grp, <span class="at">y =</span> value, <span class="at">fill =</span> grp)) <span class="sc">+</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>variable, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">fill =</span> <span class="st">"Group"</span>)<span class="sc">+</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> custom_pal) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/alpha%20diversity-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">select</span>(tukey, variable, group1,group2,p.adj, p.adj.signif))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 16%">
<col style="width: 26%">
<col style="width: 25%">
<col style="width: 13%">
<col style="width: 19%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">variable</th>
<th style="text-align: left;">group1</th>
<th style="text-align: left;">group2</th>
<th style="text-align: right;">p.adj</th>
<th style="text-align: left;">p.adj.signif</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">1.08e-02</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="even">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">1.61e-03</td>
<td style="text-align: left;">**</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: left;">****</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">2.35e-04</td>
<td style="text-align: left;">***</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">2.88e-04</td>
<td style="text-align: left;">***</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.00e+00</td>
<td style="text-align: left;">****</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">1.16e-01</td>
<td style="text-align: left;">ns</td>
</tr>
<tr class="even">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">3.48e-02</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">6.14e-05</td>
<td style="text-align: left;">****</td>
</tr>
</tbody>
</table>
</div>
</div>
<section id="effect-of-covariates" class="level3">
<h3 class="anchored" data-anchor-id="effect-of-covariates">Effect of covariates</h3>
<p>To rule out a possible effect of age on alpha diversity, pearson correlation of alpha diversity measures with age are performed within each of the 3 groups separately.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>adiv<span class="sc">$</span>data <span class="sc">%&gt;%</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable, grp) <span class="sc">%&gt;%</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cor_test</span>(value, Age) <span class="sc">%&gt;%</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(grp, variable, cor, statistic, p, conf.low, conf.high) <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 24%">
<col style="width: 14%">
<col style="width: 9%">
<col style="width: 14%">
<col style="width: 8%">
<col style="width: 14%">
<col style="width: 13%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">grp</th>
<th style="text-align: left;">variable</th>
<th style="text-align: right;">cor</th>
<th style="text-align: right;">statistic</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">conf.low</th>
<th style="text-align: right;">conf.high</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: left;">Observed</td>
<td style="text-align: right;">0.230</td>
<td style="text-align: right;">1.1510894</td>
<td style="text-align: right;">0.262</td>
<td style="text-align: right;">-0.1781717</td>
<td style="text-align: right;">0.5754530</td>
</tr>
<tr class="even">
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: left;">Observed</td>
<td style="text-align: right;">0.100</td>
<td style="text-align: right;">0.4925164</td>
<td style="text-align: right;">0.627</td>
<td style="text-align: right;">-0.3052951</td>
<td style="text-align: right;">0.4779954</td>
</tr>
<tr class="odd">
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: left;">Observed</td>
<td style="text-align: right;">-0.084</td>
<td style="text-align: right;">-0.4049561</td>
<td style="text-align: right;">0.689</td>
<td style="text-align: right;">-0.4638494</td>
<td style="text-align: right;">0.3216860</td>
</tr>
<tr class="even">
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: left;">Shannon</td>
<td style="text-align: right;">0.130</td>
<td style="text-align: right;">0.6463659</td>
<td style="text-align: right;">0.524</td>
<td style="text-align: right;">-0.2761356</td>
<td style="text-align: right;">0.5021954</td>
</tr>
<tr class="odd">
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: left;">Shannon</td>
<td style="text-align: right;">-0.076</td>
<td style="text-align: right;">-0.3639234</td>
<td style="text-align: right;">0.719</td>
<td style="text-align: right;">-0.4571294</td>
<td style="text-align: right;">0.3293109</td>
</tr>
<tr class="even">
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: left;">Shannon</td>
<td style="text-align: right;">-0.220</td>
<td style="text-align: right;">-1.0946590</td>
<td style="text-align: right;">0.285</td>
<td style="text-align: right;">-0.5677396</td>
<td style="text-align: right;">0.1892415</td>
</tr>
<tr class="odd">
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: right;">0.180</td>
<td style="text-align: right;">0.8633770</td>
<td style="text-align: right;">0.397</td>
<td style="text-align: right;">-0.2343598</td>
<td style="text-align: right;">0.5348641</td>
</tr>
<tr class="even">
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: right;">-0.062</td>
<td style="text-align: right;">-0.2976233</td>
<td style="text-align: right;">0.769</td>
<td style="text-align: right;">-0.4461513</td>
<td style="text-align: right;">0.3415505</td>
</tr>
<tr class="odd">
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: right;">-0.110</td>
<td style="text-align: right;">-0.5522881</td>
<td style="text-align: right;">0.586</td>
<td style="text-align: right;">-0.4874977</td>
<td style="text-align: right;">0.2940179</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>No comparison is significant.</p>
<section id="length-of-gaht-within-patient-group" class="level4">
<h4 class="anchored" data-anchor-id="length-of-gaht-within-patient-group">Length of GAHT within patient group</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>adiv<span class="sc">$</span>data <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(grp <span class="sc">==</span><span class="st">"03_trans-men"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cor_test</span>(value, GHAT_Length) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(variable, cor, statistic, p, conf.low, conf.high) <span class="sc">%&gt;%</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, <span class="at">caption =</span> <span class="st">"Pearson Correlation of a-diversity measures with length of GAHT in the trans-men patient group"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Pearson Correlation of a-diversity measures with length of GAHT in the trans-men patient group</caption>
<thead>
<tr class="header">
<th style="text-align: left;">variable</th>
<th style="text-align: right;">cor</th>
<th style="text-align: right;">statistic</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">conf.low</th>
<th style="text-align: right;">conf.high</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: right;">-0.15</td>
<td style="text-align: right;">-0.705</td>
<td style="text-align: right;">0.488</td>
<td style="text-align: right;">-0.511</td>
<td style="text-align: right;">0.265</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: right;">-0.40</td>
<td style="text-align: right;">-2.102</td>
<td style="text-align: right;">0.047</td>
<td style="text-align: right;">-0.688</td>
<td style="text-align: right;">-0.007</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: right;">-0.17</td>
<td style="text-align: right;">-0.844</td>
<td style="text-align: right;">0.407</td>
<td style="text-align: right;">-0.532</td>
<td style="text-align: right;">0.238</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>In this analysis, a significant effect on Shannon index with length of hormone replacement therapy could be observed.</p>
</section>
<section id="influence-of-nugent-score-within-patient-group" class="level4">
<h4 class="anchored" data-anchor-id="influence-of-nugent-score-within-patient-group">Influence of Nugent score within patient group</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>adiv<span class="sc">$</span>data <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tukey_hsd</span>(value<span class="sc">~</span>NugentScore) <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(variable, group1, group2, estimate, p.adj, p.adj.signif, conf.low, conf.high) <span class="sc">%&gt;%</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, <span class="at">caption =</span> <span class="st">"Tukey Test on alpha diversity measures by Nugent Score"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Tukey Test on alpha diversity measures by Nugent Score</caption>
<colgroup>
<col style="width: 14%">
<col style="width: 9%">
<col style="width: 15%">
<col style="width: 11%">
<col style="width: 7%">
<col style="width: 16%">
<col style="width: 11%">
<col style="width: 12%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">variable</th>
<th style="text-align: left;">group1</th>
<th style="text-align: left;">group2</th>
<th style="text-align: right;">estimate</th>
<th style="text-align: right;">p.adj</th>
<th style="text-align: left;">p.adj.signif</th>
<th style="text-align: right;">conf.low</th>
<th style="text-align: right;">conf.high</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">0-3</td>
<td style="text-align: left;">4-6</td>
<td style="text-align: right;">26.648</td>
<td style="text-align: right;">0.342</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-15.098</td>
<td style="text-align: right;">68.393</td>
</tr>
<tr class="even">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">0-3</td>
<td style="text-align: left;">7-10</td>
<td style="text-align: right;">-18.067</td>
<td style="text-align: right;">0.929</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-96.166</td>
<td style="text-align: right;">60.033</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">0-3</td>
<td style="text-align: left;">no bacteria</td>
<td style="text-align: right;">54.183</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">***</td>
<td style="text-align: right;">20.292</td>
<td style="text-align: right;">88.074</td>
</tr>
<tr class="even">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">4-6</td>
<td style="text-align: left;">7-10</td>
<td style="text-align: right;">-44.714</td>
<td style="text-align: right;">0.483</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-126.770</td>
<td style="text-align: right;">37.342</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">4-6</td>
<td style="text-align: left;">no bacteria</td>
<td style="text-align: right;">27.536</td>
<td style="text-align: right;">0.323</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-14.682</td>
<td style="text-align: right;">69.753</td>
</tr>
<tr class="even">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">7-10</td>
<td style="text-align: left;">no bacteria</td>
<td style="text-align: right;">72.250</td>
<td style="text-align: right;">0.081</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-6.102</td>
<td style="text-align: right;">150.602</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">0-3</td>
<td style="text-align: left;">4-6</td>
<td style="text-align: right;">1.167</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">***</td>
<td style="text-align: right;">0.394</td>
<td style="text-align: right;">1.939</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">0-3</td>
<td style="text-align: left;">7-10</td>
<td style="text-align: right;">0.152</td>
<td style="text-align: right;">0.993</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-1.293</td>
<td style="text-align: right;">1.597</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">0-3</td>
<td style="text-align: left;">no bacteria</td>
<td style="text-align: right;">1.250</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">****</td>
<td style="text-align: right;">0.623</td>
<td style="text-align: right;">1.877</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">4-6</td>
<td style="text-align: left;">7-10</td>
<td style="text-align: right;">-1.015</td>
<td style="text-align: right;">0.302</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-2.533</td>
<td style="text-align: right;">0.504</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">4-6</td>
<td style="text-align: left;">no bacteria</td>
<td style="text-align: right;">0.083</td>
<td style="text-align: right;">0.992</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-0.698</td>
<td style="text-align: right;">0.865</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">7-10</td>
<td style="text-align: left;">no bacteria</td>
<td style="text-align: right;">1.098</td>
<td style="text-align: right;">0.200</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-0.352</td>
<td style="text-align: right;">2.548</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">0-3</td>
<td style="text-align: left;">4-6</td>
<td style="text-align: right;">5.681</td>
<td style="text-align: right;">0.167</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-1.480</td>
<td style="text-align: right;">12.842</td>
</tr>
<tr class="even">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">0-3</td>
<td style="text-align: left;">7-10</td>
<td style="text-align: right;">-0.403</td>
<td style="text-align: right;">1.000</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-13.800</td>
<td style="text-align: right;">12.994</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">0-3</td>
<td style="text-align: left;">no bacteria</td>
<td style="text-align: right;">7.012</td>
<td style="text-align: right;">0.012</td>
<td style="text-align: left;">*</td>
<td style="text-align: right;">1.198</td>
<td style="text-align: right;">12.825</td>
</tr>
<tr class="even">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">4-6</td>
<td style="text-align: left;">7-10</td>
<td style="text-align: right;">-6.084</td>
<td style="text-align: right;">0.668</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-20.160</td>
<td style="text-align: right;">7.991</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">4-6</td>
<td style="text-align: left;">no bacteria</td>
<td style="text-align: right;">1.330</td>
<td style="text-align: right;">0.963</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-5.911</td>
<td style="text-align: right;">8.572</td>
</tr>
<tr class="even">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">7-10</td>
<td style="text-align: left;">no bacteria</td>
<td style="text-align: right;">7.415</td>
<td style="text-align: right;">0.472</td>
<td style="text-align: left;">ns</td>
<td style="text-align: right;">-6.025</td>
<td style="text-align: right;">20.855</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Notably, if analysis is split by patient group, none of these differences remain significant. This might be due to a lower sample size.</p>
</section>
</section>
</section>
<section id="beta-diversity" class="level2">
<h2 class="anchored" data-anchor-id="beta-diversity">Beta-diversity</h2>
<p>This diversity metric compares different samples or groups of samples regarding their similarity or difference in community composition. The distance matrix can be visualized using dimension reduction techniques (e.g.&nbsp;PCA) or pairwise distances can be calculated. For this analysis, the main hypothesis is, that overall community composition differs significantly between the patient group (i.e.&nbsp;<em>03_trans</em>) and the two control groups. This will be tested using <em>adonis.</em></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># first, filter out low abundance taxa</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>lowabund_filter <span class="ot">=</span> <span class="fu">genefilter_sample</span>(ps_samples, <span class="fu">filterfun_sample</span>(<span class="cf">function</span>(x) x <span class="sc">&gt;</span> <span class="dv">10</span>), <span class="at">A=</span><span class="fl">0.1</span><span class="sc">*</span><span class="fu">nsamples</span>(ps_samples))</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>ps_filtered <span class="ot">=</span> <span class="fu">prune_taxa</span>(lowabund_filter, ps_samples)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>ps_filtered</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 106 taxa and 75 samples ]
sample_data() Sample Data:       [ 75 samples by 15 sample variables ]
tax_table()   Taxonomy Table:    [ 106 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 106 tips and 105 internal nodes ]
refseq()      DNAStringSet:      [ 106 reference sequences ]</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>ps_filtered_rel <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(ps_filtered, <span class="cf">function</span>(x) <span class="dv">100</span> <span class="sc">*</span> x<span class="sc">/</span><span class="fu">sum</span>(x))</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>ord.bray<span class="ot">&lt;-</span> <span class="fu">ordinate</span>(ps_filtered_rel, <span class="st">"PCoA"</span>, <span class="st">"bray"</span>)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>ord.wunifrac <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(ps_filtered_rel, <span class="st">"PCoA"</span>, <span class="st">"unifrac"</span>, <span class="at">weighted =</span> T)</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>ord.unifrac <span class="ot">&lt;-</span> <span class="fu">ordinate</span>(ps_filtered_rel, <span class="st">"PCoA"</span>, <span class="st">"unifrac"</span>, <span class="at">weighted =</span> F)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">=</span> <span class="fu">plot_ordination</span>(ps_filtered_rel, ord.bray, <span class="at">type=</span><span class="st">"samples"</span>, <span class="at">color=</span><span class="st">"grp"</span>) <span class="sc">+</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">type =</span> <span class="st">"qual"</span>,<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Group"</span>, <span class="at">caption =</span> <span class="st">"Bray-Curtis"</span>)</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">=</span> <span class="fu">plot_ordination</span>(ps_filtered_rel, ord.wunifrac, <span class="at">type=</span><span class="st">"samples"</span>, <span class="at">color=</span><span class="st">"grp"</span>) <span class="sc">+</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">type =</span> <span class="st">"qual"</span>,<span class="at">palette =</span> <span class="st">"Dark2"</span>) <span class="sc">+</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Group"</span>, <span class="at">caption =</span> <span class="st">"Weighted Unifrac"</span>)</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">=</span> <span class="fu">plot_ordination</span>(ps_filtered_rel, ord.unifrac, <span class="at">type=</span><span class="st">"samples"</span>, <span class="at">color=</span><span class="st">"grp"</span>) <span class="sc">+</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">type =</span> <span class="st">"qual"</span>,<span class="at">palette =</span> <span class="st">"Dark2"</span>)<span class="sc">+</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>() <span class="sc">+</span></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Group"</span>, <span class="at">caption =</span> <span class="st">"Unweighted Unifrac"</span>)</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a><span class="co"># extract the legend from one of the plots</span></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>legend_b <span class="ot">&lt;-</span> <span class="fu">get_legend</span>(</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>  p1 <span class="sc">+</span> </span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">nrow =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>plot_ordinations <span class="ot">=</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"none"</span>),</span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>                   p2 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"none"</span>),</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a>                   p3 <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"none"</span>),</span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span>  <span class="st">"AUTO"</span>, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(plot_ordinations, legend_b, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, .<span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/export%20plot-1.pdf" class="img-fluid"></p>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>D_BC <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(ps_filtered_rel, <span class="st">"bray"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>D_UF <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(ps_filtered_rel,  <span class="st">"unifrac"</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>D_wUF <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(ps_filtered_rel, <span class="st">"wunifrac"</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>adonis_bc <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(D_BC <span class="sc">~</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>grp)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>adonis_pw_bc <span class="ot">=</span> <span class="fu">pairwise.adonis</span>(D_BC, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>grp)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>adonis_uf <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(D_UF <span class="sc">~</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>grp)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>adonis_pw_uf <span class="ot">=</span> <span class="fu">pairwise.adonis</span>(D_UF, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>grp)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>adonis_wUF <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(D_wUF <span class="sc">~</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>grp)</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>adonis_pw_wUF <span class="ot">=</span> <span class="fu">pairwise.adonis</span>(D_wUF, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>grp)</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(adonis_pw_bc, <span class="at">digits =</span> <span class="dv">3</span>, <span class="at">caption =</span> <span class="st">"Bray-Curtis"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Bray-Curtis</caption>
<colgroup>
<col style="width: 43%">
<col style="width: 3%">
<col style="width: 11%">
<col style="width: 9%">
<col style="width: 6%">
<col style="width: 9%">
<col style="width: 12%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">pairs</th>
<th style="text-align: right;">Df</th>
<th style="text-align: right;">SumsOfSqs</th>
<th style="text-align: right;">F.Model</th>
<th style="text-align: right;">R2</th>
<th style="text-align: right;">p.value</th>
<th style="text-align: right;">p.adjusted</th>
<th style="text-align: left;">sig</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">01_postmenopausal vs 02_premenopausal</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.839</td>
<td style="text-align: right;">4.972</td>
<td style="text-align: right;">0.094</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="even">
<td style="text-align: left;">01_postmenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.255</td>
<td style="text-align: right;">3.474</td>
<td style="text-align: right;">0.067</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="odd">
<td style="text-align: left;">02_premenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3.351</td>
<td style="text-align: right;">10.303</td>
<td style="text-align: right;">0.177</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(adonis_pw_uf, <span class="at">digits =</span> <span class="dv">3</span>, <span class="at">caption =</span><span class="st">"Unweigthed Unifrac"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Unweigthed Unifrac</caption>
<colgroup>
<col style="width: 43%">
<col style="width: 3%">
<col style="width: 11%">
<col style="width: 9%">
<col style="width: 6%">
<col style="width: 9%">
<col style="width: 12%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">pairs</th>
<th style="text-align: right;">Df</th>
<th style="text-align: right;">SumsOfSqs</th>
<th style="text-align: right;">F.Model</th>
<th style="text-align: right;">R2</th>
<th style="text-align: right;">p.value</th>
<th style="text-align: right;">p.adjusted</th>
<th style="text-align: left;">sig</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">01_postmenopausal vs 02_premenopausal</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.104</td>
<td style="text-align: right;">5.145</td>
<td style="text-align: right;">0.097</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="even">
<td style="text-align: left;">01_postmenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.704</td>
<td style="text-align: right;">4.133</td>
<td style="text-align: right;">0.079</td>
<td style="text-align: right;">0.004</td>
<td style="text-align: right;">0.012</td>
<td style="text-align: left;">.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">02_premenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2.263</td>
<td style="text-align: right;">12.762</td>
<td style="text-align: right;">0.210</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(adonis_pw_wUF, <span class="at">digits =</span> <span class="dv">3</span>, <span class="at">caption =</span> <span class="st">"Weighted Unifrac"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Weighted Unifrac</caption>
<colgroup>
<col style="width: 43%">
<col style="width: 3%">
<col style="width: 11%">
<col style="width: 9%">
<col style="width: 6%">
<col style="width: 9%">
<col style="width: 12%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">pairs</th>
<th style="text-align: right;">Df</th>
<th style="text-align: right;">SumsOfSqs</th>
<th style="text-align: right;">F.Model</th>
<th style="text-align: right;">R2</th>
<th style="text-align: right;">p.value</th>
<th style="text-align: right;">p.adjusted</th>
<th style="text-align: left;">sig</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">01_postmenopausal vs 02_premenopausal</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.559</td>
<td style="text-align: right;">8.149</td>
<td style="text-align: right;">0.145</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="even">
<td style="text-align: left;">01_postmenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.413</td>
<td style="text-align: right;">5.341</td>
<td style="text-align: right;">0.100</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="odd">
<td style="text-align: left;">02_premenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.369</td>
<td style="text-align: right;">23.593</td>
<td style="text-align: right;">0.330</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>
<section id="differential-abundance" class="level1">
<h1>Differential abundance</h1>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">phyloseq_to_deseq2</span>(ps_filtered, <span class="sc">~</span> grp)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>dds<span class="sc">$</span>grp<span class="ot">&lt;-</span><span class="fu">relevel</span>(dds<span class="sc">$</span>grp,<span class="at">ref=</span><span class="st">"03_trans-men"</span>)</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>ds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds, <span class="at">type=</span><span class="st">"poscounts"</span>)</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>ds <span class="ot">=</span> <span class="fu">DESeq</span>(ds, <span class="at">test=</span><span class="st">"Wald"</span>, <span class="at">fitType=</span><span class="st">"parametric"</span>)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>alpha <span class="ot">=</span> <span class="fl">0.01</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>res_premenopausal <span class="ot">=</span> <span class="fu">results</span>(ds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"grp"</span>, <span class="st">"02_premenopausal"</span>, <span class="st">"03_trans-men"</span>))</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>taxa_sig_premenopausal <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">as.data.frame</span>(res_premenopausal), padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ASV"</span>)</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>taxa_table <span class="ot">=</span> <span class="fu">tax_table</span>(ps_filtered) <span class="sc">%&gt;%</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ASV"</span>)</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>diffabund_pre <span class="ot">=</span> <span class="fu">left_join</span>(taxa_sig_premenopausal, taxa_table, <span class="at">by =</span> <span class="st">"ASV"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Species) <span class="sc">%&gt;%</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_csv2</span>(<span class="st">"results/microbiome/deseq_diff_trans-premenopausal.csv"</span>)</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(diffabund_pre, <span class="at">digits =</span> <span class="dv">3</span>, <span class="at">row.names =</span> F, <span class="at">caption =</span> <span class="st">"Differential abundance between premenopausal women and trans-men"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Differential abundance between premenopausal women and trans-men</caption>
<colgroup>
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 8%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 9%">
<col style="width: 10%">
<col style="width: 19%">
<col style="width: 12%">
<col style="width: 11%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">ASV</th>
<th style="text-align: right;">baseMean</th>
<th style="text-align: right;">log2FoldChange</th>
<th style="text-align: right;">lfcSE</th>
<th style="text-align: right;">stat</th>
<th style="text-align: right;">pvalue</th>
<th style="text-align: right;">padj</th>
<th style="text-align: left;">Kingdom</th>
<th style="text-align: left;">Phylum</th>
<th style="text-align: left;">Class</th>
<th style="text-align: left;">Order</th>
<th style="text-align: left;">Family</th>
<th style="text-align: left;">Genus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ASV1</td>
<td style="text-align: right;">8836.794</td>
<td style="text-align: right;">7.120</td>
<td style="text-align: right;">1.386</td>
<td style="text-align: right;">5.137</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">Lactobacillus</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV2</td>
<td style="text-align: right;">3474.423</td>
<td style="text-align: right;">4.994</td>
<td style="text-align: right;">1.272</td>
<td style="text-align: right;">3.925</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">Lactobacillus</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV3</td>
<td style="text-align: right;">296.709</td>
<td style="text-align: right;">-3.374</td>
<td style="text-align: right;">0.873</td>
<td style="text-align: right;">-3.867</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Prevotellaceae</td>
<td style="text-align: left;">Prevotella</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV6</td>
<td style="text-align: right;">23.859</td>
<td style="text-align: right;">-5.594</td>
<td style="text-align: right;">2.258</td>
<td style="text-align: right;">-2.478</td>
<td style="text-align: right;">0.013</td>
<td style="text-align: right;">0.027</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Streptococcaceae</td>
<td style="text-align: left;">Streptococcus</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV8</td>
<td style="text-align: right;">178.493</td>
<td style="text-align: right;">5.510</td>
<td style="text-align: right;">1.310</td>
<td style="text-align: right;">4.206</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">Lactobacillus</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV9</td>
<td style="text-align: right;">97.075</td>
<td style="text-align: right;">-5.076</td>
<td style="text-align: right;">1.257</td>
<td style="text-align: right;">-4.037</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Streptococcaceae</td>
<td style="text-align: left;">Streptococcus</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV10</td>
<td style="text-align: right;">145.660</td>
<td style="text-align: right;">-3.843</td>
<td style="text-align: right;">1.028</td>
<td style="text-align: right;">-3.739</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Prevotellaceae</td>
<td style="text-align: left;">Prevotella</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV11</td>
<td style="text-align: right;">182.749</td>
<td style="text-align: right;">2.610</td>
<td style="text-align: right;">1.052</td>
<td style="text-align: right;">2.480</td>
<td style="text-align: right;">0.013</td>
<td style="text-align: right;">0.027</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Coriobacteriia</td>
<td style="text-align: left;">Coriobacteriales</td>
<td style="text-align: left;">Atopobiaceae</td>
<td style="text-align: left;">Atopobium</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV12</td>
<td style="text-align: right;">193.599</td>
<td style="text-align: right;">-1.658</td>
<td style="text-align: right;">0.716</td>
<td style="text-align: right;">-2.314</td>
<td style="text-align: right;">0.021</td>
<td style="text-align: right;">0.041</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Finegoldia</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV13</td>
<td style="text-align: right;">47.479</td>
<td style="text-align: right;">4.235</td>
<td style="text-align: right;">1.757</td>
<td style="text-align: right;">2.410</td>
<td style="text-align: right;">0.016</td>
<td style="text-align: right;">0.033</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Proteobacteria</td>
<td style="text-align: left;">Gammaproteobacteria</td>
<td style="text-align: left;">Enterobacterales</td>
<td style="text-align: left;">Enterobacteriaceae</td>
<td style="text-align: left;">Escherichia-Shigella</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV15</td>
<td style="text-align: right;">66.480</td>
<td style="text-align: right;">-4.200</td>
<td style="text-align: right;">1.324</td>
<td style="text-align: right;">-3.172</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.005</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Streptococcaceae</td>
<td style="text-align: left;">Streptococcus</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV16</td>
<td style="text-align: right;">181.265</td>
<td style="text-align: right;">6.444</td>
<td style="text-align: right;">1.883</td>
<td style="text-align: right;">3.422</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">Lactobacillus</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV17</td>
<td style="text-align: right;">80.800</td>
<td style="text-align: right;">-3.940</td>
<td style="text-align: right;">1.387</td>
<td style="text-align: right;">-2.841</td>
<td style="text-align: right;">0.004</td>
<td style="text-align: right;">0.012</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Negativicutes</td>
<td style="text-align: left;">Veillonellales-Selenomonadales</td>
<td style="text-align: left;">Veillonellaceae</td>
<td style="text-align: left;">Dialister</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV20</td>
<td style="text-align: right;">6.251</td>
<td style="text-align: right;">-4.108</td>
<td style="text-align: right;">1.783</td>
<td style="text-align: right;">-2.304</td>
<td style="text-align: right;">0.021</td>
<td style="text-align: right;">0.041</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Porphyromonadaceae</td>
<td style="text-align: left;">Porphyromonas</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV21</td>
<td style="text-align: right;">40.863</td>
<td style="text-align: right;">-4.779</td>
<td style="text-align: right;">1.255</td>
<td style="text-align: right;">-3.808</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Actinobacteria</td>
<td style="text-align: left;">Bifidobacteriales</td>
<td style="text-align: left;">Bifidobacteriaceae</td>
<td style="text-align: left;">Alloscardovia</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV23</td>
<td style="text-align: right;">44.563</td>
<td style="text-align: right;">-6.795</td>
<td style="text-align: right;">1.150</td>
<td style="text-align: right;">-5.909</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Fenollaria</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV24</td>
<td style="text-align: right;">62.469</td>
<td style="text-align: right;">-2.718</td>
<td style="text-align: right;">0.675</td>
<td style="text-align: right;">-4.030</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Peptoniphilus</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV25</td>
<td style="text-align: right;">54.770</td>
<td style="text-align: right;">-3.289</td>
<td style="text-align: right;">0.916</td>
<td style="text-align: right;">-3.590</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Negativicutes</td>
<td style="text-align: left;">Veillonellales-Selenomonadales</td>
<td style="text-align: left;">Veillonellaceae</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV29</td>
<td style="text-align: right;">43.379</td>
<td style="text-align: right;">-4.231</td>
<td style="text-align: right;">1.647</td>
<td style="text-align: right;">-2.569</td>
<td style="text-align: right;">0.010</td>
<td style="text-align: right;">0.023</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Prevotellaceae</td>
<td style="text-align: left;">Prevotella</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV31</td>
<td style="text-align: right;">46.531</td>
<td style="text-align: right;">-3.379</td>
<td style="text-align: right;">1.161</td>
<td style="text-align: right;">-2.909</td>
<td style="text-align: right;">0.004</td>
<td style="text-align: right;">0.010</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Anaerococcus</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV32</td>
<td style="text-align: right;">40.807</td>
<td style="text-align: right;">-3.842</td>
<td style="text-align: right;">1.703</td>
<td style="text-align: right;">-2.256</td>
<td style="text-align: right;">0.024</td>
<td style="text-align: right;">0.045</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Prevotellaceae</td>
<td style="text-align: left;">Prevotella</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV34</td>
<td style="text-align: right;">41.188</td>
<td style="text-align: right;">-3.255</td>
<td style="text-align: right;">1.023</td>
<td style="text-align: right;">-3.182</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.005</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Campylobacterota</td>
<td style="text-align: left;">Campylobacteria</td>
<td style="text-align: left;">Campylobacterales</td>
<td style="text-align: left;">Campylobacteraceae</td>
<td style="text-align: left;">Campylobacter</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV36</td>
<td style="text-align: right;">32.757</td>
<td style="text-align: right;">-5.942</td>
<td style="text-align: right;">1.086</td>
<td style="text-align: right;">-5.472</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Negativicutes</td>
<td style="text-align: left;">Veillonellales-Selenomonadales</td>
<td style="text-align: left;">Veillonellaceae</td>
<td style="text-align: left;">Dialister</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV39</td>
<td style="text-align: right;">32.429</td>
<td style="text-align: right;">-7.577</td>
<td style="text-align: right;">1.332</td>
<td style="text-align: right;">-5.686</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Anaerococcus</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV40</td>
<td style="text-align: right;">36.919</td>
<td style="text-align: right;">-4.847</td>
<td style="text-align: right;">1.452</td>
<td style="text-align: right;">-3.339</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Prevotellaceae</td>
<td style="text-align: left;">Prevotella</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV45</td>
<td style="text-align: right;">30.382</td>
<td style="text-align: right;">-6.236</td>
<td style="text-align: right;">1.501</td>
<td style="text-align: right;">-4.154</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Anaerococcus</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV46</td>
<td style="text-align: right;">32.384</td>
<td style="text-align: right;">-4.136</td>
<td style="text-align: right;">1.161</td>
<td style="text-align: right;">-3.564</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Anaerococcus</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV47</td>
<td style="text-align: right;">18.977</td>
<td style="text-align: right;">-24.332</td>
<td style="text-align: right;">1.463</td>
<td style="text-align: right;">-16.635</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Actinobacteria</td>
<td style="text-align: left;">Actinomycetales</td>
<td style="text-align: left;">Actinomycetaceae</td>
<td style="text-align: left;">Actinomyces</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV48</td>
<td style="text-align: right;">25.838</td>
<td style="text-align: right;">-5.508</td>
<td style="text-align: right;">1.108</td>
<td style="text-align: right;">-4.972</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Ezakiella</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV54</td>
<td style="text-align: right;">7.315</td>
<td style="text-align: right;">-5.179</td>
<td style="text-align: right;">1.934</td>
<td style="text-align: right;">-2.677</td>
<td style="text-align: right;">0.007</td>
<td style="text-align: right;">0.018</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Prevotellaceae</td>
<td style="text-align: left;">Prevotella</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV56</td>
<td style="text-align: right;">90.308</td>
<td style="text-align: right;">7.488</td>
<td style="text-align: right;">1.696</td>
<td style="text-align: right;">4.415</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">Lactobacillus</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV57</td>
<td style="text-align: right;">25.517</td>
<td style="text-align: right;">-24.675</td>
<td style="text-align: right;">1.854</td>
<td style="text-align: right;">-13.312</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Parvimonas</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV62</td>
<td style="text-align: right;">16.716</td>
<td style="text-align: right;">-3.660</td>
<td style="text-align: right;">1.386</td>
<td style="text-align: right;">-2.641</td>
<td style="text-align: right;">0.008</td>
<td style="text-align: right;">0.019</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Peptoniphilus</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV64</td>
<td style="text-align: right;">17.221</td>
<td style="text-align: right;">-6.351</td>
<td style="text-align: right;">2.204</td>
<td style="text-align: right;">-2.882</td>
<td style="text-align: right;">0.004</td>
<td style="text-align: right;">0.011</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Negativicutes</td>
<td style="text-align: left;">Veillonellales-Selenomonadales</td>
<td style="text-align: left;">Veillonellaceae</td>
<td style="text-align: left;">Dialister</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV71</td>
<td style="text-align: right;">8.391</td>
<td style="text-align: right;">-4.030</td>
<td style="text-align: right;">0.983</td>
<td style="text-align: right;">-4.099</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Lachnospirales</td>
<td style="text-align: left;">Lachnospiraceae</td>
<td style="text-align: left;">Howardella</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV73</td>
<td style="text-align: right;">21.772</td>
<td style="text-align: right;">-4.249</td>
<td style="text-align: right;">1.174</td>
<td style="text-align: right;">-3.619</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Peptostreptococcaceae</td>
<td style="text-align: left;">Peptostreptococcus</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV74</td>
<td style="text-align: right;">3.931</td>
<td style="text-align: right;">-5.092</td>
<td style="text-align: right;">1.888</td>
<td style="text-align: right;">-2.697</td>
<td style="text-align: right;">0.007</td>
<td style="text-align: right;">0.017</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Prevotellaceae</td>
<td style="text-align: left;">Prevotella</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV76</td>
<td style="text-align: right;">11.488</td>
<td style="text-align: right;">-6.613</td>
<td style="text-align: right;">1.847</td>
<td style="text-align: right;">-3.580</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Porphyromonadaceae</td>
<td style="text-align: left;">Porphyromonas</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV79</td>
<td style="text-align: right;">56.695</td>
<td style="text-align: right;">7.955</td>
<td style="text-align: right;">1.627</td>
<td style="text-align: right;">4.890</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">Lactobacillus</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV80</td>
<td style="text-align: right;">28.402</td>
<td style="text-align: right;">8.726</td>
<td style="text-align: right;">2.963</td>
<td style="text-align: right;">2.945</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.009</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Mycoplasmatales</td>
<td style="text-align: left;">Mycoplasmataceae</td>
<td style="text-align: left;">Ureaplasma</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV90</td>
<td style="text-align: right;">8.425</td>
<td style="text-align: right;">-5.328</td>
<td style="text-align: right;">2.011</td>
<td style="text-align: right;">-2.649</td>
<td style="text-align: right;">0.008</td>
<td style="text-align: right;">0.019</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Negativicutes</td>
<td style="text-align: left;">Veillonellales-Selenomonadales</td>
<td style="text-align: left;">Veillonellaceae</td>
<td style="text-align: left;">Dialister</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV91</td>
<td style="text-align: right;">14.058</td>
<td style="text-align: right;">-23.131</td>
<td style="text-align: right;">2.141</td>
<td style="text-align: right;">-10.806</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Fusobacteriota</td>
<td style="text-align: left;">Fusobacteriia</td>
<td style="text-align: left;">Fusobacteriales</td>
<td style="text-align: left;">Fusobacteriaceae</td>
<td style="text-align: left;">Fusobacterium</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV96</td>
<td style="text-align: right;">6.234</td>
<td style="text-align: right;">-4.081</td>
<td style="text-align: right;">1.749</td>
<td style="text-align: right;">-2.333</td>
<td style="text-align: right;">0.020</td>
<td style="text-align: right;">0.039</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Prevotellaceae</td>
<td style="text-align: left;">Prevotella</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV97</td>
<td style="text-align: right;">7.375</td>
<td style="text-align: right;">-6.026</td>
<td style="text-align: right;">2.031</td>
<td style="text-align: right;">-2.968</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.009</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Lachnospirales</td>
<td style="text-align: left;">Lachnospiraceae</td>
<td style="text-align: left;">Moryella</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV100</td>
<td style="text-align: right;">20.126</td>
<td style="text-align: right;">6.918</td>
<td style="text-align: right;">2.032</td>
<td style="text-align: right;">3.404</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Fusobacteriota</td>
<td style="text-align: left;">Fusobacteriia</td>
<td style="text-align: left;">Fusobacteriales</td>
<td style="text-align: left;">Leptotrichiaceae</td>
<td style="text-align: left;">Sneathia</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV101</td>
<td style="text-align: right;">8.760</td>
<td style="text-align: right;">-5.947</td>
<td style="text-align: right;">1.790</td>
<td style="text-align: right;">-3.322</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Negativicutes</td>
<td style="text-align: left;">Veillonellales-Selenomonadales</td>
<td style="text-align: left;">Veillonellaceae</td>
<td style="text-align: left;">Dialister</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV102</td>
<td style="text-align: right;">14.223</td>
<td style="text-align: right;">-24.611</td>
<td style="text-align: right;">2.185</td>
<td style="text-align: right;">-11.264</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Porphyromonadaceae</td>
<td style="text-align: left;">Porphyromonas</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV103</td>
<td style="text-align: right;">9.389</td>
<td style="text-align: right;">-5.048</td>
<td style="text-align: right;">1.108</td>
<td style="text-align: right;">-4.557</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Clostridia</td>
<td style="text-align: left;">Peptostreptococcales-Tissierellales</td>
<td style="text-align: left;">Family XI</td>
<td style="text-align: left;">Peptoniphilus</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV106</td>
<td style="text-align: right;">11.386</td>
<td style="text-align: right;">-5.077</td>
<td style="text-align: right;">1.796</td>
<td style="text-align: right;">-2.827</td>
<td style="text-align: right;">0.005</td>
<td style="text-align: right;">0.012</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Porphyromonadaceae</td>
<td style="text-align: left;">Porphyromonas</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV113</td>
<td style="text-align: right;">38.903</td>
<td style="text-align: right;">7.100</td>
<td style="text-align: right;">1.758</td>
<td style="text-align: right;">4.038</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Actinobacteria</td>
<td style="text-align: left;">Bifidobacteriales</td>
<td style="text-align: left;">Bifidobacteriaceae</td>
<td style="text-align: left;">Gardnerella</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV119</td>
<td style="text-align: right;">22.514</td>
<td style="text-align: right;">6.384</td>
<td style="text-align: right;">1.676</td>
<td style="text-align: right;">3.809</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">HT002</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV120</td>
<td style="text-align: right;">6.855</td>
<td style="text-align: right;">-5.491</td>
<td style="text-align: right;">1.449</td>
<td style="text-align: right;">-3.789</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Coriobacteriia</td>
<td style="text-align: left;">Coriobacteriales</td>
<td style="text-align: left;">Atopobiaceae</td>
<td style="text-align: left;">Atopobium</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV130</td>
<td style="text-align: right;">8.747</td>
<td style="text-align: right;">-5.309</td>
<td style="text-align: right;">1.848</td>
<td style="text-align: right;">-2.873</td>
<td style="text-align: right;">0.004</td>
<td style="text-align: right;">0.011</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Fusobacteriota</td>
<td style="text-align: left;">Fusobacteriia</td>
<td style="text-align: left;">Fusobacteriales</td>
<td style="text-align: left;">Fusobacteriaceae</td>
<td style="text-align: left;">Fusobacterium</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV137</td>
<td style="text-align: right;">5.448</td>
<td style="text-align: right;">-5.111</td>
<td style="text-align: right;">2.268</td>
<td style="text-align: right;">-2.254</td>
<td style="text-align: right;">0.024</td>
<td style="text-align: right;">0.045</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Porphyromonadaceae</td>
<td style="text-align: left;">Porphyromonas</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV151</td>
<td style="text-align: right;">4.109</td>
<td style="text-align: right;">-4.040</td>
<td style="text-align: right;">1.309</td>
<td style="text-align: right;">-3.087</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.006</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Proteobacteria</td>
<td style="text-align: left;">Gammaproteobacteria</td>
<td style="text-align: left;">Burkholderiales</td>
<td style="text-align: left;">Sutterellaceae</td>
<td style="text-align: left;">Sutterella</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV168</td>
<td style="text-align: right;">3.334</td>
<td style="text-align: right;">-4.802</td>
<td style="text-align: right;">1.919</td>
<td style="text-align: right;">-2.503</td>
<td style="text-align: right;">0.012</td>
<td style="text-align: right;">0.027</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Porphyromonadaceae</td>
<td style="text-align: left;">Porphyromonas</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV181</td>
<td style="text-align: right;">4.533</td>
<td style="text-align: right;">-4.835</td>
<td style="text-align: right;">1.853</td>
<td style="text-align: right;">-2.609</td>
<td style="text-align: right;">0.009</td>
<td style="text-align: right;">0.020</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Porphyromonadaceae</td>
<td style="text-align: left;">Porphyromonas</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># repeat analysis for postmenopausal group</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>res_postmenopausal <span class="ot">=</span> <span class="fu">results</span>(ds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"grp"</span>, <span class="st">"01_postmenopausal"</span>, <span class="st">"03_trans-men"</span>))</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>taxa_sig_postmenopausal <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">as.data.frame</span>(res_postmenopausal), padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ASV"</span>)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>diffabund_post <span class="ot">=</span> <span class="fu">left_join</span>(taxa_sig_postmenopausal, taxa_table, <span class="at">by =</span> <span class="st">"ASV"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_csv2</span>(<span class="st">"results/microbiome/deseq_diff_trans-postmenopausal.csv"</span>)</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(diffabund_post, <span class="at">digits =</span> <span class="dv">3</span>, <span class="at">row.names =</span> F, <span class="at">caption =</span> <span class="st">"Differential abundance between postmenopausal women and trans-men"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Differential abundance between postmenopausal women and trans-men</caption>
<colgroup>
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 8%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 9%">
<col style="width: 10%">
<col style="width: 16%">
<col style="width: 10%">
<col style="width: 11%">
<col style="width: 5%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">ASV</th>
<th style="text-align: right;">baseMean</th>
<th style="text-align: right;">log2FoldChange</th>
<th style="text-align: right;">lfcSE</th>
<th style="text-align: right;">stat</th>
<th style="text-align: right;">pvalue</th>
<th style="text-align: right;">padj</th>
<th style="text-align: left;">Kingdom</th>
<th style="text-align: left;">Phylum</th>
<th style="text-align: left;">Class</th>
<th style="text-align: left;">Order</th>
<th style="text-align: left;">Family</th>
<th style="text-align: left;">Genus</th>
<th style="text-align: left;">Species</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ASV1</td>
<td style="text-align: right;">8836.794</td>
<td style="text-align: right;">7.158</td>
<td style="text-align: right;">1.386</td>
<td style="text-align: right;">5.165</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">Lactobacillus</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV2</td>
<td style="text-align: right;">3474.423</td>
<td style="text-align: right;">3.901</td>
<td style="text-align: right;">1.272</td>
<td style="text-align: right;">3.066</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.021</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">Lactobacillus</td>
<td style="text-align: left;">iners</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV3</td>
<td style="text-align: right;">296.709</td>
<td style="text-align: right;">-3.282</td>
<td style="text-align: right;">0.873</td>
<td style="text-align: right;">-3.760</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.004</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Prevotellaceae</td>
<td style="text-align: left;">Prevotella</td>
<td style="text-align: left;">disiens</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV6</td>
<td style="text-align: right;">23.859</td>
<td style="text-align: right;">-7.935</td>
<td style="text-align: right;">2.269</td>
<td style="text-align: right;">-3.496</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.009</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Streptococcaceae</td>
<td style="text-align: left;">Streptococcus</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV11</td>
<td style="text-align: right;">182.749</td>
<td style="text-align: right;">2.839</td>
<td style="text-align: right;">1.052</td>
<td style="text-align: right;">2.698</td>
<td style="text-align: right;">0.007</td>
<td style="text-align: right;">0.046</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Coriobacteriia</td>
<td style="text-align: left;">Coriobacteriales</td>
<td style="text-align: left;">Atopobiaceae</td>
<td style="text-align: left;">Atopobium</td>
<td style="text-align: left;">vaginae</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV13</td>
<td style="text-align: right;">47.479</td>
<td style="text-align: right;">5.899</td>
<td style="text-align: right;">1.756</td>
<td style="text-align: right;">3.359</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.012</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Proteobacteria</td>
<td style="text-align: left;">Gammaproteobacteria</td>
<td style="text-align: left;">Enterobacterales</td>
<td style="text-align: left;">Enterobacteriaceae</td>
<td style="text-align: left;">Escherichia-Shigella</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV14</td>
<td style="text-align: right;">56.671</td>
<td style="text-align: right;">-4.193</td>
<td style="text-align: right;">1.430</td>
<td style="text-align: right;">-2.933</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: right;">0.030</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Actinobacteria</td>
<td style="text-align: left;">Bifidobacteriales</td>
<td style="text-align: left;">Bifidobacteriaceae</td>
<td style="text-align: left;">Gardnerella</td>
<td style="text-align: left;">vaginalis</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV16</td>
<td style="text-align: right;">181.265</td>
<td style="text-align: right;">-22.787</td>
<td style="text-align: right;">1.927</td>
<td style="text-align: right;">-11.827</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">Lactobacillus</td>
<td style="text-align: left;">jensenii</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV17</td>
<td style="text-align: right;">80.800</td>
<td style="text-align: right;">-3.860</td>
<td style="text-align: right;">1.387</td>
<td style="text-align: right;">-2.783</td>
<td style="text-align: right;">0.005</td>
<td style="text-align: right;">0.038</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Negativicutes</td>
<td style="text-align: left;">Veillonellales-Selenomonadales</td>
<td style="text-align: left;">Veillonellaceae</td>
<td style="text-align: left;">Dialister</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV38</td>
<td style="text-align: right;">11.855</td>
<td style="text-align: right;">8.320</td>
<td style="text-align: right;">1.994</td>
<td style="text-align: right;">4.173</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Actinobacteria</td>
<td style="text-align: left;">Bifidobacteriales</td>
<td style="text-align: left;">Bifidobacteriaceae</td>
<td style="text-align: left;">Bifidobacterium</td>
<td style="text-align: left;">breve</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV83</td>
<td style="text-align: right;">9.933</td>
<td style="text-align: right;">-4.505</td>
<td style="text-align: right;">1.618</td>
<td style="text-align: right;">-2.784</td>
<td style="text-align: right;">0.005</td>
<td style="text-align: right;">0.038</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Porphyromonadaceae</td>
<td style="text-align: left;">Porphyromonas</td>
<td style="text-align: left;">uenonis</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV86</td>
<td style="text-align: right;">25.494</td>
<td style="text-align: right;">7.419</td>
<td style="text-align: right;">2.129</td>
<td style="text-align: right;">3.484</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0.009</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Actinobacteria</td>
<td style="text-align: left;">Corynebacteriales</td>
<td style="text-align: left;">Corynebacteriaceae</td>
<td style="text-align: left;">Corynebacterium</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV99</td>
<td style="text-align: right;">9.406</td>
<td style="text-align: right;">-5.846</td>
<td style="text-align: right;">1.891</td>
<td style="text-align: right;">-3.091</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.021</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Prevotellaceae</td>
<td style="text-align: left;">Prevotella</td>
<td style="text-align: left;">corporis</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV151</td>
<td style="text-align: right;">4.109</td>
<td style="text-align: right;">-4.043</td>
<td style="text-align: right;">1.301</td>
<td style="text-align: right;">-3.107</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.021</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Proteobacteria</td>
<td style="text-align: left;">Gammaproteobacteria</td>
<td style="text-align: left;">Burkholderiales</td>
<td style="text-align: left;">Sutterellaceae</td>
<td style="text-align: left;">Sutterella</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ASV181</td>
<td style="text-align: right;">4.533</td>
<td style="text-align: right;">-5.755</td>
<td style="text-align: right;">1.855</td>
<td style="text-align: right;">-3.103</td>
<td style="text-align: right;">0.002</td>
<td style="text-align: right;">0.021</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Bacteroidota</td>
<td style="text-align: left;">Bacteroidia</td>
<td style="text-align: left;">Bacteroidales</td>
<td style="text-align: left;">Porphyromonadaceae</td>
<td style="text-align: left;">Porphyromonas</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV228</td>
<td style="text-align: right;">5.726</td>
<td style="text-align: right;">-4.593</td>
<td style="text-align: right;">1.645</td>
<td style="text-align: right;">-2.793</td>
<td style="text-align: right;">0.005</td>
<td style="text-align: right;">0.038</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Coriobacteriia</td>
<td style="text-align: left;">Coriobacteriales</td>
<td style="text-align: left;">Eggerthellaceae</td>
<td style="text-align: left;">DNF00809</td>
<td style="text-align: left;">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>asvs <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">c</span>(diffabund_pre<span class="sc">$</span>ASV, diffabund_post<span class="sc">$</span>ASV)) <span class="co"># get asvs identified by Deseq2</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co"># next, melt the phyloseq object into a data.frame and filter OTUs for significant ASVs as identified by DESEQ2</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>ps_sig_melt <span class="ot">=</span> <span class="fu">psmelt</span>(ps_filtered_rel) <span class="sc">%&gt;%</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(OTU <span class="sc">%in%</span> asvs)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize Abundance at the genus level for plotting, almost no species are assigned in this dataset</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>ps_sig_melt2 <span class="ot">=</span> ps_sig_melt <span class="sc">%&gt;%</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Sample, Genus, grp) <span class="sc">%&gt;%</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Abundance =</span> <span class="fu">sum</span>(Abundance))</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the final figure panel</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>deseq_res_fig <span class="ot">=</span> <span class="fu">ggplot</span>(ps_sig_melt2, <span class="fu">aes</span>(<span class="at">x =</span> grp, <span class="at">y=</span>Abundance)) <span class="sc">+</span> </span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.3</span>, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> grp), <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Genus, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">ncol =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_sqrt</span>() <span class="sc">+</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">fill =</span> <span class="st">"Group"</span>, <span class="at">y=</span> <span class="st">"Abundance (%)"</span>) <span class="sc">+</span></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.2</span>))) <span class="sc">+</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> custom_pal)</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>deseq_res_fig</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/plot%20deseq2%20taxa-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"MbVagTrans_Results_files/figure-html/deseq_taxa_final.pdf"</span>, <span class="at">plot =</span> deseq_res_fig, </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> <span class="st">"pdf"</span>, <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">8</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># summary statistics</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>summarystat1 <span class="ot">=</span> ps_sig_melt2 <span class="sc">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Genus, grp) <span class="sc">%&gt;%</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean =</span> <span class="fu">mean</span>(Abundance, <span class="at">na.rm=</span>T),</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">median =</span> <span class="fu">median</span>(Abundance, <span class="at">na.rm =</span>T),</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">sd =</span> <span class="fu">sd</span>(Abundance, <span class="at">na.rm =</span> T),</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">min =</span> <span class="fu">min</span>(Abundance, <span class="at">na.rm =</span>T),</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">max =</span> <span class="fu">max</span>(Abundance, <span class="at">na.rm =</span>T))</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv2</span>(<span class="at">x =</span> summarystat1, <span class="st">"results/microbiome/deseq_summarystat_complete.csv"</span>)</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>summarystat2 <span class="ot">=</span> ps_sig_melt2 <span class="sc">%&gt;%</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Genus, grp) <span class="sc">%&gt;%</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean =</span> <span class="fu">mean</span>(Abundance, <span class="at">na.rm=</span>T) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>            <span class="at">sd =</span> <span class="fu">sd</span>(Abundance, <span class="at">na.rm =</span> T) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="at">digits =</span> <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stat =</span> <span class="fu">paste</span>(mean, <span class="st">"±"</span>, sd)) <span class="sc">%&gt;%</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Genus, grp, stat) <span class="sc">%&gt;%</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">"Genus"</span>, <span class="at">names_from =</span> grp, <span class="at">values_from =</span> stat)</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv2</span>(summarystat2, <span class="st">"results/microbiome/deseq_summarystat_complete.csv"</span>)</span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(summarystat1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 22%">
<col style="width: 19%">
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 10%">
<col style="width: 11%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Genus</th>
<th style="text-align: left;">grp</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">median</th>
<th style="text-align: right;">sd</th>
<th style="text-align: right;">min</th>
<th style="text-align: right;">max</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Actinomyces</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.5713867</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.8727358</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">9.0728975</td>
</tr>
<tr class="even">
<td style="text-align: left;">Actinomyces</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Actinomyces</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.6649058</td>
<td style="text-align: right;">0.0527025</td>
<td style="text-align: right;">1.3924048</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">6.0791468</td>
</tr>
<tr class="even">
<td style="text-align: left;">Alloscardovia</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">3.9311642</td>
<td style="text-align: right;">0.2332090</td>
<td style="text-align: right;">14.0753880</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">70.4386523</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Alloscardovia</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0087547</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0392597</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1959104</td>
</tr>
<tr class="even">
<td style="text-align: left;">Alloscardovia</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">1.4267694</td>
<td style="text-align: right;">0.0322511</td>
<td style="text-align: right;">5.0906189</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">25.5642757</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Anaerococcus</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">3.0717887</td>
<td style="text-align: right;">0.5136749</td>
<td style="text-align: right;">5.5732548</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">19.8298621</td>
</tr>
<tr class="even">
<td style="text-align: left;">Anaerococcus</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0932654</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.2912320</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.1904090</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Anaerococcus</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">4.0605333</td>
<td style="text-align: right;">2.9932469</td>
<td style="text-align: right;">3.3448785</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">10.9066824</td>
</tr>
<tr class="even">
<td style="text-align: left;">Atopobium</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">4.5164924</td>
<td style="text-align: right;">0.2277015</td>
<td style="text-align: right;">10.4160035</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">46.9083969</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Atopobium</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">2.3114743</td>
<td style="text-align: right;">0.0596659</td>
<td style="text-align: right;">6.5656355</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">28.4496827</td>
</tr>
<tr class="even">
<td style="text-align: left;">Atopobium</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.9158061</td>
<td style="text-align: right;">0.2469136</td>
<td style="text-align: right;">1.1828644</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">4.3279139</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bifidobacterium</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">4.0593030</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">14.9218728</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">73.1856134</td>
</tr>
<tr class="even">
<td style="text-align: left;">Bifidobacterium</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.4404757</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">2.1949415</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">10.9761388</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bifidobacterium</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.0227102</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1029739</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.5143203</td>
</tr>
<tr class="even">
<td style="text-align: left;">Campylobacter</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.9349138</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.8448353</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">6.7958783</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Campylobacter</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0562029</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1934662</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.9248210</td>
</tr>
<tr class="even">
<td style="text-align: left;">Campylobacter</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">1.4788043</td>
<td style="text-align: right;">0.4923414</td>
<td style="text-align: right;">2.8937352</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">14.3656167</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Corynebacterium</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.8283154</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">2.8155250</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">13.6609746</td>
</tr>
<tr class="even">
<td style="text-align: left;">Corynebacterium</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0192636</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0873295</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.4361759</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Corynebacterium</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.0141055</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0598928</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.2965044</td>
</tr>
<tr class="even">
<td style="text-align: left;">DNF00809</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DNF00809</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0720117</td>
<td style="text-align: right;">0.0152800</td>
<td style="text-align: right;">0.2094370</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.0476512</td>
</tr>
<tr class="even">
<td style="text-align: left;">DNF00809</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.1515091</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.4240207</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.8319212</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dialister</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">1.1172677</td>
<td style="text-align: right;">0.1631588</td>
<td style="text-align: right;">2.1203827</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">8.7569963</td>
</tr>
<tr class="even">
<td style="text-align: left;">Dialister</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.1467705</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.4491541</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.7908018</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dialister</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">6.0987898</td>
<td style="text-align: right;">3.5013155</td>
<td style="text-align: right;">6.3277216</td>
<td style="text-align: right;">0.0408521</td>
<td style="text-align: right;">25.6857604</td>
</tr>
<tr class="even">
<td style="text-align: left;">Escherichia-Shigella</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">4.4422546</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">14.4460327</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">71.3925830</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Escherichia-Shigella</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.5409810</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">2.0390260</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">9.9262483</td>
</tr>
<tr class="even">
<td style="text-align: left;">Escherichia-Shigella</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.0640695</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1839214</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.8926876</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ezakiella</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.6769803</td>
<td style="text-align: right;">0.1195764</td>
<td style="text-align: right;">1.4031535</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">6.3237888</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ezakiella</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0060341</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0174443</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0610501</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ezakiella</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.8736686</td>
<td style="text-align: right;">0.2902602</td>
<td style="text-align: right;">1.2425205</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">5.0328228</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fenollaria</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.5539208</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.7774051</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">8.6307504</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fenollaria</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0022957</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0114784</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0573921</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fenollaria</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">2.2501475</td>
<td style="text-align: right;">1.3760482</td>
<td style="text-align: right;">2.5185955</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">7.2748092</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Finegoldia</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">3.9604420</td>
<td style="text-align: right;">0.8349390</td>
<td style="text-align: right;">6.1259312</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">19.7480620</td>
</tr>
<tr class="even">
<td style="text-align: left;">Finegoldia</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.3455590</td>
<td style="text-align: right;">0.0619963</td>
<td style="text-align: right;">0.7418217</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">3.0926810</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Finegoldia</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">2.2912198</td>
<td style="text-align: right;">0.9965338</td>
<td style="text-align: right;">4.7796209</td>
<td style="text-align: right;">0.0296670</td>
<td style="text-align: right;">23.8936372</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fusobacterium</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.4976509</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.7826115</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">8.6828982</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fusobacterium</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0015992</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0079960</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0399800</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fusobacterium</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.9045715</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">2.4641849</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">11.9206638</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gardnerella</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.3933718</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.3913979</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">6.9161111</td>
</tr>
<tr class="even">
<td style="text-align: left;">Gardnerella</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">3.1539334</td>
<td style="text-align: right;">0.2571324</td>
<td style="text-align: right;">7.7565603</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">29.5627645</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gardnerella</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">2.3591859</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">5.5157218</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">19.5534425</td>
</tr>
<tr class="even">
<td style="text-align: left;">HT002</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.0753210</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1993038</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.6797486</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HT002</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.3367638</td>
<td style="text-align: right;">0.0810235</td>
<td style="text-align: right;">0.5810368</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">2.2941247</td>
</tr>
<tr class="even">
<td style="text-align: left;">HT002</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.0082445</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0412223</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.2061117</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Howardella</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">2.6314285</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">10.1000570</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">49.5850622</td>
</tr>
<tr class="even">
<td style="text-align: left;">Howardella</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0089003</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0270160</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1010995</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Howardella</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.4572855</td>
<td style="text-align: right;">0.2224469</td>
<td style="text-align: right;">1.0960540</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">5.6142399</td>
</tr>
<tr class="even">
<td style="text-align: left;">Lactobacillus</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">24.2775658</td>
<td style="text-align: right;">6.4315353</td>
<td style="text-align: right;">34.4814664</td>
<td style="text-align: right;">0.1284522</td>
<td style="text-align: right;">99.8202570</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Lactobacillus</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">71.6566839</td>
<td style="text-align: right;">87.7634476</td>
<td style="text-align: right;">35.7214691</td>
<td style="text-align: right;">0.5596256</td>
<td style="text-align: right;">99.5759004</td>
</tr>
<tr class="even">
<td style="text-align: left;">Lactobacillus</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">5.9994574</td>
<td style="text-align: right;">0.2801284</td>
<td style="text-align: right;">8.9741717</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">25.9354688</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Moryella</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.0215991</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1079956</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.5399779</td>
</tr>
<tr class="even">
<td style="text-align: left;">Moryella</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Moryella</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.4888759</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.0831026</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">3.8850376</td>
</tr>
<tr class="even">
<td style="text-align: left;">Parvimonas</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.5600192</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.8712302</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">7.5682018</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Parvimonas</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000000</td>
</tr>
<tr class="even">
<td style="text-align: left;">Parvimonas</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.7550410</td>
<td style="text-align: right;">0.1611789</td>
<td style="text-align: right;">1.4094233</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">5.5686949</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Peptoniphilus</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">1.7412584</td>
<td style="text-align: right;">0.3496503</td>
<td style="text-align: right;">2.7391621</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">9.2549871</td>
</tr>
<tr class="even">
<td style="text-align: left;">Peptoniphilus</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.1094172</td>
<td style="text-align: right;">0.0337952</td>
<td style="text-align: right;">0.2215872</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.8850438</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Peptoniphilus</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">2.7222443</td>
<td style="text-align: right;">1.8739690</td>
<td style="text-align: right;">2.4326037</td>
<td style="text-align: right;">0.1453488</td>
<td style="text-align: right;">9.3331711</td>
</tr>
<tr class="even">
<td style="text-align: left;">Peptostreptococcus</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.3240192</td>
<td style="text-align: right;">0.0492692</td>
<td style="text-align: right;">0.6665328</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">2.8736820</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Peptostreptococcus</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0224869</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0885532</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.4375497</td>
</tr>
<tr class="even">
<td style="text-align: left;">Peptostreptococcus</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.5223149</td>
<td style="text-align: right;">0.1598837</td>
<td style="text-align: right;">0.8530107</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">3.3850214</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Porphyromonas</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">1.2860890</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">3.3962705</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">14.8429833</td>
</tr>
<tr class="even">
<td style="text-align: left;">Porphyromonas</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0295355</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1029816</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.4985755</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Porphyromonas</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">5.6496421</td>
<td style="text-align: right;">2.8848984</td>
<td style="text-align: right;">10.7930069</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">50.5763397</td>
</tr>
<tr class="even">
<td style="text-align: left;">Prevotella</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">5.7985547</td>
<td style="text-align: right;">1.1067842</td>
<td style="text-align: right;">9.7530935</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">32.6725746</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Prevotella</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">1.1578882</td>
<td style="text-align: right;">0.2157130</td>
<td style="text-align: right;">2.8240835</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">12.5887124</td>
</tr>
<tr class="even">
<td style="text-align: left;">Prevotella</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">24.3020650</td>
<td style="text-align: right;">24.2634315</td>
<td style="text-align: right;">18.5127750</td>
<td style="text-align: right;">0.3851765</td>
<td style="text-align: right;">56.5387016</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Sneathia</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.1409774</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.6993965</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">3.4979846</td>
</tr>
<tr class="even">
<td style="text-align: left;">Sneathia</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.6402053</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">2.4028989</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">11.9972964</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Sneathia</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.0069812</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0349061</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.1745306</td>
</tr>
<tr class="even">
<td style="text-align: left;">Streptococcus</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">9.1904312</td>
<td style="text-align: right;">0.5155545</td>
<td style="text-align: right;">17.8955941</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">60.3249549</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Streptococcus</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">5.9064624</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">19.6198466</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">86.8697860</td>
</tr>
<tr class="even">
<td style="text-align: left;">Streptococcus</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">9.6731738</td>
<td style="text-align: right;">2.0525784</td>
<td style="text-align: right;">18.5258538</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">79.4981033</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Sutterella</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.0050866</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0202921</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0985383</td>
</tr>
<tr class="even">
<td style="text-align: left;">Sutterella</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.0027508</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0083575</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0340599</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Sutterella</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.1940975</td>
<td style="text-align: right;">0.0725191</td>
<td style="text-align: right;">0.3011899</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.2505042</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ureaplasma</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.0654674</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.2859913</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.4312977</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ureaplasma</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.5672609</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.7398576</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">7.2670503</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ureaplasma</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">0.0189886</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0757508</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.3678443</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">01_postmenopausal</td>
<td style="text-align: right;">0.8703149</td>
<td style="text-align: right;">0.2099702</td>
<td style="text-align: right;">1.9306476</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">9.2816275</td>
</tr>
<tr class="even">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">02_premenopausal</td>
<td style="text-align: right;">0.5669992</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1.8913132</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">8.8465777</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">03_trans-men</td>
<td style="text-align: right;">1.5161479</td>
<td style="text-align: right;">0.8493428</td>
<td style="text-align: right;">2.3679024</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">11.9352562</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(summarystat2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 30%">
<col style="width: 26%">
<col style="width: 24%">
<col style="width: 18%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Genus</th>
<th style="text-align: left;">01_postmenopausal</th>
<th style="text-align: left;">02_premenopausal</th>
<th style="text-align: left;">03_trans-men</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Actinomyces</td>
<td style="text-align: left;">0.57 ± 1.87</td>
<td style="text-align: left;">0 ± 0</td>
<td style="text-align: left;">0.66 ± 1.39</td>
</tr>
<tr class="even">
<td style="text-align: left;">Alloscardovia</td>
<td style="text-align: left;">3.93 ± 14.08</td>
<td style="text-align: left;">0.01 ± 0.04</td>
<td style="text-align: left;">1.43 ± 5.09</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Anaerococcus</td>
<td style="text-align: left;">3.07 ± 5.57</td>
<td style="text-align: left;">0.09 ± 0.29</td>
<td style="text-align: left;">4.06 ± 3.34</td>
</tr>
<tr class="even">
<td style="text-align: left;">Atopobium</td>
<td style="text-align: left;">4.52 ± 10.42</td>
<td style="text-align: left;">2.31 ± 6.57</td>
<td style="text-align: left;">0.92 ± 1.18</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bifidobacterium</td>
<td style="text-align: left;">4.06 ± 14.92</td>
<td style="text-align: left;">0.44 ± 2.19</td>
<td style="text-align: left;">0.02 ± 0.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">Campylobacter</td>
<td style="text-align: left;">0.93 ± 1.84</td>
<td style="text-align: left;">0.06 ± 0.19</td>
<td style="text-align: left;">1.48 ± 2.89</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Corynebacterium</td>
<td style="text-align: left;">0.83 ± 2.82</td>
<td style="text-align: left;">0.02 ± 0.09</td>
<td style="text-align: left;">0.01 ± 0.06</td>
</tr>
<tr class="even">
<td style="text-align: left;">DNF00809</td>
<td style="text-align: left;">0 ± 0</td>
<td style="text-align: left;">0.07 ± 0.21</td>
<td style="text-align: left;">0.15 ± 0.42</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dialister</td>
<td style="text-align: left;">1.12 ± 2.12</td>
<td style="text-align: left;">0.15 ± 0.45</td>
<td style="text-align: left;">6.1 ± 6.33</td>
</tr>
<tr class="even">
<td style="text-align: left;">Escherichia-Shigella</td>
<td style="text-align: left;">4.44 ± 14.45</td>
<td style="text-align: left;">0.54 ± 2.04</td>
<td style="text-align: left;">0.06 ± 0.18</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ezakiella</td>
<td style="text-align: left;">0.68 ± 1.4</td>
<td style="text-align: left;">0.01 ± 0.02</td>
<td style="text-align: left;">0.87 ± 1.24</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fenollaria</td>
<td style="text-align: left;">0.55 ± 1.78</td>
<td style="text-align: left;">0 ± 0.01</td>
<td style="text-align: left;">2.25 ± 2.52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Finegoldia</td>
<td style="text-align: left;">3.96 ± 6.13</td>
<td style="text-align: left;">0.35 ± 0.74</td>
<td style="text-align: left;">2.29 ± 4.78</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fusobacterium</td>
<td style="text-align: left;">0.5 ± 1.78</td>
<td style="text-align: left;">0 ± 0.01</td>
<td style="text-align: left;">0.9 ± 2.46</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Gardnerella</td>
<td style="text-align: left;">0.39 ± 1.39</td>
<td style="text-align: left;">3.15 ± 7.76</td>
<td style="text-align: left;">2.36 ± 5.52</td>
</tr>
<tr class="even">
<td style="text-align: left;">HT002</td>
<td style="text-align: left;">0.08 ± 0.2</td>
<td style="text-align: left;">0.34 ± 0.58</td>
<td style="text-align: left;">0.01 ± 0.04</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Howardella</td>
<td style="text-align: left;">2.63 ± 10.1</td>
<td style="text-align: left;">0.01 ± 0.03</td>
<td style="text-align: left;">0.46 ± 1.1</td>
</tr>
<tr class="even">
<td style="text-align: left;">Lactobacillus</td>
<td style="text-align: left;">24.28 ± 34.48</td>
<td style="text-align: left;">71.66 ± 35.72</td>
<td style="text-align: left;">6 ± 8.97</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Moryella</td>
<td style="text-align: left;">0.02 ± 0.11</td>
<td style="text-align: left;">0 ± 0</td>
<td style="text-align: left;">0.49 ± 1.08</td>
</tr>
<tr class="even">
<td style="text-align: left;">Parvimonas</td>
<td style="text-align: left;">0.56 ± 1.87</td>
<td style="text-align: left;">0 ± 0</td>
<td style="text-align: left;">0.76 ± 1.41</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Peptoniphilus</td>
<td style="text-align: left;">1.74 ± 2.74</td>
<td style="text-align: left;">0.11 ± 0.22</td>
<td style="text-align: left;">2.72 ± 2.43</td>
</tr>
<tr class="even">
<td style="text-align: left;">Peptostreptococcus</td>
<td style="text-align: left;">0.32 ± 0.67</td>
<td style="text-align: left;">0.02 ± 0.09</td>
<td style="text-align: left;">0.52 ± 0.85</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Porphyromonas</td>
<td style="text-align: left;">1.29 ± 3.4</td>
<td style="text-align: left;">0.03 ± 0.1</td>
<td style="text-align: left;">5.65 ± 10.79</td>
</tr>
<tr class="even">
<td style="text-align: left;">Prevotella</td>
<td style="text-align: left;">5.8 ± 9.75</td>
<td style="text-align: left;">1.16 ± 2.82</td>
<td style="text-align: left;">24.3 ± 18.51</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Sneathia</td>
<td style="text-align: left;">0.14 ± 0.7</td>
<td style="text-align: left;">0.64 ± 2.4</td>
<td style="text-align: left;">0.01 ± 0.03</td>
</tr>
<tr class="even">
<td style="text-align: left;">Streptococcus</td>
<td style="text-align: left;">9.19 ± 17.9</td>
<td style="text-align: left;">5.91 ± 19.62</td>
<td style="text-align: left;">9.67 ± 18.53</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Sutterella</td>
<td style="text-align: left;">0.01 ± 0.02</td>
<td style="text-align: left;">0 ± 0.01</td>
<td style="text-align: left;">0.19 ± 0.3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ureaplasma</td>
<td style="text-align: left;">0.07 ± 0.29</td>
<td style="text-align: left;">0.57 ± 1.74</td>
<td style="text-align: left;">0.02 ± 0.08</td>
</tr>
<tr class="odd">
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.87 ± 1.93</td>
<td style="text-align: left;">0.57 ± 1.89</td>
<td style="text-align: left;">1.52 ± 2.37</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This analysis reveals, that the <em>Trans-men</em> group is characterized by a loss of <em>Lactobacillus</em> and an increase in bacteria more commonly associated with the intestinal flora (e.g.&nbsp;<em>Campylobacter, Anaerococcus, Dialister, Prevotella)</em> compared to the premenopausal group.</p>
<p>Comparing the <em>Trans-men</em> to the <em>postmenopausal</em> group only revealed an even larger reduction in <em>Lactobacillus</em>.</p>
<p>Thus, overall the <em>trans-men</em> group is strinkingly similar in its taxonomic composition compared to postmenopausal women.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>deseq_res_fig</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/plot%20deseq2-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(deseq_res_fig, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">10</span>, <span class="at">filename =</span> <span class="st">"results/microbiome/deseq_asv_figure.pdf"</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(deseq_res_fig, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">filename =</span> <span class="st">"results/microbiome/deseq_asv_figure.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="taxonomic-composition" class="level1">
<h1>Taxonomic composition</h1>
<p>The following plots illustrate taxonomic composition at all levels of the taxonomy, The taxon table is filtered to only differentiate the most abundant taxa at each level (depending on the level).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>ps.ns <span class="ot">=</span> <span class="fu">prune_taxa</span>(<span class="fu">taxa_sums</span>(ps_samples) <span class="sc">&gt;</span><span class="dv">1</span>, ps_samples) <span class="co"># remove singleton ASVs</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>ps_rel <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(ps.ns, <span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x) <span class="sc">*</span><span class="dv">100</span>) <span class="co"># transform to rel counts</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate agglomerated phyloseq objects</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>genus_rel <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_rel, <span class="at">taxrank=</span><span class="st">"Genus"</span>, <span class="at">NArm =</span> F)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>genus_rel <span class="co"># 383 genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 398 taxa and 75 samples ]
sample_data() Sample Data:       [ 75 samples by 15 sample variables ]
tax_table()   Taxonomy Table:    [ 398 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 398 tips and 397 internal nodes ]
refseq()      DNAStringSet:      [ 398 reference sequences ]</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>genus <span class="ot">=</span> <span class="fu">psmelt</span>(genus_rel) <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxonomy =</span> <span class="fu">paste0</span>(<span class="st">"p_"</span>, Phylum, <span class="st">";c_"</span>, Class, <span class="st">";o_"</span>, Order, <span class="st">";f_"</span>, Family, <span class="st">";g_"</span>, Genus)) <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Sample, Taxonomy, grp, Age<span class="sc">:</span>Estradiole, Abundance)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>genus<span class="sc">$</span>Taxonomy[genus<span class="sc">$</span>Abundance <span class="sc">&lt;</span> <span class="dv">15</span>] <span class="ot">&lt;-</span> <span class="st">"&lt; 15 % abund."</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>genusplot <span class="ot">=</span> <span class="fu">ggplot</span>(genus, <span class="fu">aes</span>(<span class="at">x =</span> grp, <span class="at">y =</span> Abundance<span class="sc">/</span><span class="dv">25</span>, <span class="at">fill =</span> Taxonomy)) <span class="sc">+</span> </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Abundance (%)"</span>, <span class="at">x=</span> <span class="st">""</span>, <span class="at">caption =</span> <span class="st">"taxonomic level: genus"</span>) <span class="sc">+</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>() </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Family Level</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>family_rel <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_rel, <span class="at">taxrank=</span><span class="st">"Family"</span>, <span class="at">NArm =</span> F)</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>family_rel <span class="co"># 383 genera</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 152 taxa and 75 samples ]
sample_data() Sample Data:       [ 75 samples by 15 sample variables ]
tax_table()   Taxonomy Table:    [ 152 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 152 tips and 151 internal nodes ]
refseq()      DNAStringSet:      [ 152 reference sequences ]</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>family <span class="ot">=</span> <span class="fu">psmelt</span>(family_rel) <span class="sc">%&gt;%</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxonomy =</span> <span class="fu">paste0</span>(<span class="st">"p_"</span>, Phylum, <span class="st">";c_"</span>, Class, <span class="st">";o_"</span>, Order, <span class="st">";f_"</span>, Family)) <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Sample, Taxonomy, grp, Age<span class="sc">:</span>Estradiole, Abundance)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>family<span class="sc">$</span>Taxonomy[family<span class="sc">$</span>Abundance <span class="sc">&lt;</span> <span class="dv">15</span>] <span class="ot">&lt;-</span> <span class="st">"&lt; 15 % abund."</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>familyplot <span class="ot">=</span> <span class="fu">ggplot</span>(family, <span class="fu">aes</span>(<span class="at">x =</span> grp, <span class="at">y =</span> Abundance<span class="sc">/</span><span class="dv">25</span>, <span class="at">fill =</span> Taxonomy)) <span class="sc">+</span> </span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Abundance (%)"</span>, <span class="at">x=</span> <span class="st">""</span>, <span class="at">caption =</span> <span class="st">"taxonomic level: family"</span>) <span class="sc">+</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>()</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Order Level</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>order_rel <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_rel, <span class="at">taxrank=</span><span class="st">"Order"</span>, <span class="at">NArm =</span> F)</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>order_rel <span class="co"># 75 taxa</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 78 taxa and 75 samples ]
sample_data() Sample Data:       [ 75 samples by 15 sample variables ]
tax_table()   Taxonomy Table:    [ 78 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 78 tips and 77 internal nodes ]
refseq()      DNAStringSet:      [ 78 reference sequences ]</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>order <span class="ot">=</span> <span class="fu">psmelt</span>(order_rel) <span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxonomy =</span> <span class="fu">paste0</span>(<span class="st">"p_"</span>, Phylum, <span class="st">";c_"</span>, Class, <span class="st">";o_"</span>, Order)) <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Sample, Taxonomy, grp, Age<span class="sc">:</span>Estradiole, Abundance)</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>order<span class="sc">$</span>Taxonomy[order<span class="sc">$</span>Abundance <span class="sc">&lt;</span> <span class="dv">20</span>] <span class="ot">&lt;-</span> <span class="st">"&lt; 20 % abund."</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>orderplot <span class="ot">=</span> <span class="fu">ggplot</span>(order, <span class="fu">aes</span>(<span class="at">x =</span> grp, <span class="at">y =</span> Abundance<span class="sc">/</span><span class="dv">25</span>, <span class="at">fill =</span> Taxonomy)) <span class="sc">+</span> </span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Abundance (%)"</span>, <span class="at">x=</span> <span class="st">""</span>, <span class="at">caption =</span> <span class="st">"taxonomic level: order"</span>) <span class="sc">+</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>()</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Class Level</span></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>class_rel <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_rel, <span class="at">taxrank=</span><span class="st">"Class"</span>, <span class="at">NArm =</span> F)</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>class_rel <span class="co"># 30 taxa</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 32 taxa and 75 samples ]
sample_data() Sample Data:       [ 75 samples by 15 sample variables ]
tax_table()   Taxonomy Table:    [ 32 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 32 tips and 31 internal nodes ]
refseq()      DNAStringSet:      [ 32 reference sequences ]</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>class <span class="ot">=</span> <span class="fu">psmelt</span>(class_rel) <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxonomy =</span> <span class="fu">paste0</span>(<span class="st">"p_"</span>, Phylum, <span class="st">";c_"</span>, Class)) <span class="sc">%&gt;%</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Sample, Taxonomy, grp, Age<span class="sc">:</span>Estradiole, Abundance)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>class<span class="sc">$</span>Taxonomy[class<span class="sc">$</span>Abundance <span class="sc">&lt;</span> <span class="dv">25</span>] <span class="ot">&lt;-</span> <span class="st">"&lt; 25 % abund."</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>classplot <span class="ot">=</span> <span class="fu">ggplot</span>(class, <span class="fu">aes</span>(<span class="at">x =</span> grp, <span class="at">y =</span> Abundance<span class="sc">/</span><span class="dv">25</span>, <span class="at">fill =</span> Taxonomy)) <span class="sc">+</span> </span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Abundance (%)"</span>, <span class="at">x=</span> <span class="st">""</span>, <span class="at">caption =</span> <span class="st">"taxonomic level: class"</span>) <span class="sc">+</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>()</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Phylum Level</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>phylum_rel <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_rel, <span class="at">taxrank=</span><span class="st">"Phylum"</span>, <span class="at">NArm =</span> F)</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>phylum_rel <span class="co"># 18 taxa</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 19 taxa and 75 samples ]
sample_data() Sample Data:       [ 75 samples by 15 sample variables ]
tax_table()   Taxonomy Table:    [ 19 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 19 tips and 18 internal nodes ]
refseq()      DNAStringSet:      [ 19 reference sequences ]</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>phylum <span class="ot">=</span> <span class="fu">psmelt</span>(phylum_rel) <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Taxonomy =</span> <span class="fu">paste0</span>(<span class="st">"p_"</span>, Phylum)) <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Sample, Taxonomy, grp, Age<span class="sc">:</span>Estradiole, Abundance)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>phylum<span class="sc">$</span>Taxonomy[phylum<span class="sc">$</span>Abundance <span class="sc">&lt;</span> <span class="dv">25</span>] <span class="ot">&lt;-</span> <span class="st">"&lt; 25 % abund."</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>phylumplot <span class="ot">=</span> <span class="fu">ggplot</span>(phylum, <span class="fu">aes</span>(<span class="at">x =</span> grp, <span class="at">y =</span> Abundance<span class="sc">/</span><span class="dv">25</span>, <span class="at">fill =</span> Taxonomy)) <span class="sc">+</span> </span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">"stack"</span>) <span class="sc">+</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Abundance (%)"</span>, <span class="at">x=</span> <span class="st">""</span>, <span class="at">caption =</span> <span class="st">"taxonomic level: phylum"</span>) <span class="sc">+</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="phylum-level" class="level2">
<h2 class="anchored" data-anchor-id="phylum-level">Phylum level</h2>
<p>On the phylum level, the taxonomic distribution is dominated by <em>Firmicutes</em> (these correspond to <em>Lactobacilli</em>, see below). The <em>Trans-men</em> group also shows an increase in <em>Bacteroidota</em> compared to the other 2 groups.</p>
<div class="cell" height="5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>phylumplot <span class="ot">=</span> phylumplot <span class="sc">+</span> <span class="fu">scale_x_discrete</span>(<span class="at">limits =</span> <span class="fu">rev</span>(<span class="fu">levels</span>(phylum<span class="sc">$</span>grp))) <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>phylumplot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/phylum%20plot-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> phylumplot, <span class="at">filename =</span> <span class="st">"results/microbiome/taxplots_phylum.pdf"</span>, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="class-level" class="level2">
<h2 class="anchored" data-anchor-id="class-level">Class level</h2>
<div class="cell" height="5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>classplot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/class%20plot-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> classplot, <span class="at">filename =</span> <span class="st">"results/microbiome/taxplots_class.pdf"</span>, <span class="at">width =</span> <span class="dv">6</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> classplot, <span class="at">filename =</span> <span class="st">"results/microbiome/taxplots_class.png"</span>, <span class="at">width =</span> <span class="dv">6</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="order-level" class="level2">
<h2 class="anchored" data-anchor-id="order-level">Order level</h2>
<div class="cell" height="5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>orderplot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/order%20plot-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> orderplot, <span class="at">filename =</span> <span class="st">"results/microbiome/taxplots_order.pdf"</span>, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> orderplot, <span class="at">filename =</span> <span class="st">"results/microbiome/taxplots_order.png"</span>, <span class="at">width =</span> <span class="dv">7</span>, <span class="at">height =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="family-level" class="level2">
<h2 class="anchored" data-anchor-id="family-level">Family level</h2>
<div class="cell" height="6">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>familyplot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/family%20plot-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> familyplot, <span class="at">filename =</span> <span class="st">"results/microbiome/taxplots_family.pdf"</span>, <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> familyplot, <span class="at">filename =</span> <span class="st">"results/microbiome/taxplots_family.png"</span>, <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="genus-level" class="level2">
<h2 class="anchored" data-anchor-id="genus-level">Genus level</h2>
<div class="cell" height="6">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>genusplot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/genus%20plot-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> genusplot, <span class="at">filename =</span> <span class="st">"results/microbiome/taxplots_genus.pdf"</span>, <span class="at">width =</span> <span class="dv">15</span>, <span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> genusplot, <span class="at">filename =</span> <span class="st">"results/microbiome/taxplots_family.png"</span>, <span class="at">width =</span> <span class="dv">15</span>, <span class="at">height =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="secondary-analyses" class="level1">
<h1>Secondary analyses</h1>
<section id="duration-of-menopause" class="level3">
<h3 class="anchored" data-anchor-id="duration-of-menopause">Duration of menopause</h3>
<p>This analysis is only performed in the postmenopausal group</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">sample_data</span>(ps_samples)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>           CID               grp Age GHAT_Length GNRHA Virgo SexuallyActive
22010-0001 101 01_postmenopausal  64           0  &lt;NA&gt;    no            yes
22010-0002 102 01_postmenopausal  49           0  &lt;NA&gt;    no            yes
22010-0003 103 01_postmenopausal  61           0  &lt;NA&gt;    no            yes
22010-0004 104 01_postmenopausal  70           0  &lt;NA&gt;    no            yes
22010-0005 105 01_postmenopausal  66           0  &lt;NA&gt;    no            yes
22010-0006 106 01_postmenopausal  71           0  &lt;NA&gt;    no             no
           SexOfPartner NugentScore Testosterone Estradiole DurationMenopause
22010-0001            M        7-10           NA         NA                11
22010-0002            M         0-3           NA         NA                 2
22010-0003            M        7-10           NA         NA                10
22010-0004            M         4-6           NA         NA                22
22010-0005            M no bacteria           NA         NA                12
22010-0006            M         0-3           NA         NA                22
           DurationAmenorrhea CycleDaySampling Duration_GNRH
22010-0001                 NA               NA            NA
22010-0002                 NA               NA            NA
22010-0003                 NA               NA            NA
22010-0004                 NA               NA            NA
22010-0005                 NA               NA            NA
22010-0006                 NA               NA            NA</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># for menopause duration, use maaslin2</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Maaslin2)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="co"># only 01_postmenopausal samples</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>ps_menopause <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_samples, grp <span class="sc">==</span><span class="st">"01_postmenopausal"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x) <span class="sc">*</span> <span class="dv">100</span>) <span class="co"># relative abundances are necessary</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>asv_durationMenopause <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_menopause),</span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_menopause), <span class="st">"data.frame"</span>),</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/asv_menopause-duration"</span>,</span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="fl">2.5</span>,</span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.1</span>,</span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"DurationMenopause"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:58:39.004283 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/asv_menopause-duration/maaslin2.log"
2024-03-06 21:58:40.821048 INFO::Writing function arguments to log file
2024-03-06 21:58:40.82766 INFO::Verifying options selected are valid
2024-03-06 21:58:40.851213 INFO::Determining format of input files
2024-03-06 21:58:40.851688 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:58:40.862723 INFO::Formula for fixed effects: expr ~  DurationMenopause
2024-03-06 21:58:40.863224 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:58:40.863474 INFO::Total samples in data: 25
2024-03-06 21:58:40.863699 INFO::Min samples required with min abundance for a feature not to be filtered: 2.500000
2024-03-06 21:58:40.874914 INFO::Total filtered features: 2545
2024-03-06 21:58:40.876026 INFO::Filtered feature names from abundance and prevalence filtering: ASV3, ASV5, ASV6, ASV14, ASV15, ASV16, ASV17, ASV18, ASV19, ASV20, ASV22, ASV23, ASV25, ASV26, ASV27, ASV28, ASV29, ASV30, ASV31, ASV32, ASV33, ASV34, ASV35, ASV36, ASV37, ASV39, ASV40, ASV41, ASV42, ASV43, ASV44, ASV45, ASV46, ASV47, ASV48, ASV49, ASV50, ASV51, ASV52, ASV53, ASV54, ASV55, ASV56, ASV57, ASV58, ASV59, ASV61, ASV62, ASV63, ASV64, ASV65, ASV66, ASV67, ASV68, ASV69, ASV70, ASV71, ASV72, ASV73, ASV74, ASV75, ASV76, ASV77, ASV78, ASV79, ASV80, ASV81, ASV82, ASV83, ASV84, ASV85, ASV86, ASV87, ASV88, ASV89, ASV90, ASV91, ASV92, ASV93, ASV95, ASV96, ASV97, ASV98, ASV99, ASV100, ASV101, ASV102, ASV103, ASV104, ASV105, ASV106, ASV107, ASV108, ASV109, ASV110, ASV111, ASV112, ASV113, ASV114, ASV115, ASV116, ASV117, ASV118, ASV119, ASV120, ASV121, ASV122, ASV123, ASV124, ASV125, ASV126, ASV127, ASV128, ASV129, ASV130, ASV131, ASV132, ASV133, ASV134, ASV135, ASV136, ASV137, ASV138, ASV139, ASV140, ASV141, ASV142, ASV143, ASV144, ASV145, ASV146, ASV147, ASV148, ASV149, ASV150, ASV151, ASV152, ASV153, ASV154, ASV155, ASV156, ASV157, ASV158, ASV159, ASV160, ASV161, ASV162, ASV163, ASV164, ASV165, ASV166, ASV167, ASV168, ASV169, ASV170, ASV171, ASV172, ASV173, ASV174, ASV175, ASV176, ASV177, ASV178, ASV179, ASV180, ASV181, ASV182, ASV183, ASV184, ASV185, ASV186, ASV187, ASV188, ASV189, ASV190, ASV191, ASV192, ASV193, ASV194, ASV195, ASV196, ASV197, ASV198, ASV199, ASV200, ASV201, ASV202, ASV203, ASV204, ASV205, ASV206, ASV207, ASV208, ASV209, ASV210, ASV211, ASV212, ASV213, ASV214, ASV215, ASV216, ASV217, ASV218, ASV219, ASV220, ASV221, ASV222, ASV223, ASV224, ASV225, ASV226, ASV227, ASV228, ASV229, ASV230, ASV231, ASV232, ASV233, ASV234, ASV235, ASV236, ASV237, ASV238, ASV239, ASV240, ASV241, ASV242, ASV243, ASV244, ASV245, ASV246, ASV247, ASV248, ASV249, ASV250, ASV251, ASV252, ASV253, ASV254, ASV255, ASV256, ASV257, ASV258, ASV259, ASV260, ASV261, ASV262, ASV263, ASV264, ASV265, ASV266, ASV267, ASV268, ASV269, ASV270, ASV271, ASV272, ASV273, ASV274, ASV275, ASV276, ASV277, ASV278, ASV279, ASV280, ASV281, ASV282, ASV283, ASV284, ASV285, ASV286, ASV287, ASV288, ASV289, ASV290, ASV291, ASV292, ASV293, ASV294, ASV295, ASV296, ASV297, ASV298, ASV299, ASV300, ASV301, ASV302, ASV303, ASV304, ASV305, ASV306, ASV307, ASV308, ASV309, ASV310, ASV311, ASV312, ASV313, ASV314, ASV315, ASV316, ASV317, ASV318, ASV319, ASV320, ASV321, ASV322, ASV323, ASV324, ASV325, ASV326, ASV327, ASV328, ASV329, ASV330, ASV331, ASV332, ASV333, ASV334, ASV335, ASV336, ASV337, ASV338, ASV339, ASV340, ASV341, ASV342, ASV343, ASV344, ASV345, ASV346, ASV347, ASV348, ASV349, ASV350, ASV351, ASV352, ASV353, ASV354, ASV355, ASV356, ASV357, ASV358, ASV359, ASV360, ASV361, ASV362, ASV363, ASV364, ASV365, ASV366, ASV367, ASV368, ASV369, ASV370, ASV371, ASV372, ASV373, ASV374, ASV375, ASV376, ASV377, ASV378, ASV379, ASV380, ASV381, ASV382, ASV383, ASV384, ASV385, ASV386, ASV387, ASV388, ASV389, ASV390, ASV391, ASV392, ASV393, ASV394, ASV395, ASV396, ASV397, ASV398, ASV399, ASV400, ASV401, ASV402, ASV403, ASV404, ASV405, ASV406, ASV407, ASV408, ASV409, ASV410, ASV411, ASV412, ASV413, ASV414, ASV415, ASV416, ASV417, ASV418, ASV419, ASV420, ASV421, ASV422, ASV423, ASV424, ASV425, ASV426, ASV427, ASV428, ASV429, ASV430, ASV431, ASV432, ASV433, ASV434, ASV435, ASV436, ASV437, ASV438, ASV439, ASV440, ASV441, ASV442, ASV443, ASV444, ASV445, ASV446, ASV447, ASV448, ASV449, ASV450, ASV451, ASV452, ASV453, ASV454, ASV455, ASV456, ASV457, ASV458, ASV459, ASV460, ASV461, ASV462, ASV463, ASV464, ASV465, ASV466, ASV467, ASV468, ASV469, ASV470, ASV471, ASV472, ASV473, ASV474, ASV475, ASV476, ASV477, ASV478, ASV479, ASV480, ASV481, ASV482, ASV483, ASV484, ASV485, ASV486, ASV487, ASV488, ASV489, ASV490, ASV491, ASV492, ASV493, ASV494, ASV495, ASV496, ASV497, ASV498, ASV499, ASV500, ASV501, ASV502, ASV503, ASV504, ASV505, ASV506, ASV507, ASV508, ASV509, ASV510, ASV511, ASV512, ASV513, ASV514, ASV515, ASV516, ASV517, ASV518, ASV519, ASV520, ASV521, ASV522, ASV523, ASV524, ASV525, ASV526, ASV527, ASV528, ASV529, ASV530, ASV531, ASV532, ASV533, ASV534, ASV535, ASV536, ASV537, ASV538, ASV539, ASV540, ASV541, ASV542, ASV543, ASV544, ASV545, ASV546, ASV547, ASV548, ASV549, ASV550, ASV551, ASV552, ASV553, ASV554, ASV555, ASV556, ASV557, ASV558, ASV559, ASV560, ASV561, ASV562, ASV563, ASV564, ASV565, ASV566, ASV567, ASV568, ASV569, ASV570, ASV571, ASV572, ASV573, ASV574, ASV575, ASV576, ASV577, ASV578, ASV579, ASV580, ASV581, ASV582, ASV583, ASV584, ASV585, ASV586, ASV587, ASV588, ASV589, ASV590, ASV591, ASV592, ASV593, ASV594, ASV595, ASV596, ASV597, ASV598, ASV599, ASV600, ASV601, ASV602, ASV603, ASV604, ASV605, ASV606, ASV607, ASV608, ASV609, ASV610, ASV611, ASV612, ASV613, ASV614, ASV615, ASV616, ASV617, ASV618, ASV619, ASV620, ASV621, ASV622, ASV623, ASV624, ASV625, ASV626, ASV627, ASV628, ASV629, ASV630, ASV631, ASV632, ASV633, ASV634, ASV635, ASV636, ASV637, ASV638, ASV639, ASV640, ASV641, ASV642, ASV643, ASV644, ASV645, ASV646, ASV647, ASV648, ASV649, ASV650, ASV651, ASV652, ASV653, ASV654, ASV655, ASV656, ASV657, ASV658, ASV659, ASV660, ASV661, ASV662, ASV663, ASV664, ASV665, ASV666, ASV667, ASV668, ASV669, ASV670, ASV671, ASV672, ASV673, ASV674, ASV675, ASV676, ASV677, ASV678, ASV679, ASV680, ASV681, ASV682, ASV683, ASV684, ASV685, ASV686, ASV687, ASV688, ASV689, ASV690, ASV691, ASV692, ASV693, ASV694, ASV695, ASV696, ASV697, ASV698, ASV699, ASV700, ASV701, ASV702, ASV703, ASV704, ASV705, ASV706, ASV707, ASV708, ASV709, ASV710, ASV711, ASV712, ASV713, ASV714, ASV715, ASV716, ASV717, ASV718, ASV719, ASV720, ASV721, ASV722, ASV723, ASV724, ASV725, ASV726, ASV727, ASV728, ASV729, ASV730, ASV731, ASV732, ASV733, ASV734, ASV735, ASV736, ASV737, ASV738, ASV739, ASV740, ASV741, ASV742, ASV743, ASV744, ASV745, ASV746, ASV747, ASV748, ASV749, ASV750, ASV751, ASV752, ASV753, ASV754, ASV755, ASV756, ASV757, ASV758, ASV759, ASV760, ASV761, ASV762, ASV763, ASV764, ASV765, ASV766, ASV767, ASV768, ASV769, ASV770, ASV771, ASV772, ASV773, ASV774, ASV775, ASV776, ASV777, ASV778, ASV779, ASV780, ASV781, ASV782, ASV783, ASV784, ASV785, ASV786, ASV787, ASV788, ASV789, ASV790, ASV791, ASV792, ASV793, ASV794, ASV795, ASV796, ASV797, ASV798, ASV799, ASV800, ASV801, ASV802, ASV803, ASV804, ASV805, ASV806, ASV807, ASV808, ASV809, ASV810, ASV811, ASV812, ASV813, ASV814, ASV815, ASV816, ASV817, ASV818, ASV819, ASV820, ASV821, ASV822, ASV823, ASV824, ASV825, ASV826, ASV827, ASV828, ASV829, ASV830, ASV831, ASV832, ASV833, ASV834, ASV835, ASV836, ASV837, ASV838, ASV839, ASV840, ASV841, ASV842, ASV843, ASV844, ASV845, ASV846, ASV847, ASV848, ASV849, ASV850, ASV851, ASV852, ASV853, ASV854, ASV855, ASV856, ASV857, ASV858, ASV859, ASV860, ASV861, ASV862, ASV863, ASV864, ASV865, ASV866, ASV867, ASV868, ASV869, ASV870, ASV871, ASV872, ASV873, ASV874, ASV875, ASV876, ASV877, ASV878, ASV879, ASV880, ASV881, ASV882, ASV883, ASV884, ASV885, ASV886, ASV887, ASV888, ASV889, ASV890, ASV891, ASV892, ASV893, ASV894, ASV895, ASV896, ASV897, ASV898, ASV899, ASV900, ASV901, ASV902, ASV903, ASV904, ASV905, ASV906, ASV907, ASV908, ASV909, ASV910, ASV911, ASV912, ASV913, ASV914, ASV915, ASV916, ASV917, ASV918, ASV919, ASV920, ASV921, ASV922, ASV923, ASV924, ASV925, ASV926, ASV927, ASV928, ASV929, ASV930, ASV931, ASV932, ASV933, ASV934, ASV935, ASV936, ASV937, ASV938, ASV939, ASV940, ASV941, ASV942, ASV943, ASV944, ASV945, ASV946, ASV947, ASV948, ASV949, ASV950, ASV951, ASV952, ASV953, ASV954, ASV955, ASV956, ASV957, ASV958, ASV959, ASV960, ASV961, ASV962, ASV963, ASV964, ASV965, ASV966, ASV967, ASV968, ASV969, ASV970, ASV971, ASV972, ASV973, ASV974, ASV975, ASV976, ASV977, ASV978, ASV979, ASV980, ASV981, ASV982, ASV983, ASV984, ASV985, ASV986, ASV987, ASV988, ASV989, ASV990, ASV991, ASV992, ASV993, ASV994, ASV995, ASV996, ASV997, ASV998, ASV999, ASV1000, ASV1001, ASV1002, ASV1003, ASV1004, ASV1005, ASV1006, ASV1007, ASV1008, ASV1009, ASV1010, ASV1011, ASV1012, ASV1013, ASV1014, ASV1015, ASV1016, ASV1017, ASV1018, ASV1019, ASV1020, ASV1021, ASV1022, ASV1023, ASV1024, ASV1025, ASV1026, ASV1027, ASV1028, ASV1029, ASV1030, ASV1031, ASV1032, ASV1033, ASV1034, ASV1035, ASV1036, ASV1037, ASV1038, ASV1039, ASV1040, ASV1041, ASV1042, ASV1043, ASV1044, ASV1045, ASV1046, ASV1047, ASV1048, ASV1049, ASV1050, ASV1051, ASV1052, ASV1053, ASV1054, ASV1055, ASV1056, ASV1057, ASV1058, ASV1059, ASV1060, ASV1061, ASV1062, ASV1063, ASV1064, ASV1065, ASV1066, ASV1067, ASV1068, ASV1069, ASV1070, ASV1071, ASV1072, ASV1073, ASV1074, ASV1075, ASV1076, ASV1077, ASV1078, ASV1079, ASV1080, ASV1081, ASV1082, ASV1083, ASV1084, ASV1085, ASV1086, ASV1087, ASV1088, ASV1089, ASV1090, ASV1091, ASV1092, ASV1093, ASV1094, ASV1095, ASV1096, ASV1097, ASV1098, ASV1099, ASV1100, ASV1101, ASV1102, ASV1103, ASV1104, ASV1105, ASV1106, ASV1107, ASV1108, ASV1109, ASV1110, ASV1111, ASV1112, ASV1113, ASV1114, ASV1115, ASV1116, ASV1117, ASV1118, ASV1119, ASV1120, ASV1121, ASV1122, ASV1123, ASV1124, ASV1125, ASV1126, ASV1127, ASV1128, ASV1129, ASV1130, ASV1131, ASV1132, ASV1133, ASV1134, ASV1135, ASV1136, ASV1137, ASV1138, ASV1139, ASV1140, ASV1141, ASV1142, ASV1143, ASV1144, ASV1145, ASV1146, ASV1147, ASV1148, ASV1149, ASV1150, ASV1151, ASV1152, ASV1153, ASV1154, ASV1155, ASV1156, ASV1157, ASV1158, ASV1159, ASV1160, ASV1161, ASV1162, ASV1163, ASV1164, ASV1165, ASV1166, ASV1167, ASV1168, ASV1169, ASV1170, ASV1171, ASV1172, ASV1173, ASV1174, ASV1175, ASV1176, ASV1177, ASV1178, ASV1179, ASV1180, ASV1181, ASV1182, ASV1183, ASV1184, ASV1185, ASV1186, ASV1187, ASV1188, ASV1189, ASV1190, ASV1191, ASV1192, ASV1193, ASV1194, ASV1195, ASV1196, ASV1197, ASV1198, ASV1199, ASV1200, ASV1201, ASV1202, ASV1203, ASV1204, ASV1205, ASV1206, ASV1207, ASV1208, ASV1209, ASV1210, ASV1211, ASV1212, ASV1213, ASV1214, ASV1215, ASV1216, ASV1217, ASV1218, ASV1219, ASV1220, ASV1221, ASV1222, ASV1223, ASV1224, ASV1225, ASV1226, ASV1227, ASV1228, ASV1229, ASV1230, ASV1231, ASV1232, ASV1233, ASV1234, ASV1235, ASV1236, ASV1237, ASV1238, ASV1239, ASV1240, ASV1241, ASV1242, ASV1243, ASV1244, ASV1245, ASV1246, ASV1247, ASV1248, ASV1249, ASV1250, ASV1251, ASV1252, ASV1253, ASV1254, ASV1255, ASV1256, ASV1257, ASV1258, ASV1259, ASV1260, ASV1261, ASV1262, ASV1263, ASV1264, ASV1265, ASV1266, ASV1267, ASV1268, ASV1269, ASV1270, ASV1271, ASV1272, ASV1273, ASV1274, ASV1275, ASV1276, ASV1277, ASV1278, ASV1279, ASV1280, ASV1281, ASV1282, ASV1283, ASV1284, ASV1285, ASV1286, ASV1287, ASV1288, ASV1289, ASV1290, ASV1291, ASV1292, ASV1293, ASV1294, ASV1295, ASV1296, ASV1297, ASV1298, ASV1299, ASV1300, ASV1301, ASV1302, ASV1303, ASV1304, ASV1305, ASV1306, ASV1307, ASV1308, ASV1309, ASV1310, ASV1311, ASV1312, ASV1313, ASV1314, ASV1315, ASV1316, ASV1317, ASV1318, ASV1319, ASV1320, ASV1321, ASV1322, ASV1323, ASV1324, ASV1325, ASV1326, ASV1327, ASV1328, ASV1329, ASV1330, ASV1331, ASV1332, ASV1333, ASV1334, ASV1335, ASV1336, ASV1337, ASV1338, ASV1339, ASV1340, ASV1341, ASV1342, ASV1343, ASV1344, ASV1345, ASV1346, ASV1347, ASV1348, ASV1349, ASV1350, ASV1351, ASV1352, ASV1353, ASV1354, ASV1355, ASV1356, ASV1357, ASV1358, ASV1359, ASV1360, ASV1361, ASV1362, ASV1363, ASV1364, ASV1365, ASV1366, ASV1367, ASV1368, ASV1369, ASV1370, ASV1371, ASV1372, ASV1373, ASV1374, ASV1375, ASV1376, ASV1377, ASV1378, ASV1379, ASV1380, ASV1381, ASV1382, ASV1383, ASV1384, ASV1385, ASV1386, ASV1387, ASV1388, ASV1389, ASV1390, ASV1391, ASV1392, ASV1393, ASV1394, ASV1395, ASV1396, ASV1397, ASV1398, ASV1399, ASV1400, ASV1401, ASV1402, ASV1403, ASV1404, ASV1405, ASV1406, ASV1407, ASV1408, ASV1409, ASV1410, ASV1411, ASV1412, ASV1413, ASV1414, ASV1415, ASV1416, ASV1417, ASV1418, ASV1419, ASV1420, ASV1421, ASV1422, ASV1423, ASV1424, ASV1425, ASV1426, ASV1427, ASV1428, ASV1429, ASV1430, ASV1431, ASV1432, ASV1433, ASV1434, ASV1435, ASV1436, ASV1437, ASV1438, ASV1439, ASV1440, ASV1441, ASV1442, ASV1443, ASV1444, ASV1445, ASV1446, ASV1447, ASV1448, ASV1449, ASV1450, ASV1451, ASV1452, ASV1453, ASV1454, ASV1455, ASV1456, ASV1457, ASV1458, ASV1459, ASV1460, ASV1461, ASV1462, ASV1463, ASV1464, ASV1465, ASV1466, ASV1467, ASV1468, ASV1469, ASV1470, ASV1471, ASV1472, ASV1473, ASV1474, ASV1475, ASV1476, ASV1477, ASV1478, ASV1479, ASV1480, ASV1481, ASV1482, ASV1483, ASV1484, ASV1485, ASV1486, ASV1487, ASV1488, ASV1489, ASV1490, ASV1491, ASV1492, ASV1493, ASV1494, ASV1495, ASV1496, ASV1497, ASV1498, ASV1499, ASV1500, ASV1501, ASV1502, ASV1503, ASV1504, ASV1505, ASV1506, ASV1507, ASV1508, ASV1509, ASV1510, ASV1511, ASV1512, ASV1513, ASV1514, ASV1515, ASV1516, ASV1517, ASV1518, ASV1519, ASV1520, ASV1521, ASV1522, ASV1523, ASV1524, ASV1525, ASV1526, ASV1527, ASV1528, ASV1529, ASV1530, ASV1531, ASV1532, ASV1533, ASV1534, ASV1535, ASV1536, ASV1537, ASV1538, ASV1539, ASV1540, ASV1541, ASV1542, ASV1543, ASV1544, ASV1545, ASV1546, ASV1547, ASV1548, ASV1549, ASV1550, ASV1551, ASV1552, ASV1553, ASV1554, ASV1555, ASV1556, ASV1557, ASV1558, ASV1559, ASV1560, ASV1561, ASV1562, ASV1563, ASV1564, ASV1565, ASV1566, ASV1567, ASV1568, ASV1569, ASV1570, ASV1571, ASV1572, ASV1573, ASV1574, ASV1575, ASV1576, ASV1577, ASV1578, ASV1579, ASV1580, ASV1581, ASV1582, ASV1583, ASV1584, ASV1585, ASV1586, ASV1587, ASV1588, ASV1589, ASV1590, ASV1591, ASV1592, ASV1593, ASV1594, ASV1595, ASV1596, ASV1597, ASV1598, ASV1599, ASV1600, ASV1601, ASV1602, ASV1603, ASV1604, ASV1605, ASV1606, ASV1607, ASV1608, ASV1609, ASV1610, ASV1611, ASV1612, ASV1613, ASV1614, ASV1615, ASV1616, ASV1617, ASV1618, ASV1619, ASV1620, ASV1621, ASV1622, ASV1623, ASV1624, ASV1625, ASV1626, ASV1627, ASV1628, ASV1629, ASV1630, ASV1631, ASV1632, ASV1633, ASV1634, ASV1635, ASV1636, ASV1637, ASV1638, ASV1639, ASV1640, ASV1641, ASV1642, ASV1643, ASV1644, ASV1645, ASV1646, ASV1647, ASV1648, ASV1649, ASV1650, ASV1651, ASV1652, ASV1653, ASV1654, ASV1655, ASV1656, ASV1657, ASV1658, ASV1659, ASV1660, ASV1661, ASV1662, ASV1663, ASV1664, ASV1665, ASV1666, ASV1667, ASV1668, ASV1669, ASV1670, ASV1671, ASV1672, ASV1673, ASV1674, ASV1675, ASV1676, ASV1677, ASV1678, ASV1679, ASV1680, ASV1681, ASV1682, ASV1683, ASV1684, ASV1685, ASV1686, ASV1687, ASV1688, ASV1689, ASV1690, ASV1691, ASV1692, ASV1693, ASV1694, ASV1695, ASV1696, ASV1697, ASV1698, ASV1699, ASV1700, ASV1701, ASV1702, ASV1703, ASV1704, ASV1705, ASV1706, ASV1707, ASV1708, ASV1709, ASV1710, ASV1711, ASV1712, ASV1713, ASV1714, ASV1715, ASV1716, ASV1717, ASV1718, ASV1719, ASV1720, ASV1721, ASV1722, ASV1723, ASV1724, ASV1725, ASV1726, ASV1727, ASV1728, ASV1729, ASV1730, ASV1731, ASV1732, ASV1733, ASV1734, ASV1735, ASV1736, ASV1737, ASV1738, ASV1739, ASV1740, ASV1741, ASV1742, ASV1743, ASV1744, ASV1745, ASV1746, ASV1747, ASV1748, ASV1749, ASV1750, ASV1751, ASV1752, ASV1753, ASV1754, ASV1755, ASV1756, ASV1757, ASV1758, ASV1759, ASV1760, ASV1761, ASV1762, ASV1763, ASV1764, ASV1765, ASV1766, ASV1767, ASV1768, ASV1769, ASV1770, ASV1771, ASV1772, ASV1773, ASV1774, ASV1775, ASV1776, ASV1777, ASV1778, ASV1779, ASV1780, ASV1781, ASV1782, ASV1783, ASV1784, ASV1785, ASV1786, ASV1787, ASV1788, ASV1789, ASV1790, ASV1791, ASV1792, ASV1793, ASV1794, ASV1795, ASV1796, ASV1797, ASV1798, ASV1799, ASV1800, ASV1801, ASV1802, ASV1803, ASV1804, ASV1805, ASV1806, ASV1807, ASV1808, ASV1809, ASV1810, ASV1811, ASV1812, ASV1813, ASV1814, ASV1815, ASV1816, ASV1817, ASV1818, ASV1819, ASV1820, ASV1821, ASV1822, ASV1823, ASV1824, ASV1825, ASV1826, ASV1827, ASV1828, ASV1829, ASV1830, ASV1831, ASV1832, ASV1833, ASV1834, ASV1835, ASV1836, ASV1837, ASV1838, ASV1839, ASV1840, ASV1841, ASV1842, ASV1843, ASV1844, ASV1845, ASV1846, ASV1847, ASV1848, ASV1849, ASV1850, ASV1851, ASV1852, ASV1853, ASV1854, ASV1855, ASV1856, ASV1857, ASV1858, ASV1859, ASV1860, ASV1861, ASV1862, ASV1863, ASV1864, ASV1865, ASV1866, ASV1867, ASV1868, ASV1869, ASV1870, ASV1871, ASV1872, ASV1873, ASV1874, ASV1875, ASV1876, ASV1877, ASV1878, ASV1879, ASV1880, ASV1881, ASV1882, ASV1883, ASV1884, ASV1885, ASV1886, ASV1887, ASV1888, ASV1889, ASV1890, ASV1891, ASV1892, ASV1893, ASV1894, ASV1895, ASV1896, ASV1897, ASV1898, ASV1899, ASV1900, ASV1901, ASV1902, ASV1903, ASV1904, ASV1905, ASV1906, ASV1907, ASV1908, ASV1909, ASV1910, ASV1911, ASV1912, ASV1913, ASV1914, ASV1915, ASV1916, ASV1917, ASV1918, ASV1919, ASV1920, ASV1921, ASV1922, ASV1923, ASV1924, ASV1925, ASV1926, ASV1927, ASV1928, ASV1929, ASV1930, ASV1931, ASV1932, ASV1933, ASV1934, ASV1935, ASV1936, ASV1937, ASV1938, ASV1939, ASV1940, ASV1941, ASV1942, ASV1943, ASV1944, ASV1945, ASV1946, ASV1947, ASV1948, ASV1949, ASV1950, ASV1951, ASV1952, ASV1953, ASV1954, ASV1955, ASV1956, ASV1957, ASV1958, ASV1959, ASV1960, ASV1961, ASV1962, ASV1963, ASV1964, ASV1965, ASV1966, ASV1967, ASV1968, ASV1969, ASV1970, ASV1971, ASV1972, ASV1973, ASV1974, ASV1975, ASV1976, ASV1977, ASV1978, ASV1979, ASV1980, ASV1981, ASV1982, ASV1983, ASV1984, ASV1985, ASV1986, ASV1987, ASV1988, ASV1989, ASV1990, ASV1991, ASV1992, ASV1993, ASV1994, ASV1995, ASV1996, ASV1997, ASV1998, ASV1999, ASV2000, ASV2001, ASV2002, ASV2003, ASV2004, ASV2005, ASV2006, ASV2007, ASV2008, ASV2009, ASV2010, ASV2011, ASV2012, ASV2013, ASV2014, ASV2015, ASV2016, ASV2017, ASV2018, ASV2019, ASV2020, ASV2021, ASV2022, ASV2023, ASV2024, ASV2025, ASV2026, ASV2027, ASV2028, ASV2029, ASV2030, ASV2031, ASV2032, ASV2033, ASV2034, ASV2035, ASV2036, ASV2037, ASV2038, ASV2039, ASV2040, ASV2041, ASV2042, ASV2043, ASV2044, ASV2045, ASV2046, ASV2047, ASV2048, ASV2049, ASV2050, ASV2051, ASV2052, ASV2053, ASV2054, ASV2055, ASV2056, ASV2057, ASV2058, ASV2059, ASV2060, ASV2061, ASV2062, ASV2063, ASV2064, ASV2065, ASV2066, ASV2067, ASV2068, ASV2069, ASV2070, ASV2071, ASV2072, ASV2073, ASV2074, ASV2075, ASV2076, ASV2077, ASV2078, ASV2079, ASV2080, ASV2081, ASV2082, ASV2083, ASV2084, ASV2085, ASV2086, ASV2087, ASV2088, ASV2089, ASV2090, ASV2091, ASV2092, ASV2093, ASV2094, ASV2095, ASV2096, ASV2097, ASV2098, ASV2099, ASV2100, ASV2101, ASV2102, ASV2103, ASV2104, ASV2105, ASV2106, ASV2107, ASV2108, ASV2109, ASV2110, ASV2111, ASV2112, ASV2113, ASV2114, ASV2115, ASV2116, ASV2117, ASV2118, ASV2119, ASV2120, ASV2121, ASV2122, ASV2123, ASV2124, ASV2125, ASV2126, ASV2127, ASV2128, ASV2129, ASV2130, ASV2131, ASV2132, ASV2133, ASV2134, ASV2135, ASV2136, ASV2137, ASV2138, ASV2139, ASV2140, ASV2141, ASV2142, ASV2143, ASV2144, ASV2145, ASV2146, ASV2147, ASV2148, ASV2149, ASV2150, ASV2151, ASV2152, ASV2153, ASV2154, ASV2155, ASV2156, ASV2157, ASV2158, ASV2159, ASV2160, ASV2161, ASV2162, ASV2163, ASV2164, ASV2165, ASV2166, ASV2167, ASV2168, ASV2169, ASV2170, ASV2171, ASV2172, ASV2173, ASV2174, ASV2175, ASV2176, ASV2177, ASV2178, ASV2179, ASV2180, ASV2181, ASV2182, ASV2183, ASV2184, ASV2185, ASV2186, ASV2187, ASV2188, ASV2189, ASV2190, ASV2191, ASV2192, ASV2193, ASV2194, ASV2195, ASV2196, ASV2197, ASV2198, ASV2199, ASV2200, ASV2201, ASV2202, ASV2203, ASV2204, ASV2205, ASV2206, ASV2207, ASV2208, ASV2209, ASV2210, ASV2211, ASV2212, ASV2213, ASV2214, ASV2215, ASV2216, ASV2217, ASV2218, ASV2219, ASV2220, ASV2221, ASV2222, ASV2223, ASV2224, ASV2225, ASV2226, ASV2227, ASV2228, ASV2229, ASV2230, ASV2231, ASV2232, ASV2233, ASV2234, ASV2235, ASV2236, ASV2237, ASV2238, ASV2239, ASV2240, ASV2241, ASV2242, ASV2243, ASV2244, ASV2245, ASV2246, ASV2247, ASV2248, ASV2249, ASV2250, ASV2251, ASV2252, ASV2253, ASV2254, ASV2255, ASV2256, ASV2257, ASV2258, ASV2259, ASV2260, ASV2261, ASV2262, ASV2263, ASV2264, ASV2265, ASV2266, ASV2267, ASV2268, ASV2269, ASV2270, ASV2271, ASV2272, ASV2273, ASV2274, ASV2275, ASV2276, ASV2277, ASV2278, ASV2279, ASV2280, ASV2281, ASV2282, ASV2283, ASV2284, ASV2285, ASV2286, ASV2287, ASV2288, ASV2289, ASV2290, ASV2291, ASV2292, ASV2293, ASV2294, ASV2295, ASV2296, ASV2297, ASV2298, ASV2299, ASV2300, ASV2301, ASV2302, ASV2303, ASV2304, ASV2305, ASV2306, ASV2307, ASV2308, ASV2309, ASV2310, ASV2311, ASV2312, ASV2313, ASV2314, ASV2315, ASV2316, ASV2317, ASV2318, ASV2319, ASV2320, ASV2321, ASV2322, ASV2323, ASV2324, ASV2325, ASV2326, ASV2327, ASV2328, ASV2329, ASV2330, ASV2331, ASV2332, ASV2333, ASV2334, ASV2335, ASV2336, ASV2337, ASV2338, ASV2339, ASV2340, ASV2341, ASV2342, ASV2343, ASV2344, ASV2345, ASV2346, ASV2347, ASV2348, ASV2349, ASV2350, ASV2351, ASV2352, ASV2353, ASV2354, ASV2355, ASV2356, ASV2357, ASV2358, ASV2359, ASV2360, ASV2361, ASV2362, ASV2363, ASV2364, ASV2365, ASV2366, ASV2367, ASV2368, ASV2369, ASV2370, ASV2371, ASV2372, ASV2373, ASV2374, ASV2375, ASV2376, ASV2377, ASV2378, ASV2379, ASV2380, ASV2381, ASV2382, ASV2383, ASV2384, ASV2385, ASV2386, ASV2387, ASV2388, ASV2389, ASV2390, ASV2391, ASV2392, ASV2393, ASV2394, ASV2395, ASV2396, ASV2397, ASV2398, ASV2399, ASV2400, ASV2401, ASV2402, ASV2403, ASV2404, ASV2405, ASV2406, ASV2407, ASV2408, ASV2409, ASV2410, ASV2411, ASV2412, ASV2413, ASV2414, ASV2415, ASV2416, ASV2417, ASV2418, ASV2419, ASV2420, ASV2421, ASV2422, ASV2423, ASV2424, ASV2425, ASV2426, ASV2427, ASV2428, ASV2429, ASV2430, ASV2431, ASV2432, ASV2433, ASV2434, ASV2435, ASV2436, ASV2437, ASV2438, ASV2439, ASV2440, ASV2441, ASV2442, ASV2443, ASV2444, ASV2445, ASV2446, ASV2447, ASV2448, ASV2449, ASV2450, ASV2451, ASV2452, ASV2453, ASV2454, ASV2455, ASV2456, ASV2457, ASV2458, ASV2459, ASV2460, ASV2461, ASV2462, ASV2463, ASV2464, ASV2465, ASV2466, ASV2467, ASV2468, ASV2469, ASV2470, ASV2471, ASV2472, ASV2473, ASV2474, ASV2475, ASV2476, ASV2477, ASV2478, ASV2479, ASV2480, ASV2481, ASV2482, ASV2483, ASV2484, ASV2485, ASV2486, ASV2487, ASV2488, ASV2489, ASV2490, ASV2491, ASV2492, ASV2493, ASV2494, ASV2495, ASV2496, ASV2497, ASV2498, ASV2499, ASV2500, ASV2501, ASV2502, ASV2503, ASV2504, ASV2505, ASV2506, ASV2507, ASV2508, ASV2509, ASV2510, ASV2511, ASV2512, ASV2513, ASV2514, ASV2515, ASV2516, ASV2517, ASV2518, ASV2519, ASV2520, ASV2521, ASV2522, ASV2523, ASV2524, ASV2525, ASV2526, ASV2527, ASV2528, ASV2529, ASV2530, ASV2531, ASV2532, ASV2533, ASV2534, ASV2535, ASV2536, ASV2537, ASV2538, ASV2539, ASV2540, ASV2541, ASV2542, ASV2543, ASV2544, ASV2545, ASV2546, ASV2547, ASV2548, ASV2549, ASV2550, ASV2551, ASV2552, ASV2553, ASV2554, ASV2555, ASV2556, ASV2557, ASV2558, ASV2559, ASV2560
2024-03-06 21:58:40.877044 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:58:40.877317 INFO::Filtered feature names from variance filtering:
2024-03-06 21:58:40.877547 INFO::Running selected normalization method: TSS
2024-03-06 21:58:40.8785 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:58:40.882077 INFO::Running selected transform method: LOG
2024-03-06 21:58:40.882707 INFO::Running selected analysis method: LM
2024-03-06 21:58:40.886751 INFO::Fitting model to feature number 1, ASV1
2024-03-06 21:58:40.888849 INFO::Fitting model to feature number 2, ASV2
2024-03-06 21:58:40.889946 INFO::Fitting model to feature number 3, ASV4
2024-03-06 21:58:40.891012 INFO::Fitting model to feature number 4, ASV7
2024-03-06 21:58:40.892039 INFO::Fitting model to feature number 5, ASV8
2024-03-06 21:58:40.892983 INFO::Fitting model to feature number 6, ASV9
2024-03-06 21:58:40.893888 INFO::Fitting model to feature number 7, ASV10
2024-03-06 21:58:40.894781 INFO::Fitting model to feature number 8, ASV11
2024-03-06 21:58:40.89572 INFO::Fitting model to feature number 9, ASV12
2024-03-06 21:58:40.896661 INFO::Fitting model to feature number 10, ASV13
2024-03-06 21:58:40.89765 INFO::Fitting model to feature number 11, ASV21
2024-03-06 21:58:40.898565 INFO::Fitting model to feature number 12, ASV24
2024-03-06 21:58:40.899419 INFO::Fitting model to feature number 13, ASV38
2024-03-06 21:58:40.900265 INFO::Fitting model to feature number 14, ASV60
2024-03-06 21:58:40.901101 INFO::Fitting model to feature number 15, ASV94
2024-03-06 21:58:40.904224 INFO::Counting total values for each feature
2024-03-06 21:58:40.905088 INFO::Writing filtered data to file results/maaslin2/asv_menopause-duration/features/filtered_data.tsv
2024-03-06 21:58:40.90691 INFO::Writing filtered, normalized data to file results/maaslin2/asv_menopause-duration/features/filtered_data_norm.tsv
2024-03-06 21:58:40.907857 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/asv_menopause-duration/features/filtered_data_norm_transformed.tsv
2024-03-06 21:58:40.908743 WARNING::Deleting existing residuals file: results/maaslin2/asv_menopause-duration/fits/residuals.rds
2024-03-06 21:58:40.909255 INFO::Writing residuals to file results/maaslin2/asv_menopause-duration/fits/residuals.rds
2024-03-06 21:58:40.909741 WARNING::Deleting existing fitted file: results/maaslin2/asv_menopause-duration/fits/fitted.rds
2024-03-06 21:58:40.910123 INFO::Writing fitted values to file results/maaslin2/asv_menopause-duration/fits/fitted.rds
2024-03-06 21:58:40.910533 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/asv_menopause-duration/all_results.tsv
2024-03-06 21:58:40.911204 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/asv_menopause-duration/significant_results.tsv
2024-03-06 21:58:40.911583 INFO::Writing heatmap of significant results to file: results/maaslin2/asv_menopause-duration/heatmap.pdf
[1] "There are no associations to plot!"
2024-03-06 21:58:40.912631 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/asv_menopause-duration
[1] "There are no associations to plot!"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># no significant effect on ASV level, try glom at genuslevel</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>ps_menopause_genus <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_menopause, <span class="at">taxrank =</span> <span class="st">"Genus"</span>)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>genus_durationMenopause <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_menopause_genus),</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_menopause_genus), <span class="st">"data.frame"</span>),</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/genus_menopause-duration"</span>,</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">5</span>,</span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.2</span>,</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"DurationMenopause"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:58:45.381516 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/genus_menopause-duration/maaslin2.log"
2024-03-06 21:58:45.55419 INFO::Writing function arguments to log file
2024-03-06 21:58:45.556922 INFO::Verifying options selected are valid
2024-03-06 21:58:45.557199 INFO::Determining format of input files
2024-03-06 21:58:45.557443 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:58:45.559453 INFO::Formula for fixed effects: expr ~  DurationMenopause
2024-03-06 21:58:45.560162 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:58:45.560616 INFO::Total samples in data: 25
2024-03-06 21:58:45.560876 INFO::Min samples required with min abundance for a feature not to be filtered: 5.000000
2024-03-06 21:58:45.562438 INFO::Total filtered features: 316
2024-03-06 21:58:45.562779 INFO::Filtered feature names from abundance and prevalence filtering: ASV5, ASV11, ASV12, ASV13, ASV20, ASV21, ASV23, ASV24, ASV34, ASV35, ASV36, ASV38, ASV45, ASV47, ASV48, ASV50, ASV53, ASV57, ASV59, ASV63, ASV68, ASV69, ASV71, ASV73, ASV95, ASV104, ASV115, ASV119, ASV125, ASV128, ASV131, ASV136, ASV148, ASV151, ASV152, ASV155, ASV170, ASV174, ASV191, ASV193, ASV196, ASV201, ASV204, ASV207, ASV208, ASV209, ASV218, ASV224, ASV227, ASV254, ASV255, ASV256, ASV259, ASV264, ASV266, ASV267, ASV275, ASV276, ASV277, ASV280, ASV284, ASV291, ASV300, ASV307, ASV311, ASV313, ASV315, ASV325, ASV330, ASV331, ASV338, ASV347, ASV349, ASV353, ASV362, ASV363, ASV371, ASV375, ASV378, ASV384, ASV390, ASV395, ASV398, ASV405, ASV410, ASV414, ASV417, ASV419, ASV421, ASV431, ASV432, ASV440, ASV443, ASV459, ASV466, ASV470, ASV475, ASV479, ASV482, ASV483, ASV494, ASV518, ASV526, ASV532, ASV556, ASV557, ASV558, ASV570, ASV571, ASV573, ASV586, ASV601, ASV605, ASV610, ASV612, ASV615, ASV641, ASV642, ASV653, ASV658, ASV664, ASV671, ASV674, ASV683, ASV685, ASV686, ASV690, ASV693, ASV699, ASV709, ASV711, ASV728, ASV734, ASV739, ASV740, ASV744, ASV764, ASV785, ASV789, ASV793, ASV801, ASV803, ASV808, ASV821, ASV823, ASV824, ASV829, ASV840, ASV849, ASV853, ASV855, ASV858, ASV860, ASV863, ASV864, ASV865, ASV881, ASV891, ASV892, ASV913, ASV918, ASV925, ASV927, ASV936, ASV939, ASV948, ASV956, ASV967, ASV979, ASV980, ASV984, ASV989, ASV991, ASV1000, ASV1004, ASV1009, ASV1011, ASV1026, ASV1038, ASV1040, ASV1044, ASV1050, ASV1053, ASV1054, ASV1055, ASV1058, ASV1068, ASV1070, ASV1083, ASV1088, ASV1089, ASV1103, ASV1109, ASV1111, ASV1117, ASV1121, ASV1123, ASV1132, ASV1134, ASV1136, ASV1145, ASV1151, ASV1159, ASV1160, ASV1172, ASV1176, ASV1194, ASV1195, ASV1196, ASV1197, ASV1200, ASV1201, ASV1203, ASV1221, ASV1222, ASV1235, ASV1236, ASV1238, ASV1242, ASV1243, ASV1249, ASV1251, ASV1253, ASV1255, ASV1260, ASV1275, ASV1282, ASV1284, ASV1285, ASV1291, ASV1293, ASV1304, ASV1312, ASV1316, ASV1317, ASV1321, ASV1322, ASV1344, ASV1353, ASV1382, ASV1383, ASV1411, ASV1412, ASV1414, ASV1428, ASV1438, ASV1439, ASV1440, ASV1476, ASV1483, ASV1489, ASV1499, ASV1501, ASV1503, ASV1517, ASV1520, ASV1534, ASV1535, ASV1536, ASV1537, ASV1539, ASV1550, ASV1565, ASV1601, ASV1602, ASV1609, ASV1615, ASV1616, ASV1617, ASV1624, ASV1625, ASV1626, ASV1627, ASV1630, ASV1636, ASV1639, ASV1647, ASV1672, ASV1724, ASV1732, ASV1734, ASV1767, ASV1769, ASV1774, ASV1778, ASV1783, ASV1812, ASV1857, ASV1859, ASV1862, ASV1869, ASV1887, ASV1926, ASV1935, ASV1980, ASV2008, ASV2029, ASV2033, ASV2047, ASV2071, ASV2087, ASV2119, ASV2149, ASV2159, ASV2214, ASV2267, ASV2269, ASV2301, ASV2322, ASV2328, ASV2329, ASV2331, ASV2387, ASV2395, ASV2396, ASV2480
2024-03-06 21:58:45.563228 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:58:45.563476 INFO::Filtered feature names from variance filtering:
2024-03-06 21:58:45.563696 INFO::Running selected normalization method: TSS
2024-03-06 21:58:45.564146 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:58:45.567505 INFO::Running selected transform method: LOG
2024-03-06 21:58:45.568039 INFO::Running selected analysis method: LM
2024-03-06 21:58:45.568362 INFO::Fitting model to feature number 1, ASV2
2024-03-06 21:58:45.569632 INFO::Fitting model to feature number 2, ASV7
2024-03-06 21:58:45.570694 INFO::Fitting model to feature number 3, ASV18
2024-03-06 21:58:45.571728 INFO::Fitting model to feature number 4, ASV33
2024-03-06 21:58:45.574589 INFO::Counting total values for each feature
2024-03-06 21:58:45.575296 INFO::Writing filtered data to file results/maaslin2/genus_menopause-duration/features/filtered_data.tsv
2024-03-06 21:58:45.576513 INFO::Writing filtered, normalized data to file results/maaslin2/genus_menopause-duration/features/filtered_data_norm.tsv
2024-03-06 21:58:45.577024 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/genus_menopause-duration/features/filtered_data_norm_transformed.tsv
2024-03-06 21:58:45.577553 WARNING::Deleting existing residuals file: results/maaslin2/genus_menopause-duration/fits/residuals.rds
2024-03-06 21:58:45.578017 INFO::Writing residuals to file results/maaslin2/genus_menopause-duration/fits/residuals.rds
2024-03-06 21:58:45.578431 WARNING::Deleting existing fitted file: results/maaslin2/genus_menopause-duration/fits/fitted.rds
2024-03-06 21:58:45.578807 INFO::Writing fitted values to file results/maaslin2/genus_menopause-duration/fits/fitted.rds
2024-03-06 21:58:45.57917 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/genus_menopause-duration/all_results.tsv
2024-03-06 21:58:45.579717 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/genus_menopause-duration/significant_results.tsv
2024-03-06 21:58:45.580063 INFO::Writing heatmap of significant results to file: results/maaslin2/genus_menopause-duration/heatmap.pdf
[1] "There are no associations to plot!"
2024-03-06 21:58:45.580553 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/genus_menopause-duration
[1] "There are no associations to plot!"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># no significant effect on genus level, try on family level</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>ps_menopause_family <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_menopause, <span class="at">taxrank =</span> <span class="st">"Family"</span>)</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>family_durationMenopause <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_menopause_family),</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_menopause_family), <span class="st">"data.frame"</span>),</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/family_menopause-duration"</span>,</span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">10</span>,</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.3</span>,</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"DurationMenopause"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:58:47.396412 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/family_menopause-duration/maaslin2.log"
2024-03-06 21:58:47.460983 INFO::Writing function arguments to log file
2024-03-06 21:58:47.464302 INFO::Verifying options selected are valid
2024-03-06 21:58:47.464571 INFO::Determining format of input files
2024-03-06 21:58:47.464795 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:58:47.46591 INFO::Formula for fixed effects: expr ~  DurationMenopause
2024-03-06 21:58:47.466277 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:58:47.466582 INFO::Total samples in data: 25
2024-03-06 21:58:47.466827 INFO::Min samples required with min abundance for a feature not to be filtered: 7.500000
2024-03-06 21:58:47.467619 INFO::Total filtered features: 117
2024-03-06 21:58:47.467885 INFO::Filtered feature names from abundance and prevalence filtering: ASV5, ASV11, ASV13, ASV20, ASV33, ASV34, ASV35, ASV36, ASV47, ASV50, ASV53, ASV59, ASV63, ASV68, ASV71, ASV73, ASV125, ASV128, ASV131, ASV148, ASV152, ASV157, ASV170, ASV174, ASV193, ASV201, ASV204, ASV224, ASV227, ASV259, ASV267, ASV273, ASV275, ASV276, ASV315, ASV362, ASV363, ASV378, ASV379, ASV382, ASV390, ASV398, ASV410, ASV414, ASV419, ASV439, ASV443, ASV466, ASV471, ASV482, ASV483, ASV494, ASV532, ASV583, ASV601, ASV610, ASV634, ASV641, ASV658, ASV683, ASV690, ASV764, ASV801, ASV823, ASV858, ASV864, ASV891, ASV913, ASV925, ASV956, ASV984, ASV1000, ASV1009, ASV1026, ASV1049, ASV1055, ASV1068, ASV1083, ASV1088, ASV1089, ASV1103, ASV1151, ASV1159, ASV1193, ASV1201, ASV1243, ASV1249, ASV1284, ASV1291, ASV1304, ASV1321, ASV1322, ASV1412, ASV1428, ASV1439, ASV1483, ASV1503, ASV1516, ASV1537, ASV1566, ASV1573, ASV1609, ASV1615, ASV1630, ASV1636, ASV1639, ASV1641, ASV1734, ASV1769, ASV1775, ASV1857, ASV1869, ASV1887, ASV1929, ASV2071, ASV2197, ASV2384
2024-03-06 21:58:47.46824 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:58:47.468472 INFO::Filtered feature names from variance filtering:
2024-03-06 21:58:47.46868 INFO::Running selected normalization method: TSS
2024-03-06 21:58:47.46905 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:58:47.471391 INFO::Running selected transform method: LOG
2024-03-06 21:58:47.471745 INFO::Running selected analysis method: LM
2024-03-06 21:58:47.472014 INFO::Fitting model to feature number 1, ASV2
2024-03-06 21:58:47.47304 INFO::Fitting model to feature number 2, ASV7
2024-03-06 21:58:47.473921 INFO::Fitting model to feature number 3, ASV12
2024-03-06 21:58:47.474767 INFO::Fitting model to feature number 4, ASV18
2024-03-06 21:58:47.477207 INFO::Counting total values for each feature
2024-03-06 21:58:47.477793 INFO::Writing filtered data to file results/maaslin2/family_menopause-duration/features/filtered_data.tsv
2024-03-06 21:58:47.478402 INFO::Writing filtered, normalized data to file results/maaslin2/family_menopause-duration/features/filtered_data_norm.tsv
2024-03-06 21:58:47.478913 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/family_menopause-duration/features/filtered_data_norm_transformed.tsv
2024-03-06 21:58:47.479442 WARNING::Deleting existing residuals file: results/maaslin2/family_menopause-duration/fits/residuals.rds
2024-03-06 21:58:47.479892 INFO::Writing residuals to file results/maaslin2/family_menopause-duration/fits/residuals.rds
2024-03-06 21:58:47.480292 WARNING::Deleting existing fitted file: results/maaslin2/family_menopause-duration/fits/fitted.rds
2024-03-06 21:58:47.480637 INFO::Writing fitted values to file results/maaslin2/family_menopause-duration/fits/fitted.rds
2024-03-06 21:58:47.480991 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/family_menopause-duration/all_results.tsv
2024-03-06 21:58:47.481528 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/family_menopause-duration/significant_results.tsv
2024-03-06 21:58:47.481864 INFO::Writing heatmap of significant results to file: results/maaslin2/family_menopause-duration/heatmap.pdf
[1] "There are no associations to plot!"
2024-03-06 21:58:47.482309 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/family_menopause-duration
[1] "There are no associations to plot!"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Alpha diversity by duration of menopause</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>ps_menopause_abs <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_samples, grp <span class="sc">==</span><span class="st">"01_postmenopausal"</span>)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>adiv_menopause <span class="ot">=</span> <span class="fu">plot_richness</span>(ps_menopause_abs, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">"Observed"</span>, <span class="st">"Shannon"</span>, <span class="st">"InvSimpson"</span>))</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>adiv_menopause<span class="sc">$</span>data <span class="sc">%&gt;%</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(DurationMenopause, variable, value) <span class="sc">%&gt;%</span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cor_test</span>(value, DurationMenopause) <span class="sc">%&gt;%</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 13%">
<col style="width: 7%">
<col style="width: 21%">
<col style="width: 6%">
<col style="width: 12%">
<col style="width: 7%">
<col style="width: 10%">
<col style="width: 12%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">variable</th>
<th style="text-align: left;">var1</th>
<th style="text-align: left;">var2</th>
<th style="text-align: right;">cor</th>
<th style="text-align: right;">statistic</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">conf.low</th>
<th style="text-align: right;">conf.high</th>
<th style="text-align: left;">method</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationMenopause</td>
<td style="text-align: right;">0.18</td>
<td style="text-align: right;">0.861</td>
<td style="text-align: right;">0.398</td>
<td style="text-align: right;">-0.235</td>
<td style="text-align: right;">0.534</td>
<td style="text-align: left;">Pearson</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationMenopause</td>
<td style="text-align: right;">0.11</td>
<td style="text-align: right;">0.528</td>
<td style="text-align: right;">0.602</td>
<td style="text-align: right;">-0.299</td>
<td style="text-align: right;">0.484</td>
<td style="text-align: left;">Pearson</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationMenopause</td>
<td style="text-align: right;">0.16</td>
<td style="text-align: right;">0.767</td>
<td style="text-align: right;">0.451</td>
<td style="text-align: right;">-0.253</td>
<td style="text-align: right;">0.521</td>
<td style="text-align: left;">Pearson</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>adiv_menopause<span class="sc">$</span>data <span class="sc">%&gt;%</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> DurationMenopause, <span class="at">y =</span> value)) <span class="sc">+</span> </span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span><span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>variable, <span class="at">scales =</span> <span class="st">"free_y"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/postmenopausal_menopause%20duration-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>BC_menopause_genus <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(<span class="fu">otu_table</span>(ps_menopause_genus))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>menopauseduration_adonis_bc <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(BC_menopause_genus <span class="sc">~</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_menopause_genus)<span class="sc">$</span>DurationMenopause)</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a><span class="co"># no significant effect on the beta diversity level as well</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary statistics of duration menopause</span></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(ps_menopause_genus)<span class="sc">$</span>DurationMenopause <span class="sc">%&gt;%</span></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1.00    7.00   13.00   14.48   22.00   28.00 </code></pre>
</div>
</div>
<p>There is no significant correlation of duration of menopause (years) with alpha diversity of vaginal microbiomes and overall community composition (beta-diversity). Duration of menopause in this sample is rather varied ranging from 1 to 28 years (mean 14.5 years).</p>
</section>
<section id="duration-of-gender-affirming-hormone-therapy-gaht" class="level3">
<h3 class="anchored" data-anchor-id="duration-of-gender-affirming-hormone-therapy-gaht">Duration of gender-affirming hormone therapy (GAHT)</h3>
<p>This analysis is only performed in the trans-men group.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># only 03_trans samples</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>ps_trans <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_samples, grp <span class="sc">==</span><span class="st">"03_trans-men"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x) <span class="sc">*</span> <span class="dv">100</span>) <span class="co"># relative abundances are necessary</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>asv_GHAT <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_trans),</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_trans), <span class="st">"data.frame"</span>),</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/asv_GHAT_Length"</span>,</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">1</span>,</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.1</span>,</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"GHAT_Length"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:58:47.752434 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/asv_GHAT_Length/maaslin2.log"
2024-03-06 21:58:49.153699 INFO::Writing function arguments to log file
2024-03-06 21:58:49.15652 INFO::Verifying options selected are valid
2024-03-06 21:58:49.15681 INFO::Determining format of input files
2024-03-06 21:58:49.157064 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:58:49.167435 INFO::Formula for fixed effects: expr ~  GHAT_Length
2024-03-06 21:58:49.167911 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:58:49.16817 INFO::Total samples in data: 25
2024-03-06 21:58:49.16839 INFO::Min samples required with min abundance for a feature not to be filtered: 2.500000
2024-03-06 21:58:49.178432 INFO::Total filtered features: 2511
2024-03-06 21:58:49.179375 INFO::Filtered feature names from abundance and prevalence filtering: ASV1, ASV5, ASV8, ASV13, ASV16, ASV19, ASV20, ASV22, ASV26, ASV30, ASV33, ASV37, ASV38, ASV41, ASV42, ASV43, ASV44, ASV50, ASV51, ASV52, ASV53, ASV55, ASV56, ASV58, ASV59, ASV60, ASV61, ASV63, ASV65, ASV67, ASV68, ASV69, ASV71, ASV72, ASV75, ASV77, ASV78, ASV79, ASV80, ASV81, ASV82, ASV84, ASV85, ASV86, ASV87, ASV88, ASV89, ASV90, ASV93, ASV94, ASV95, ASV96, ASV98, ASV99, ASV100, ASV101, ASV103, ASV105, ASV106, ASV107, ASV108, ASV109, ASV110, ASV111, ASV112, ASV113, ASV114, ASV115, ASV117, ASV118, ASV119, ASV120, ASV121, ASV122, ASV123, ASV124, ASV125, ASV126, ASV127, ASV128, ASV129, ASV130, ASV131, ASV132, ASV133, ASV134, ASV135, ASV136, ASV137, ASV138, ASV139, ASV140, ASV141, ASV142, ASV143, ASV145, ASV146, ASV147, ASV148, ASV149, ASV150, ASV151, ASV152, ASV153, ASV154, ASV155, ASV156, ASV157, ASV158, ASV159, ASV160, ASV161, ASV162, ASV163, ASV164, ASV165, ASV166, ASV167, ASV168, ASV169, ASV170, ASV171, ASV172, ASV173, ASV174, ASV175, ASV176, ASV177, ASV178, ASV179, ASV180, ASV181, ASV182, ASV183, ASV184, ASV185, ASV186, ASV187, ASV188, ASV189, ASV190, ASV191, ASV192, ASV193, ASV194, ASV195, ASV196, ASV197, ASV198, ASV199, ASV200, ASV201, ASV202, ASV203, ASV204, ASV205, ASV206, ASV207, ASV208, ASV209, ASV210, ASV211, ASV212, ASV213, ASV214, ASV215, ASV216, ASV217, ASV218, ASV219, ASV220, ASV221, ASV222, ASV223, ASV224, ASV225, ASV226, ASV227, ASV228, ASV229, ASV230, ASV231, ASV232, ASV233, ASV234, ASV235, ASV236, ASV237, ASV238, ASV239, ASV240, ASV241, ASV242, ASV243, ASV244, ASV245, ASV246, ASV247, ASV248, ASV249, ASV250, ASV251, ASV252, ASV253, ASV254, ASV255, ASV256, ASV257, ASV258, ASV259, ASV260, ASV261, ASV262, ASV263, ASV264, ASV265, ASV266, ASV267, ASV268, ASV269, ASV270, ASV271, ASV272, ASV273, ASV274, ASV275, ASV276, ASV277, ASV278, ASV279, ASV280, ASV281, ASV282, ASV283, ASV284, ASV285, ASV286, ASV287, ASV288, ASV289, ASV290, ASV291, ASV292, ASV293, ASV294, ASV295, ASV296, ASV297, ASV298, ASV299, ASV300, ASV301, ASV302, ASV303, ASV304, ASV305, ASV306, ASV307, ASV308, ASV309, ASV310, ASV311, ASV312, ASV313, ASV314, ASV315, ASV316, ASV317, ASV318, ASV319, ASV320, ASV321, ASV322, ASV323, ASV324, ASV325, ASV326, ASV327, ASV328, ASV329, ASV330, ASV331, ASV332, ASV333, ASV334, ASV335, ASV336, ASV337, ASV338, ASV339, ASV340, ASV341, ASV342, ASV343, ASV344, ASV345, ASV346, ASV347, ASV348, ASV349, ASV350, ASV351, ASV352, ASV353, ASV354, ASV355, ASV356, ASV357, ASV358, ASV359, ASV360, ASV361, ASV362, ASV363, ASV364, ASV365, ASV366, ASV367, ASV368, ASV369, ASV370, ASV371, ASV372, ASV373, ASV374, ASV375, ASV376, ASV377, ASV378, ASV379, ASV380, ASV381, ASV382, ASV383, ASV384, ASV385, ASV386, ASV387, ASV388, ASV389, ASV390, ASV391, ASV392, ASV393, ASV394, ASV395, ASV396, ASV397, ASV398, ASV399, ASV400, ASV401, ASV402, ASV403, ASV404, ASV405, ASV406, ASV407, ASV408, ASV409, ASV410, ASV411, ASV412, ASV413, ASV414, ASV415, ASV416, ASV417, ASV418, ASV419, ASV420, ASV421, ASV422, ASV423, ASV424, ASV425, ASV426, ASV427, ASV428, ASV429, ASV430, ASV431, ASV432, ASV433, ASV434, ASV435, ASV436, ASV437, ASV438, ASV439, ASV440, ASV441, ASV442, ASV443, ASV444, ASV445, ASV446, ASV447, ASV448, ASV449, ASV450, ASV451, ASV452, ASV453, ASV454, ASV455, ASV456, ASV457, ASV458, ASV459, ASV460, ASV461, ASV462, ASV463, ASV464, ASV465, ASV466, ASV467, ASV468, ASV469, ASV470, ASV471, ASV472, ASV473, ASV474, ASV475, ASV476, ASV477, ASV478, ASV479, ASV480, ASV481, ASV482, ASV483, ASV484, ASV485, ASV486, ASV487, ASV488, ASV489, ASV490, ASV491, ASV492, ASV493, ASV494, ASV495, ASV496, ASV497, ASV498, ASV499, ASV500, ASV501, ASV502, ASV503, ASV504, ASV505, ASV506, ASV507, ASV508, ASV509, ASV510, ASV511, ASV512, ASV513, ASV514, ASV515, ASV516, ASV517, ASV518, ASV519, ASV520, ASV521, ASV522, ASV523, ASV524, ASV525, ASV526, ASV527, ASV528, ASV529, ASV530, ASV531, ASV532, ASV533, ASV534, ASV535, ASV536, ASV537, ASV538, ASV539, ASV540, ASV541, ASV542, ASV543, ASV544, ASV545, ASV546, ASV547, ASV548, ASV549, ASV550, ASV551, ASV552, ASV553, ASV554, ASV555, ASV556, ASV557, ASV558, ASV559, ASV560, ASV561, ASV562, ASV563, ASV564, ASV565, ASV566, ASV567, ASV568, ASV569, ASV570, ASV571, ASV572, ASV573, ASV574, ASV575, ASV576, ASV577, ASV578, ASV579, ASV580, ASV581, ASV582, ASV583, ASV584, ASV585, ASV586, ASV587, ASV588, ASV589, ASV590, ASV591, ASV592, ASV593, ASV594, ASV595, ASV596, ASV597, ASV598, ASV599, ASV600, ASV601, ASV602, ASV603, ASV604, ASV605, ASV606, ASV607, ASV608, ASV609, ASV610, ASV611, ASV612, ASV613, ASV614, ASV615, ASV616, ASV617, ASV618, ASV619, ASV620, ASV621, ASV622, ASV623, ASV624, ASV625, ASV626, ASV627, ASV628, ASV629, ASV630, ASV631, ASV632, ASV633, ASV634, ASV635, ASV636, ASV637, ASV638, ASV639, ASV640, ASV641, ASV642, ASV643, ASV644, ASV645, ASV646, ASV647, ASV648, ASV649, ASV650, ASV651, ASV652, ASV653, ASV654, ASV655, ASV656, ASV657, ASV658, ASV659, ASV660, ASV661, ASV662, ASV663, ASV664, ASV665, ASV666, ASV667, ASV668, ASV669, ASV670, ASV671, ASV672, ASV673, ASV674, ASV675, ASV676, ASV677, ASV678, ASV679, ASV680, ASV681, ASV682, ASV683, ASV684, ASV685, ASV686, ASV687, ASV688, ASV689, ASV690, ASV691, ASV692, ASV693, ASV694, ASV695, ASV696, ASV697, ASV698, ASV699, ASV700, ASV701, ASV702, ASV703, ASV704, ASV705, ASV706, ASV707, ASV708, ASV709, ASV710, ASV711, ASV712, ASV713, ASV714, ASV715, ASV716, ASV717, ASV718, ASV719, ASV720, ASV721, ASV722, ASV723, ASV724, ASV725, ASV726, ASV727, ASV728, ASV729, ASV730, ASV731, ASV732, ASV733, ASV734, ASV735, ASV736, ASV737, ASV738, ASV739, ASV740, ASV741, ASV742, ASV743, ASV744, ASV745, ASV746, ASV747, ASV748, ASV749, ASV750, ASV751, ASV752, ASV753, ASV754, ASV755, ASV756, ASV757, ASV758, ASV759, ASV760, ASV761, ASV762, ASV763, ASV764, ASV765, ASV766, ASV767, ASV768, ASV769, ASV770, ASV771, ASV772, ASV773, ASV774, ASV775, ASV776, ASV777, ASV778, ASV779, ASV780, ASV781, ASV782, ASV783, ASV784, ASV785, ASV786, ASV787, ASV788, ASV789, ASV790, ASV791, ASV792, ASV793, ASV794, ASV795, ASV796, ASV797, ASV798, ASV799, ASV800, ASV801, ASV802, ASV803, ASV804, ASV805, ASV806, ASV807, ASV808, ASV809, ASV810, ASV811, ASV812, ASV813, ASV814, ASV815, ASV816, ASV817, ASV818, ASV819, ASV820, ASV821, ASV822, ASV823, ASV824, ASV825, ASV826, ASV827, ASV828, ASV829, ASV830, ASV831, ASV832, ASV833, ASV834, ASV835, ASV836, ASV837, ASV838, ASV839, ASV840, ASV841, ASV842, ASV843, ASV844, ASV845, ASV846, ASV847, ASV848, ASV849, ASV850, ASV851, ASV852, ASV853, ASV854, ASV855, ASV856, ASV857, ASV858, ASV859, ASV860, ASV861, ASV862, ASV863, ASV864, ASV865, ASV866, ASV867, ASV868, ASV869, ASV870, ASV871, ASV872, ASV873, ASV874, ASV875, ASV876, ASV877, ASV878, ASV879, ASV880, ASV881, ASV882, ASV883, ASV884, ASV885, ASV886, ASV887, ASV888, ASV889, ASV890, ASV891, ASV892, ASV893, ASV894, ASV895, ASV896, ASV897, ASV898, ASV899, ASV900, ASV901, ASV902, ASV903, ASV904, ASV905, ASV906, ASV907, ASV908, ASV909, ASV910, ASV911, ASV912, ASV913, ASV914, ASV915, ASV916, ASV917, ASV918, ASV919, ASV920, ASV921, ASV922, ASV923, ASV924, ASV925, ASV926, ASV927, ASV928, ASV929, ASV930, ASV931, ASV932, ASV933, ASV934, ASV935, ASV936, ASV937, ASV938, ASV939, ASV940, ASV941, ASV942, ASV943, ASV944, ASV945, ASV946, ASV947, ASV948, ASV949, ASV950, ASV951, ASV952, ASV953, ASV954, ASV955, ASV956, ASV957, ASV958, ASV959, ASV960, ASV961, ASV962, ASV963, ASV964, ASV965, ASV966, ASV967, ASV968, ASV969, ASV970, ASV971, ASV972, ASV973, ASV974, ASV975, ASV976, ASV977, ASV978, ASV979, ASV980, ASV981, ASV982, ASV983, ASV984, ASV985, ASV986, ASV987, ASV988, ASV989, ASV990, ASV991, ASV992, ASV993, ASV994, ASV995, ASV996, ASV997, ASV998, ASV999, ASV1000, ASV1001, ASV1002, ASV1003, ASV1004, ASV1005, ASV1006, ASV1007, ASV1008, ASV1009, ASV1010, ASV1011, ASV1012, ASV1013, ASV1014, ASV1015, ASV1016, ASV1017, ASV1018, ASV1019, ASV1020, ASV1021, ASV1022, ASV1023, ASV1024, ASV1025, ASV1026, ASV1027, ASV1028, ASV1029, ASV1030, ASV1031, ASV1032, ASV1033, ASV1034, ASV1035, ASV1036, ASV1037, ASV1038, ASV1039, ASV1040, ASV1041, ASV1042, ASV1043, ASV1044, ASV1045, ASV1046, ASV1047, ASV1048, ASV1049, ASV1050, ASV1051, ASV1052, ASV1053, ASV1054, ASV1055, ASV1056, ASV1057, ASV1058, ASV1059, ASV1060, ASV1061, ASV1062, ASV1063, ASV1064, ASV1065, ASV1066, ASV1067, ASV1068, ASV1069, ASV1070, ASV1071, ASV1072, ASV1073, ASV1074, ASV1075, ASV1076, ASV1077, ASV1078, ASV1079, ASV1080, ASV1081, ASV1082, ASV1083, ASV1084, ASV1085, ASV1086, ASV1087, ASV1088, ASV1089, ASV1090, ASV1091, ASV1092, ASV1093, ASV1094, ASV1095, ASV1096, ASV1097, ASV1098, ASV1099, ASV1100, ASV1101, ASV1102, ASV1103, ASV1104, ASV1105, ASV1106, ASV1107, ASV1108, ASV1109, ASV1110, ASV1111, ASV1112, ASV1113, ASV1114, ASV1115, ASV1116, ASV1117, ASV1118, ASV1119, ASV1120, ASV1121, ASV1122, ASV1123, ASV1124, ASV1125, ASV1126, ASV1127, ASV1128, ASV1129, ASV1130, ASV1131, ASV1132, ASV1133, ASV1134, ASV1135, ASV1136, ASV1137, ASV1138, ASV1139, ASV1140, ASV1141, ASV1142, ASV1143, ASV1144, ASV1145, ASV1146, ASV1147, ASV1148, ASV1149, ASV1150, ASV1151, ASV1152, ASV1153, ASV1154, ASV1155, ASV1156, ASV1157, ASV1158, ASV1159, ASV1160, ASV1161, ASV1162, ASV1163, ASV1164, ASV1165, ASV1166, ASV1167, ASV1168, ASV1169, ASV1170, ASV1171, ASV1172, ASV1173, ASV1174, ASV1175, ASV1176, ASV1177, ASV1178, ASV1179, ASV1180, ASV1181, ASV1182, ASV1183, ASV1184, ASV1185, ASV1186, ASV1187, ASV1188, ASV1189, ASV1190, ASV1191, ASV1192, ASV1193, ASV1194, ASV1195, ASV1196, ASV1197, ASV1198, ASV1199, ASV1200, ASV1201, ASV1202, ASV1203, ASV1204, ASV1205, ASV1206, ASV1207, ASV1208, ASV1209, ASV1210, ASV1211, ASV1212, ASV1213, ASV1214, ASV1215, ASV1216, ASV1217, ASV1218, ASV1219, ASV1220, ASV1221, ASV1222, ASV1223, ASV1224, ASV1225, ASV1226, ASV1227, ASV1228, ASV1229, ASV1230, ASV1231, ASV1232, ASV1233, ASV1234, ASV1235, ASV1236, ASV1237, ASV1238, ASV1239, ASV1240, ASV1241, ASV1242, ASV1243, ASV1244, ASV1245, ASV1246, ASV1247, ASV1248, ASV1249, ASV1250, ASV1251, ASV1252, ASV1253, ASV1254, ASV1255, ASV1256, ASV1257, ASV1258, ASV1259, ASV1260, ASV1261, ASV1262, ASV1263, ASV1264, ASV1265, ASV1266, ASV1267, ASV1268, ASV1269, ASV1270, ASV1271, ASV1272, ASV1273, ASV1274, ASV1275, ASV1276, ASV1277, ASV1278, ASV1279, ASV1280, ASV1281, ASV1282, ASV1283, ASV1284, ASV1285, ASV1286, ASV1287, ASV1288, ASV1289, ASV1290, ASV1291, ASV1292, ASV1293, ASV1294, ASV1295, ASV1296, ASV1297, ASV1298, ASV1299, ASV1300, ASV1301, ASV1302, ASV1303, ASV1304, ASV1305, ASV1306, ASV1307, ASV1308, ASV1309, ASV1310, ASV1311, ASV1312, ASV1313, ASV1314, ASV1315, ASV1316, ASV1317, ASV1318, ASV1319, ASV1320, ASV1321, ASV1322, ASV1323, ASV1324, ASV1325, ASV1326, ASV1327, ASV1328, ASV1329, ASV1330, ASV1331, ASV1332, ASV1333, ASV1334, ASV1335, ASV1336, ASV1337, ASV1338, ASV1339, ASV1340, ASV1341, ASV1342, ASV1343, ASV1344, ASV1345, ASV1346, ASV1347, ASV1348, ASV1349, ASV1350, ASV1351, ASV1352, ASV1353, ASV1354, ASV1355, ASV1356, ASV1357, ASV1358, ASV1359, ASV1360, ASV1361, ASV1362, ASV1363, ASV1364, ASV1365, ASV1366, ASV1367, ASV1368, ASV1369, ASV1370, ASV1371, ASV1372, ASV1373, ASV1374, ASV1375, ASV1376, ASV1377, ASV1378, ASV1379, ASV1380, ASV1381, ASV1382, ASV1383, ASV1384, ASV1385, ASV1386, ASV1387, ASV1388, ASV1389, ASV1390, ASV1391, ASV1392, ASV1393, ASV1394, ASV1395, ASV1396, ASV1397, ASV1398, ASV1399, ASV1400, ASV1401, ASV1402, ASV1403, ASV1404, ASV1405, ASV1406, ASV1407, ASV1408, ASV1409, ASV1410, ASV1411, ASV1412, ASV1413, ASV1414, ASV1415, ASV1416, ASV1417, ASV1418, ASV1419, ASV1420, ASV1421, ASV1422, ASV1423, ASV1424, ASV1425, ASV1426, ASV1427, ASV1428, ASV1429, ASV1430, ASV1431, ASV1432, ASV1433, ASV1434, ASV1435, ASV1436, ASV1437, ASV1438, ASV1439, ASV1440, ASV1441, ASV1442, ASV1443, ASV1444, ASV1445, ASV1446, ASV1447, ASV1448, ASV1449, ASV1450, ASV1451, ASV1452, ASV1453, ASV1454, ASV1455, ASV1456, ASV1457, ASV1458, ASV1459, ASV1460, ASV1461, ASV1462, ASV1463, ASV1464, ASV1465, ASV1466, ASV1467, ASV1468, ASV1469, ASV1470, ASV1471, ASV1472, ASV1473, ASV1474, ASV1475, ASV1476, ASV1477, ASV1478, ASV1479, ASV1480, ASV1481, ASV1482, ASV1483, ASV1484, ASV1485, ASV1486, ASV1487, ASV1488, ASV1489, ASV1490, ASV1491, ASV1492, ASV1493, ASV1494, ASV1495, ASV1496, ASV1497, ASV1498, ASV1499, ASV1500, ASV1501, ASV1502, ASV1503, ASV1504, ASV1505, ASV1506, ASV1507, ASV1508, ASV1509, ASV1510, ASV1511, ASV1512, ASV1513, ASV1514, ASV1515, ASV1516, ASV1517, ASV1518, ASV1519, ASV1520, ASV1521, ASV1522, ASV1523, ASV1524, ASV1525, ASV1526, ASV1527, ASV1528, ASV1529, ASV1530, ASV1531, ASV1532, ASV1533, ASV1534, ASV1535, ASV1536, ASV1537, ASV1538, ASV1539, ASV1540, ASV1541, ASV1542, ASV1543, ASV1544, ASV1545, ASV1546, ASV1547, ASV1548, ASV1549, ASV1550, ASV1551, ASV1552, ASV1553, ASV1554, ASV1555, ASV1556, ASV1557, ASV1558, ASV1559, ASV1560, ASV1561, ASV1562, ASV1563, ASV1564, ASV1565, ASV1566, ASV1567, ASV1568, ASV1569, ASV1570, ASV1571, ASV1572, ASV1573, ASV1574, ASV1575, ASV1576, ASV1577, ASV1578, ASV1579, ASV1580, ASV1581, ASV1582, ASV1583, ASV1584, ASV1585, ASV1586, ASV1587, ASV1588, ASV1589, ASV1590, ASV1591, ASV1592, ASV1593, ASV1594, ASV1595, ASV1596, ASV1597, ASV1598, ASV1599, ASV1600, ASV1601, ASV1602, ASV1603, ASV1604, ASV1605, ASV1606, ASV1607, ASV1608, ASV1609, ASV1610, ASV1611, ASV1612, ASV1613, ASV1614, ASV1615, ASV1616, ASV1617, ASV1618, ASV1619, ASV1620, ASV1621, ASV1622, ASV1623, ASV1624, ASV1625, ASV1626, ASV1627, ASV1628, ASV1629, ASV1630, ASV1631, ASV1632, ASV1633, ASV1634, ASV1635, ASV1636, ASV1637, ASV1638, ASV1639, ASV1640, ASV1641, ASV1642, ASV1643, ASV1644, ASV1645, ASV1646, ASV1647, ASV1648, ASV1649, ASV1650, ASV1651, ASV1652, ASV1653, ASV1654, ASV1655, ASV1656, ASV1657, ASV1658, ASV1659, ASV1660, ASV1661, ASV1662, ASV1663, ASV1664, ASV1665, ASV1666, ASV1667, ASV1668, ASV1669, ASV1670, ASV1671, ASV1672, ASV1673, ASV1674, ASV1675, ASV1676, ASV1677, ASV1678, ASV1679, ASV1680, ASV1681, ASV1682, ASV1683, ASV1684, ASV1685, ASV1686, ASV1687, ASV1688, ASV1689, ASV1690, ASV1691, ASV1692, ASV1693, ASV1694, ASV1695, ASV1696, ASV1697, ASV1698, ASV1699, ASV1700, ASV1701, ASV1702, ASV1703, ASV1704, ASV1705, ASV1706, ASV1707, ASV1708, ASV1709, ASV1710, ASV1711, ASV1712, ASV1713, ASV1714, ASV1715, ASV1716, ASV1717, ASV1718, ASV1719, ASV1720, ASV1721, ASV1722, ASV1723, ASV1724, ASV1725, ASV1726, ASV1727, ASV1728, ASV1729, ASV1730, ASV1731, ASV1732, ASV1733, ASV1734, ASV1735, ASV1736, ASV1737, ASV1738, ASV1739, ASV1740, ASV1741, ASV1742, ASV1743, ASV1744, ASV1745, ASV1746, ASV1747, ASV1748, ASV1749, ASV1750, ASV1751, ASV1752, ASV1753, ASV1754, ASV1755, ASV1756, ASV1757, ASV1758, ASV1759, ASV1760, ASV1761, ASV1762, ASV1763, ASV1764, ASV1765, ASV1766, ASV1767, ASV1768, ASV1769, ASV1770, ASV1771, ASV1772, ASV1773, ASV1774, ASV1775, ASV1776, ASV1777, ASV1778, ASV1779, ASV1780, ASV1781, ASV1782, ASV1783, ASV1784, ASV1785, ASV1786, ASV1787, ASV1788, ASV1789, ASV1790, ASV1791, ASV1792, ASV1793, ASV1794, ASV1795, ASV1796, ASV1797, ASV1798, ASV1799, ASV1800, ASV1801, ASV1802, ASV1803, ASV1804, ASV1805, ASV1806, ASV1807, ASV1808, ASV1809, ASV1810, ASV1811, ASV1812, ASV1813, ASV1814, ASV1815, ASV1816, ASV1817, ASV1818, ASV1819, ASV1820, ASV1821, ASV1822, ASV1823, ASV1824, ASV1825, ASV1826, ASV1827, ASV1828, ASV1829, ASV1830, ASV1831, ASV1832, ASV1833, ASV1834, ASV1835, ASV1836, ASV1837, ASV1838, ASV1839, ASV1840, ASV1841, ASV1842, ASV1843, ASV1844, ASV1845, ASV1846, ASV1847, ASV1848, ASV1849, ASV1850, ASV1851, ASV1852, ASV1853, ASV1854, ASV1855, ASV1856, ASV1857, ASV1858, ASV1859, ASV1860, ASV1861, ASV1862, ASV1863, ASV1864, ASV1865, ASV1866, ASV1867, ASV1868, ASV1869, ASV1870, ASV1871, ASV1872, ASV1873, ASV1874, ASV1875, ASV1876, ASV1877, ASV1878, ASV1879, ASV1880, ASV1881, ASV1882, ASV1883, ASV1884, ASV1885, ASV1886, ASV1887, ASV1888, ASV1889, ASV1890, ASV1891, ASV1892, ASV1893, ASV1894, ASV1895, ASV1896, ASV1897, ASV1898, ASV1899, ASV1900, ASV1901, ASV1902, ASV1903, ASV1904, ASV1905, ASV1906, ASV1907, ASV1908, ASV1909, ASV1910, ASV1911, ASV1912, ASV1913, ASV1914, ASV1915, ASV1916, ASV1917, ASV1918, ASV1919, ASV1920, ASV1921, ASV1922, ASV1923, ASV1924, ASV1925, ASV1926, ASV1927, ASV1928, ASV1929, ASV1930, ASV1931, ASV1932, ASV1933, ASV1934, ASV1935, ASV1936, ASV1937, ASV1938, ASV1939, ASV1940, ASV1941, ASV1942, ASV1943, ASV1944, ASV1945, ASV1946, ASV1947, ASV1948, ASV1949, ASV1950, ASV1951, ASV1952, ASV1953, ASV1954, ASV1955, ASV1956, ASV1957, ASV1958, ASV1959, ASV1960, ASV1961, ASV1962, ASV1963, ASV1964, ASV1965, ASV1966, ASV1967, ASV1968, ASV1969, ASV1970, ASV1971, ASV1972, ASV1973, ASV1974, ASV1975, ASV1976, ASV1977, ASV1978, ASV1979, ASV1980, ASV1981, ASV1982, ASV1983, ASV1984, ASV1985, ASV1986, ASV1987, ASV1988, ASV1989, ASV1990, ASV1991, ASV1992, ASV1993, ASV1994, ASV1995, ASV1996, ASV1997, ASV1998, ASV1999, ASV2000, ASV2001, ASV2002, ASV2003, ASV2004, ASV2005, ASV2006, ASV2007, ASV2008, ASV2009, ASV2010, ASV2011, ASV2012, ASV2013, ASV2014, ASV2015, ASV2016, ASV2017, ASV2018, ASV2019, ASV2020, ASV2021, ASV2022, ASV2023, ASV2024, ASV2025, ASV2026, ASV2027, ASV2028, ASV2029, ASV2030, ASV2031, ASV2032, ASV2033, ASV2034, ASV2035, ASV2036, ASV2037, ASV2038, ASV2039, ASV2040, ASV2041, ASV2042, ASV2043, ASV2044, ASV2045, ASV2046, ASV2047, ASV2048, ASV2049, ASV2050, ASV2051, ASV2052, ASV2053, ASV2054, ASV2055, ASV2056, ASV2057, ASV2058, ASV2059, ASV2060, ASV2061, ASV2062, ASV2063, ASV2064, ASV2065, ASV2066, ASV2067, ASV2068, ASV2069, ASV2070, ASV2071, ASV2072, ASV2073, ASV2074, ASV2075, ASV2076, ASV2077, ASV2078, ASV2079, ASV2080, ASV2081, ASV2082, ASV2083, ASV2084, ASV2085, ASV2086, ASV2087, ASV2088, ASV2089, ASV2090, ASV2091, ASV2092, ASV2093, ASV2094, ASV2095, ASV2096, ASV2097, ASV2098, ASV2099, ASV2100, ASV2101, ASV2102, ASV2103, ASV2104, ASV2105, ASV2106, ASV2107, ASV2108, ASV2109, ASV2110, ASV2111, ASV2112, ASV2113, ASV2114, ASV2115, ASV2116, ASV2117, ASV2118, ASV2119, ASV2120, ASV2121, ASV2122, ASV2123, ASV2124, ASV2125, ASV2126, ASV2127, ASV2128, ASV2129, ASV2130, ASV2131, ASV2132, ASV2133, ASV2134, ASV2135, ASV2136, ASV2137, ASV2138, ASV2139, ASV2140, ASV2141, ASV2142, ASV2143, ASV2144, ASV2145, ASV2146, ASV2147, ASV2148, ASV2149, ASV2150, ASV2151, ASV2152, ASV2153, ASV2154, ASV2155, ASV2156, ASV2157, ASV2158, ASV2159, ASV2160, ASV2161, ASV2162, ASV2163, ASV2164, ASV2165, ASV2166, ASV2167, ASV2168, ASV2169, ASV2170, ASV2171, ASV2172, ASV2173, ASV2174, ASV2175, ASV2176, ASV2177, ASV2178, ASV2179, ASV2180, ASV2181, ASV2182, ASV2183, ASV2184, ASV2185, ASV2186, ASV2187, ASV2188, ASV2189, ASV2190, ASV2191, ASV2192, ASV2193, ASV2194, ASV2195, ASV2196, ASV2197, ASV2198, ASV2199, ASV2200, ASV2201, ASV2202, ASV2203, ASV2204, ASV2205, ASV2206, ASV2207, ASV2208, ASV2209, ASV2210, ASV2211, ASV2212, ASV2213, ASV2214, ASV2215, ASV2216, ASV2217, ASV2218, ASV2219, ASV2220, ASV2221, ASV2222, ASV2223, ASV2224, ASV2225, ASV2226, ASV2227, ASV2228, ASV2229, ASV2230, ASV2231, ASV2232, ASV2233, ASV2234, ASV2235, ASV2236, ASV2237, ASV2238, ASV2239, ASV2240, ASV2241, ASV2242, ASV2243, ASV2244, ASV2245, ASV2246, ASV2247, ASV2248, ASV2249, ASV2250, ASV2251, ASV2252, ASV2253, ASV2254, ASV2255, ASV2256, ASV2257, ASV2258, ASV2259, ASV2260, ASV2261, ASV2262, ASV2263, ASV2264, ASV2265, ASV2266, ASV2267, ASV2268, ASV2269, ASV2270, ASV2271, ASV2272, ASV2273, ASV2274, ASV2275, ASV2276, ASV2277, ASV2278, ASV2279, ASV2280, ASV2281, ASV2282, ASV2283, ASV2284, ASV2285, ASV2286, ASV2287, ASV2288, ASV2289, ASV2290, ASV2291, ASV2292, ASV2293, ASV2294, ASV2295, ASV2296, ASV2297, ASV2298, ASV2299, ASV2300, ASV2301, ASV2302, ASV2303, ASV2304, ASV2305, ASV2306, ASV2307, ASV2308, ASV2309, ASV2310, ASV2311, ASV2312, ASV2313, ASV2314, ASV2315, ASV2316, ASV2317, ASV2318, ASV2319, ASV2320, ASV2321, ASV2322, ASV2323, ASV2324, ASV2325, ASV2326, ASV2327, ASV2328, ASV2329, ASV2330, ASV2331, ASV2332, ASV2333, ASV2334, ASV2335, ASV2336, ASV2337, ASV2338, ASV2339, ASV2340, ASV2341, ASV2342, ASV2343, ASV2344, ASV2345, ASV2346, ASV2347, ASV2348, ASV2349, ASV2350, ASV2351, ASV2352, ASV2353, ASV2354, ASV2355, ASV2356, ASV2357, ASV2358, ASV2359, ASV2360, ASV2361, ASV2362, ASV2363, ASV2364, ASV2365, ASV2366, ASV2367, ASV2368, ASV2369, ASV2370, ASV2371, ASV2372, ASV2373, ASV2374, ASV2375, ASV2376, ASV2377, ASV2378, ASV2379, ASV2380, ASV2381, ASV2382, ASV2383, ASV2384, ASV2385, ASV2386, ASV2387, ASV2388, ASV2389, ASV2390, ASV2391, ASV2392, ASV2393, ASV2394, ASV2395, ASV2396, ASV2397, ASV2398, ASV2399, ASV2400, ASV2401, ASV2402, ASV2403, ASV2404, ASV2405, ASV2406, ASV2407, ASV2408, ASV2409, ASV2410, ASV2411, ASV2412, ASV2413, ASV2414, ASV2415, ASV2416, ASV2417, ASV2418, ASV2419, ASV2420, ASV2421, ASV2422, ASV2423, ASV2424, ASV2425, ASV2426, ASV2427, ASV2428, ASV2429, ASV2430, ASV2431, ASV2432, ASV2433, ASV2434, ASV2435, ASV2436, ASV2437, ASV2438, ASV2439, ASV2440, ASV2441, ASV2442, ASV2443, ASV2444, ASV2445, ASV2446, ASV2447, ASV2448, ASV2449, ASV2450, ASV2451, ASV2452, ASV2453, ASV2454, ASV2455, ASV2456, ASV2457, ASV2458, ASV2459, ASV2460, ASV2461, ASV2462, ASV2463, ASV2464, ASV2465, ASV2466, ASV2467, ASV2468, ASV2469, ASV2470, ASV2471, ASV2472, ASV2473, ASV2474, ASV2475, ASV2476, ASV2477, ASV2478, ASV2479, ASV2480, ASV2481, ASV2482, ASV2483, ASV2484, ASV2485, ASV2486, ASV2487, ASV2488, ASV2489, ASV2490, ASV2491, ASV2492, ASV2493, ASV2494, ASV2495, ASV2496, ASV2497, ASV2498, ASV2499, ASV2500, ASV2501, ASV2502, ASV2503, ASV2504, ASV2505, ASV2506, ASV2507, ASV2508, ASV2509, ASV2510, ASV2511, ASV2512, ASV2513, ASV2514, ASV2515, ASV2516, ASV2517, ASV2518, ASV2519, ASV2520, ASV2521, ASV2522, ASV2523, ASV2524, ASV2525, ASV2526, ASV2527, ASV2528, ASV2529, ASV2530, ASV2531, ASV2532, ASV2533, ASV2534, ASV2535, ASV2536, ASV2537, ASV2538, ASV2539, ASV2540, ASV2541, ASV2542, ASV2543, ASV2544, ASV2545, ASV2546, ASV2547, ASV2548, ASV2549, ASV2550, ASV2551, ASV2552, ASV2553, ASV2554, ASV2555, ASV2556, ASV2557, ASV2558, ASV2559, ASV2560
2024-03-06 21:58:49.180546 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:58:49.180809 INFO::Filtered feature names from variance filtering:
2024-03-06 21:58:49.181028 INFO::Running selected normalization method: TSS
2024-03-06 21:58:49.181574 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:58:49.183903 INFO::Running selected transform method: LOG
2024-03-06 21:58:49.184518 INFO::Running selected analysis method: LM
2024-03-06 21:58:49.184799 INFO::Fitting model to feature number 1, ASV2
2024-03-06 21:58:49.185843 INFO::Fitting model to feature number 2, ASV3
2024-03-06 21:58:49.186798 INFO::Fitting model to feature number 3, ASV4
2024-03-06 21:58:49.187977 INFO::Fitting model to feature number 4, ASV6
2024-03-06 21:58:49.191878 INFO::Fitting model to feature number 5, ASV7
2024-03-06 21:58:49.193387 INFO::Fitting model to feature number 6, ASV9
2024-03-06 21:58:49.194218 INFO::Fitting model to feature number 7, ASV10
2024-03-06 21:58:49.195055 INFO::Fitting model to feature number 8, ASV11
2024-03-06 21:58:49.195886 INFO::Fitting model to feature number 9, ASV12
2024-03-06 21:58:49.196694 INFO::Fitting model to feature number 10, ASV14
2024-03-06 21:58:49.197506 INFO::Fitting model to feature number 11, ASV15
2024-03-06 21:58:49.198355 INFO::Fitting model to feature number 12, ASV17
2024-03-06 21:58:49.199173 INFO::Fitting model to feature number 13, ASV18
2024-03-06 21:58:49.199976 INFO::Fitting model to feature number 14, ASV21
2024-03-06 21:58:49.200777 INFO::Fitting model to feature number 15, ASV23
2024-03-06 21:58:49.201594 INFO::Fitting model to feature number 16, ASV24
2024-03-06 21:58:49.202407 INFO::Fitting model to feature number 17, ASV25
2024-03-06 21:58:49.203227 INFO::Fitting model to feature number 18, ASV27
2024-03-06 21:58:49.20405 INFO::Fitting model to feature number 19, ASV28
2024-03-06 21:58:49.204853 INFO::Fitting model to feature number 20, ASV29
2024-03-06 21:58:49.205664 INFO::Fitting model to feature number 21, ASV31
2024-03-06 21:58:49.206473 INFO::Fitting model to feature number 22, ASV32
2024-03-06 21:58:49.207301 INFO::Fitting model to feature number 23, ASV34
2024-03-06 21:58:49.208146 INFO::Fitting model to feature number 24, ASV35
2024-03-06 21:58:49.208956 INFO::Fitting model to feature number 25, ASV36
2024-03-06 21:58:49.209767 INFO::Fitting model to feature number 26, ASV39
2024-03-06 21:58:49.210573 INFO::Fitting model to feature number 27, ASV40
2024-03-06 21:58:49.211374 INFO::Fitting model to feature number 28, ASV45
2024-03-06 21:58:49.212195 INFO::Fitting model to feature number 29, ASV46
2024-03-06 21:58:49.213004 INFO::Fitting model to feature number 30, ASV47
2024-03-06 21:58:49.213806 INFO::Fitting model to feature number 31, ASV48
2024-03-06 21:58:49.214609 INFO::Fitting model to feature number 32, ASV49
2024-03-06 21:58:49.215412 INFO::Fitting model to feature number 33, ASV54
2024-03-06 21:58:49.216246 INFO::Fitting model to feature number 34, ASV57
2024-03-06 21:58:49.217131 INFO::Fitting model to feature number 35, ASV62
2024-03-06 21:58:49.21798 INFO::Fitting model to feature number 36, ASV64
2024-03-06 21:58:49.218799 INFO::Fitting model to feature number 37, ASV66
2024-03-06 21:58:49.219618 INFO::Fitting model to feature number 38, ASV70
2024-03-06 21:58:49.220414 INFO::Fitting model to feature number 39, ASV73
2024-03-06 21:58:49.221222 INFO::Fitting model to feature number 40, ASV74
2024-03-06 21:58:49.222037 INFO::Fitting model to feature number 41, ASV76
2024-03-06 21:58:49.222845 INFO::Fitting model to feature number 42, ASV83
2024-03-06 21:58:49.223655 INFO::Fitting model to feature number 43, ASV91
2024-03-06 21:58:49.224498 INFO::Fitting model to feature number 44, ASV92
2024-03-06 21:58:49.225311 INFO::Fitting model to feature number 45, ASV97
2024-03-06 21:58:49.226133 INFO::Fitting model to feature number 46, ASV102
2024-03-06 21:58:49.226968 INFO::Fitting model to feature number 47, ASV104
2024-03-06 21:58:49.227778 INFO::Fitting model to feature number 48, ASV116
2024-03-06 21:58:49.228584 INFO::Fitting model to feature number 49, ASV144
2024-03-06 21:58:49.232646 INFO::Counting total values for each feature
2024-03-06 21:58:49.233851 INFO::Writing filtered data to file results/maaslin2/asv_GHAT_Length/features/filtered_data.tsv
2024-03-06 21:58:49.235006 INFO::Writing filtered, normalized data to file results/maaslin2/asv_GHAT_Length/features/filtered_data_norm.tsv
2024-03-06 21:58:49.236038 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/asv_GHAT_Length/features/filtered_data_norm_transformed.tsv
2024-03-06 21:58:49.237279 WARNING::Deleting existing residuals file: results/maaslin2/asv_GHAT_Length/fits/residuals.rds
2024-03-06 21:58:49.237865 INFO::Writing residuals to file results/maaslin2/asv_GHAT_Length/fits/residuals.rds
2024-03-06 21:58:49.238562 WARNING::Deleting existing fitted file: results/maaslin2/asv_GHAT_Length/fits/fitted.rds
2024-03-06 21:58:49.238945 INFO::Writing fitted values to file results/maaslin2/asv_GHAT_Length/fits/fitted.rds
2024-03-06 21:58:49.239621 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/asv_GHAT_Length/all_results.tsv
2024-03-06 21:58:49.24052 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/asv_GHAT_Length/significant_results.tsv
2024-03-06 21:58:49.240905 INFO::Writing heatmap of significant results to file: results/maaslin2/asv_GHAT_Length/heatmap.pdf
[1] "There are no associations to plot!"
2024-03-06 21:58:49.241378 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/asv_GHAT_Length
[1] "There are no associations to plot!"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_trans) <span class="sc">%&gt;%</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ASV"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ASV <span class="sc">%in%</span> <span class="fu">filter</span>(asv_GHAT<span class="sc">$</span>results, qval <span class="sc">&lt;</span><span class="fl">0.2</span>)<span class="sc">$</span>feature)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] ASV     Kingdom Phylum  Class   Order   Family  Genus   Species
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>ps_trans_genus <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_trans, <span class="at">taxrank =</span> <span class="st">"Genus"</span>)</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>genus_GHAT <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_trans_genus),</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_trans_genus), <span class="st">"data.frame"</span>),</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/genus_GHAT_Length"</span>,</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">5</span>,</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.1</span>,</span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"GHAT_Length"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:58:53.221919 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/genus_GHAT_Length/maaslin2.log"
2024-03-06 21:58:53.415442 INFO::Writing function arguments to log file
2024-03-06 21:58:53.418391 INFO::Verifying options selected are valid
2024-03-06 21:58:53.418727 INFO::Determining format of input files
2024-03-06 21:58:53.418996 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:58:53.421023 INFO::Formula for fixed effects: expr ~  GHAT_Length
2024-03-06 21:58:53.421391 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:58:53.421665 INFO::Total samples in data: 25
2024-03-06 21:58:53.421889 INFO::Min samples required with min abundance for a feature not to be filtered: 2.500000
2024-03-06 21:58:53.42356 INFO::Total filtered features: 310
2024-03-06 21:58:53.424095 INFO::Filtered feature names from abundance and prevalence filtering: ASV5, ASV11, ASV12, ASV13, ASV21, ASV22, ASV23, ASV30, ASV34, ASV38, ASV47, ASV48, ASV51, ASV53, ASV57, ASV59, ASV63, ASV69, ASV71, ASV73, ASV75, ASV88, ASV97, ASV104, ASV115, ASV119, ASV125, ASV131, ASV136, ASV140, ASV147, ASV148, ASV151, ASV152, ASV154, ASV155, ASV174, ASV177, ASV183, ASV191, ASV196, ASV197, ASV200, ASV207, ASV209, ASV215, ASV218, ASV224, ASV228, ASV249, ASV254, ASV255, ASV256, ASV259, ASV264, ASV266, ASV267, ASV276, ASV277, ASV284, ASV289, ASV291, ASV299, ASV306, ASV307, ASV309, ASV312, ASV319, ASV325, ASV331, ASV347, ASV349, ASV353, ASV356, ASV358, ASV362, ASV363, ASV367, ASV371, ASV375, ASV378, ASV390, ASV392, ASV395, ASV407, ASV410, ASV417, ASV421, ASV431, ASV440, ASV443, ASV459, ASV466, ASV468, ASV470, ASV473, ASV475, ASV483, ASV494, ASV518, ASV526, ASV531, ASV557, ASV558, ASV570, ASV571, ASV573, ASV586, ASV596, ASV601, ASV605, ASV608, ASV612, ASV618, ASV626, ASV637, ASV641, ASV642, ASV650, ASV653, ASV660, ASV667, ASV671, ASV674, ASV685, ASV686, ASV690, ASV693, ASV699, ASV703, ASV709, ASV711, ASV728, ASV734, ASV740, ASV751, ASV760, ASV764, ASV785, ASV791, ASV793, ASV803, ASV808, ASV821, ASV824, ASV829, ASV840, ASV847, ASV849, ASV850, ASV853, ASV855, ASV860, ASV863, ASV881, ASV891, ASV903, ASV913, ASV915, ASV918, ASV925, ASV927, ASV936, ASV948, ASV956, ASV967, ASV979, ASV980, ASV984, ASV1000, ASV1004, ASV1009, ASV1011, ASV1038, ASV1039, ASV1042, ASV1050, ASV1055, ASV1068, ASV1069, ASV1070, ASV1083, ASV1087, ASV1088, ASV1089, ASV1103, ASV1109, ASV1111, ASV1115, ASV1117, ASV1118, ASV1121, ASV1123, ASV1132, ASV1134, ASV1136, ASV1144, ASV1145, ASV1172, ASV1176, ASV1194, ASV1196, ASV1197, ASV1200, ASV1201, ASV1203, ASV1221, ASV1222, ASV1235, ASV1236, ASV1238, ASV1242, ASV1249, ASV1251, ASV1253, ASV1255, ASV1260, ASV1275, ASV1282, ASV1284, ASV1285, ASV1291, ASV1304, ASV1312, ASV1313, ASV1314, ASV1315, ASV1316, ASV1317, ASV1321, ASV1322, ASV1344, ASV1345, ASV1361, ASV1367, ASV1370, ASV1382, ASV1383, ASV1414, ASV1438, ASV1439, ASV1440, ASV1489, ASV1499, ASV1501, ASV1503, ASV1517, ASV1520, ASV1534, ASV1535, ASV1536, ASV1537, ASV1539, ASV1550, ASV1560, ASV1565, ASV1601, ASV1602, ASV1615, ASV1617, ASV1624, ASV1627, ASV1647, ASV1669, ASV1672, ASV1724, ASV1732, ASV1734, ASV1767, ASV1769, ASV1774, ASV1778, ASV1783, ASV1812, ASV1857, ASV1859, ASV1862, ASV1869, ASV1887, ASV1926, ASV1935, ASV1966, ASV1980, ASV2008, ASV2029, ASV2033, ASV2047, ASV2055, ASV2071, ASV2087, ASV2159, ASV2214, ASV2267, ASV2269, ASV2301, ASV2321, ASV2322, ASV2323, ASV2328, ASV2329, ASV2331, ASV2358, ASV2387, ASV2395, ASV2396, ASV2414, ASV2421, ASV2433, ASV2480, ASV2559
2024-03-06 21:58:53.424658 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:58:53.42496 INFO::Filtered feature names from variance filtering:
2024-03-06 21:58:53.425193 INFO::Running selected normalization method: TSS
2024-03-06 21:58:53.425614 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:58:53.428095 INFO::Running selected transform method: LOG
2024-03-06 21:58:53.428585 INFO::Running selected analysis method: LM
2024-03-06 21:58:53.428866 INFO::Fitting model to feature number 1, ASV2
2024-03-06 21:58:53.429941 INFO::Fitting model to feature number 2, ASV3
2024-03-06 21:58:53.431029 INFO::Fitting model to feature number 3, ASV4
2024-03-06 21:58:53.432243 INFO::Fitting model to feature number 4, ASV6
2024-03-06 21:58:53.433324 INFO::Fitting model to feature number 5, ASV17
2024-03-06 21:58:53.434287 INFO::Fitting model to feature number 6, ASV20
2024-03-06 21:58:53.435264 INFO::Fitting model to feature number 7, ASV24
2024-03-06 21:58:53.436332 INFO::Fitting model to feature number 8, ASV31
2024-03-06 21:58:53.43726 INFO::Fitting model to feature number 9, ASV70
2024-03-06 21:58:53.438296 INFO::Fitting model to feature number 10, ASV105
2024-03-06 21:58:53.441573 INFO::Counting total values for each feature
2024-03-06 21:58:53.442429 INFO::Writing filtered data to file results/maaslin2/genus_GHAT_Length/features/filtered_data.tsv
2024-03-06 21:58:53.443179 INFO::Writing filtered, normalized data to file results/maaslin2/genus_GHAT_Length/features/filtered_data_norm.tsv
2024-03-06 21:58:53.443974 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/genus_GHAT_Length/features/filtered_data_norm_transformed.tsv
2024-03-06 21:58:53.44481 WARNING::Deleting existing residuals file: results/maaslin2/genus_GHAT_Length/fits/residuals.rds
2024-03-06 21:58:53.445329 INFO::Writing residuals to file results/maaslin2/genus_GHAT_Length/fits/residuals.rds
2024-03-06 21:58:53.445784 WARNING::Deleting existing fitted file: results/maaslin2/genus_GHAT_Length/fits/fitted.rds
2024-03-06 21:58:53.446358 INFO::Writing fitted values to file results/maaslin2/genus_GHAT_Length/fits/fitted.rds
2024-03-06 21:58:53.446836 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/genus_GHAT_Length/all_results.tsv
2024-03-06 21:58:53.447469 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/genus_GHAT_Length/significant_results.tsv
2024-03-06 21:58:53.447871 INFO::Writing heatmap of significant results to file: results/maaslin2/genus_GHAT_Length/heatmap.pdf
[1] "There is not enough metadata in the associations to create a heatmap plot. Please review the associations in text output file."
2024-03-06 21:58:53.448424 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/genus_GHAT_Length
2024-03-06 21:58:53.448977 INFO::Plotting associations from most to least significant, grouped by metadata
2024-03-06 21:58:53.449268 INFO::Plotting data for metadata number 1, GHAT_Length
2024-03-06 21:58:53.450232 INFO::Creating scatter plot for continuous data, GHAT_Length vs ASV17</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:58:53.521648 INFO::Creating scatter plot for continuous data, GHAT_Length vs ASV3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values (`geom_point()`).
Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values (`geom_point()`).</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_trans_genus) <span class="sc">%&gt;%</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ASV"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ASV <span class="sc">%in%</span> <span class="fu">filter</span>(genus_GHAT<span class="sc">$</span>results, qval <span class="sc">&lt;</span> <span class="fl">0.2</span>)<span class="sc">$</span>feature)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>    ASV  Kingdom       Phylum         Class                          Order
1  ASV3 Bacteria Bacteroidota   Bacteroidia                  Bacteroidales
2 ASV17 Bacteria   Firmicutes Negativicutes Veillonellales-Selenomonadales
           Family      Genus Species
1  Prevotellaceae Prevotella    &lt;NA&gt;
2 Veillonellaceae  Dialister    &lt;NA&gt;</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot correlation for these 2 genera</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>psmelt_trans_genus <span class="ot">=</span> <span class="fu">psmelt</span>(ps_trans_genus)</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>maaslin_res_genus_ghat <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">filter</span>(psmelt_trans_genus, OTU <span class="sc">%in%</span> <span class="fu">filter</span>(genus_GHAT<span class="sc">$</span>results, qval <span class="sc">&lt;</span> <span class="fl">0.2</span>)<span class="sc">$</span>feature),</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">aes</span>(<span class="at">x =</span> GHAT_Length, <span class="at">y =</span> Abundance)) <span class="sc">+</span></span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Genus, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="cf">function</span>(x){<span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fl">0.1</span>, x))}) <span class="sc">+</span></span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Length of Testosterone Therapy"</span>,</span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Genus Abundance (%)"</span>,</span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Maaslin2 results"</span>)</span>
<span id="cb67-13"><a href="#cb67-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-14"><a href="#cb67-14" aria-hidden="true" tabindex="-1"></a>maaslin_res_genus_ghat</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 29 rows containing missing values (`geom_smooth()`).</code></pre>
</div>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/trans%20testosterone-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(maaslin_res_genus_ghat, <span class="at">filename =</span> <span class="st">"results/maaslin2/genus_GHAT_Length_panelplot.pdf"</span>,</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">4</span>, <span class="at">width =</span><span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 29 rows containing missing values (`geom_smooth()`).</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Alpha diversity by duration of gaht</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>ps_trans_abs <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_samples, grp <span class="sc">==</span><span class="st">"03_trans-men"</span>)</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>ps_trans_genus <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_trans_abs, <span class="at">taxrank =</span><span class="st">"Genus"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x) <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>adiv_trans <span class="ot">=</span> <span class="fu">plot_richness</span>(ps_trans_abs, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">"Observed"</span>, <span class="st">"Shannon"</span>, <span class="st">"InvSimpson"</span>))</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>adiv_trans<span class="sc">$</span>data <span class="sc">%&gt;%</span></span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(GHAT_Length, variable, value) <span class="sc">%&gt;%</span></span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cor_test</span>(value, GHAT_Length) <span class="sc">%&gt;%</span></span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 14%">
<col style="width: 7%">
<col style="width: 15%">
<col style="width: 7%">
<col style="width: 12%">
<col style="width: 7%">
<col style="width: 11%">
<col style="width: 12%">
<col style="width: 10%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">variable</th>
<th style="text-align: left;">var1</th>
<th style="text-align: left;">var2</th>
<th style="text-align: right;">cor</th>
<th style="text-align: right;">statistic</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">conf.low</th>
<th style="text-align: right;">conf.high</th>
<th style="text-align: left;">method</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">GHAT_Length</td>
<td style="text-align: right;">-0.15</td>
<td style="text-align: right;">-0.705</td>
<td style="text-align: right;">0.488</td>
<td style="text-align: right;">-0.511</td>
<td style="text-align: right;">0.265</td>
<td style="text-align: left;">Pearson</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">GHAT_Length</td>
<td style="text-align: right;">-0.40</td>
<td style="text-align: right;">-2.102</td>
<td style="text-align: right;">0.047</td>
<td style="text-align: right;">-0.688</td>
<td style="text-align: right;">-0.007</td>
<td style="text-align: left;">Pearson</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">GHAT_Length</td>
<td style="text-align: right;">-0.17</td>
<td style="text-align: right;">-0.844</td>
<td style="text-align: right;">0.407</td>
<td style="text-align: right;">-0.532</td>
<td style="text-align: right;">0.238</td>
<td style="text-align: left;">Pearson</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>adiv_trans<span class="sc">$</span>data <span class="sc">%&gt;%</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> GHAT_Length, <span class="at">y =</span> value)) <span class="sc">+</span> </span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span><span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>variable, <span class="at">scales =</span> <span class="st">"free_y"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/trans%20testosterone-2.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>BC_trans_genus <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(<span class="fu">otu_table</span>(ps_trans_genus))</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>ord.bc <span class="ot">=</span> <span class="fu">ordinate</span>(ps_trans_genus,<span class="at">method =</span> <span class="st">"PCoA"</span>)</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>gaht_adonis_bc <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(BC_trans_genus <span class="sc">~</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_trans_genus)<span class="sc">$</span>GHAT_Length)</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a><span class="co">#This is highly significant on the genus level!</span></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(gaht_adonis_bc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 56%">
<col style="width: 3%">
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 10%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Df</th>
<th style="text-align: right;">SumOfSqs</th>
<th style="text-align: right;">R2</th>
<th style="text-align: right;">F</th>
<th style="text-align: right;">Pr(&gt;F)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">phyloseq::sample_data(ps_trans_genus)$GHAT_Length</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.6500245</td>
<td style="text-align: right;">0.1102164</td>
<td style="text-align: right;">2.848982</td>
<td style="text-align: right;">0.014</td>
</tr>
<tr class="even">
<td style="text-align: left;">Residual</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">5.2476854</td>
<td style="text-align: right;">0.8897836</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Total</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">5.8977098</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Summary statistics of duration menopause</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(ps_trans_genus)<span class="sc">$</span>GHAT_Length <span class="sc">%&gt;%</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  10.00   16.00   26.00   33.56   38.00  146.00 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>p_gaht_beta <span class="ot">=</span> <span class="fu">plot_ordination</span>(ps_trans_genus, ord.bc, <span class="at">type=</span><span class="st">"samples"</span>, <span class="at">color=</span><span class="st">"GHAT_Length"</span>) <span class="sc">+</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">"Length of GAHT"</span>, <span class="at">caption =</span> <span class="st">"Bray-Curtis"</span>) <span class="sc">+</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>))</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>p_gaht_beta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/trans%20testosterone-3.pdf" class="img-fluid"></p>
</div>
</div>
<p>There is a significant effect of duration of GAHT on the shannon index value in the trans group and on overall community composition although this seems to depend on a single outlier with a duration of 150 of hormone therapy. On the genus level, there are 2 taxa (Dialister and Prevotella) who are inversely correlated to duration of testosterone supplementation.</p>
</section>
<section id="absolute-height-of-testosterone-in-the-trans-men-group" class="level3">
<h3 class="anchored" data-anchor-id="absolute-height-of-testosterone-in-the-trans-men-group">Absolute Height of Testosterone in the trans-men group</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Height of testosterone</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>genus_testosterone <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_trans_genus),</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_trans_genus), <span class="st">"data.frame"</span>),</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/genus_Testosterone"</span>,</span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">5</span>,</span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.3</span>,</span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"Testosterone"</span>) <span class="co"># no significant effect</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:58:58.512654 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/genus_Testosterone/maaslin2.log"
2024-03-06 21:58:58.691886 INFO::Writing function arguments to log file
2024-03-06 21:58:58.693938 INFO::Verifying options selected are valid
2024-03-06 21:58:58.69418 INFO::Determining format of input files
2024-03-06 21:58:58.694402 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:58:58.696169 INFO::Formula for fixed effects: expr ~  Testosterone
2024-03-06 21:58:58.696431 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:58:58.696652 INFO::Total samples in data: 25
2024-03-06 21:58:58.696857 INFO::Min samples required with min abundance for a feature not to be filtered: 7.500000
2024-03-06 21:58:58.698317 INFO::Total filtered features: 314
2024-03-06 21:58:58.698598 INFO::Filtered feature names from abundance and prevalence filtering: ASV5, ASV11, ASV12, ASV13, ASV20, ASV21, ASV22, ASV23, ASV24, ASV30, ASV34, ASV38, ASV47, ASV48, ASV51, ASV53, ASV57, ASV59, ASV63, ASV66, ASV69, ASV70, ASV71, ASV73, ASV75, ASV88, ASV97, ASV104, ASV115, ASV125, ASV131, ASV136, ASV140, ASV143, ASV147, ASV148, ASV151, ASV152, ASV154, ASV155, ASV170, ASV174, ASV177, ASV183, ASV191, ASV196, ASV197, ASV200, ASV201, ASV202, ASV207, ASV209, ASV215, ASV218, ASV224, ASV228, ASV249, ASV254, ASV255, ASV256, ASV259, ASV264, ASV267, ASV276, ASV277, ASV284, ASV289, ASV291, ASV299, ASV306, ASV307, ASV312, ASV319, ASV325, ASV331, ASV337, ASV347, ASV349, ASV353, ASV356, ASV358, ASV362, ASV363, ASV367, ASV371, ASV375, ASV378, ASV390, ASV392, ASV395, ASV407, ASV410, ASV417, ASV421, ASV431, ASV440, ASV443, ASV455, ASV459, ASV466, ASV470, ASV473, ASV475, ASV483, ASV494, ASV518, ASV526, ASV531, ASV557, ASV558, ASV570, ASV571, ASV573, ASV586, ASV596, ASV601, ASV605, ASV608, ASV612, ASV618, ASV626, ASV628, ASV637, ASV641, ASV650, ASV653, ASV660, ASV671, ASV674, ASV685, ASV686, ASV690, ASV693, ASV699, ASV703, ASV709, ASV711, ASV728, ASV734, ASV740, ASV744, ASV751, ASV760, ASV764, ASV785, ASV791, ASV793, ASV803, ASV808, ASV821, ASV824, ASV829, ASV838, ASV840, ASV847, ASV849, ASV850, ASV853, ASV855, ASV860, ASV863, ASV881, ASV891, ASV913, ASV918, ASV925, ASV927, ASV936, ASV948, ASV956, ASV967, ASV979, ASV980, ASV984, ASV995, ASV1000, ASV1004, ASV1009, ASV1011, ASV1038, ASV1039, ASV1040, ASV1042, ASV1044, ASV1050, ASV1055, ASV1068, ASV1069, ASV1070, ASV1083, ASV1087, ASV1088, ASV1089, ASV1103, ASV1109, ASV1111, ASV1115, ASV1117, ASV1118, ASV1121, ASV1123, ASV1132, ASV1134, ASV1136, ASV1144, ASV1145, ASV1172, ASV1176, ASV1194, ASV1196, ASV1197, ASV1200, ASV1201, ASV1203, ASV1221, ASV1222, ASV1235, ASV1236, ASV1238, ASV1242, ASV1249, ASV1251, ASV1253, ASV1255, ASV1259, ASV1260, ASV1275, ASV1282, ASV1284, ASV1285, ASV1291, ASV1304, ASV1312, ASV1316, ASV1317, ASV1321, ASV1322, ASV1331, ASV1344, ASV1345, ASV1361, ASV1367, ASV1382, ASV1383, ASV1414, ASV1438, ASV1439, ASV1440, ASV1489, ASV1499, ASV1501, ASV1503, ASV1517, ASV1520, ASV1534, ASV1535, ASV1536, ASV1537, ASV1539, ASV1550, ASV1565, ASV1601, ASV1602, ASV1615, ASV1617, ASV1624, ASV1627, ASV1647, ASV1669, ASV1672, ASV1723, ASV1724, ASV1732, ASV1734, ASV1767, ASV1769, ASV1774, ASV1778, ASV1783, ASV1812, ASV1857, ASV1859, ASV1862, ASV1869, ASV1887, ASV1926, ASV1935, ASV1980, ASV2008, ASV2029, ASV2033, ASV2047, ASV2055, ASV2071, ASV2087, ASV2159, ASV2214, ASV2267, ASV2269, ASV2301, ASV2322, ASV2323, ASV2328, ASV2329, ASV2331, ASV2358, ASV2365, ASV2387, ASV2395, ASV2396, ASV2414, ASV2433, ASV2480, ASV2559
2024-03-06 21:58:58.699016 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:58:58.699243 INFO::Filtered feature names from variance filtering:
2024-03-06 21:58:58.699449 INFO::Running selected normalization method: TSS
2024-03-06 21:58:58.699801 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:58:58.702043 INFO::Running selected transform method: LOG
2024-03-06 21:58:58.702403 INFO::Running selected analysis method: LM
2024-03-06 21:58:58.702674 INFO::Fitting model to feature number 1, ASV2
2024-03-06 21:58:58.703877 INFO::Fitting model to feature number 2, ASV3
2024-03-06 21:58:58.704772 INFO::Fitting model to feature number 3, ASV4
2024-03-06 21:58:58.705658 INFO::Fitting model to feature number 4, ASV6
2024-03-06 21:58:58.70652 INFO::Fitting model to feature number 5, ASV17
2024-03-06 21:58:58.707385 INFO::Fitting model to feature number 6, ASV31
2024-03-06 21:58:58.709921 INFO::Counting total values for each feature
2024-03-06 21:58:58.710535 INFO::Writing filtered data to file results/maaslin2/genus_Testosterone/features/filtered_data.tsv
2024-03-06 21:58:58.711187 INFO::Writing filtered, normalized data to file results/maaslin2/genus_Testosterone/features/filtered_data_norm.tsv
2024-03-06 21:58:58.711722 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/genus_Testosterone/features/filtered_data_norm_transformed.tsv
2024-03-06 21:58:58.712286 WARNING::Deleting existing residuals file: results/maaslin2/genus_Testosterone/fits/residuals.rds
2024-03-06 21:58:58.712737 INFO::Writing residuals to file results/maaslin2/genus_Testosterone/fits/residuals.rds
2024-03-06 21:58:58.713169 WARNING::Deleting existing fitted file: results/maaslin2/genus_Testosterone/fits/fitted.rds
2024-03-06 21:58:58.713523 INFO::Writing fitted values to file results/maaslin2/genus_Testosterone/fits/fitted.rds
2024-03-06 21:58:58.713901 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/genus_Testosterone/all_results.tsv
2024-03-06 21:58:58.714454 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/genus_Testosterone/significant_results.tsv
2024-03-06 21:58:58.714813 INFO::Writing heatmap of significant results to file: results/maaslin2/genus_Testosterone/heatmap.pdf
[1] "There are no associations to plot!"
2024-03-06 21:58:58.715266 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/genus_Testosterone
[1] "There are no associations to plot!"</code></pre>
</div>
</div>
<p>No significant associations.</p>
</section>
<section id="duration-of-gnrh-therapy-in-trans-men-group" class="level3">
<h3 class="anchored" data-anchor-id="duration-of-gnrh-therapy-in-trans-men-group">Duration of GnRH therapy in trans-men group</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">sample_data</span>(ps_trans_genus)<span class="sc">$</span>Duration_GNRH) <span class="co"># only 10 values!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
   12.0    18.0    36.0    37.6    56.0    74.0      15 </code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>genus_gnrh<span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_trans_genus),</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_trans_genus), <span class="st">"data.frame"</span>),</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/genus_DurationGNRH"</span>,</span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">5</span>,</span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.3</span>,</span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"Duration_GNRH"</span>) <span class="co"># no significant effect</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:58:58.720945 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/genus_DurationGNRH/maaslin2.log"
2024-03-06 21:58:58.89665 INFO::Writing function arguments to log file
2024-03-06 21:58:58.899383 INFO::Verifying options selected are valid
2024-03-06 21:58:58.899693 INFO::Determining format of input files
2024-03-06 21:58:58.899945 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:58:58.901896 INFO::Formula for fixed effects: expr ~  Duration_GNRH
2024-03-06 21:58:58.90222 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:58:58.902454 INFO::Total samples in data: 25
2024-03-06 21:58:58.902665 INFO::Min samples required with min abundance for a feature not to be filtered: 7.500000
2024-03-06 21:58:58.904331 INFO::Total filtered features: 314
2024-03-06 21:58:58.904859 INFO::Filtered feature names from abundance and prevalence filtering: ASV5, ASV11, ASV12, ASV13, ASV20, ASV21, ASV22, ASV23, ASV24, ASV30, ASV34, ASV38, ASV47, ASV48, ASV51, ASV53, ASV57, ASV59, ASV63, ASV66, ASV69, ASV70, ASV71, ASV73, ASV75, ASV88, ASV97, ASV104, ASV115, ASV125, ASV131, ASV136, ASV140, ASV143, ASV147, ASV148, ASV151, ASV152, ASV154, ASV155, ASV170, ASV174, ASV177, ASV183, ASV191, ASV196, ASV197, ASV200, ASV201, ASV202, ASV207, ASV209, ASV215, ASV218, ASV224, ASV228, ASV249, ASV254, ASV255, ASV256, ASV259, ASV264, ASV267, ASV276, ASV277, ASV284, ASV289, ASV291, ASV299, ASV306, ASV307, ASV312, ASV319, ASV325, ASV331, ASV337, ASV347, ASV349, ASV353, ASV356, ASV358, ASV362, ASV363, ASV367, ASV371, ASV375, ASV378, ASV390, ASV392, ASV395, ASV407, ASV410, ASV417, ASV421, ASV431, ASV440, ASV443, ASV455, ASV459, ASV466, ASV470, ASV473, ASV475, ASV483, ASV494, ASV518, ASV526, ASV531, ASV557, ASV558, ASV570, ASV571, ASV573, ASV586, ASV596, ASV601, ASV605, ASV608, ASV612, ASV618, ASV626, ASV628, ASV637, ASV641, ASV650, ASV653, ASV660, ASV671, ASV674, ASV685, ASV686, ASV690, ASV693, ASV699, ASV703, ASV709, ASV711, ASV728, ASV734, ASV740, ASV744, ASV751, ASV760, ASV764, ASV785, ASV791, ASV793, ASV803, ASV808, ASV821, ASV824, ASV829, ASV838, ASV840, ASV847, ASV849, ASV850, ASV853, ASV855, ASV860, ASV863, ASV881, ASV891, ASV913, ASV918, ASV925, ASV927, ASV936, ASV948, ASV956, ASV967, ASV979, ASV980, ASV984, ASV995, ASV1000, ASV1004, ASV1009, ASV1011, ASV1038, ASV1039, ASV1040, ASV1042, ASV1044, ASV1050, ASV1055, ASV1068, ASV1069, ASV1070, ASV1083, ASV1087, ASV1088, ASV1089, ASV1103, ASV1109, ASV1111, ASV1115, ASV1117, ASV1118, ASV1121, ASV1123, ASV1132, ASV1134, ASV1136, ASV1144, ASV1145, ASV1172, ASV1176, ASV1194, ASV1196, ASV1197, ASV1200, ASV1201, ASV1203, ASV1221, ASV1222, ASV1235, ASV1236, ASV1238, ASV1242, ASV1249, ASV1251, ASV1253, ASV1255, ASV1259, ASV1260, ASV1275, ASV1282, ASV1284, ASV1285, ASV1291, ASV1304, ASV1312, ASV1316, ASV1317, ASV1321, ASV1322, ASV1331, ASV1344, ASV1345, ASV1361, ASV1367, ASV1382, ASV1383, ASV1414, ASV1438, ASV1439, ASV1440, ASV1489, ASV1499, ASV1501, ASV1503, ASV1517, ASV1520, ASV1534, ASV1535, ASV1536, ASV1537, ASV1539, ASV1550, ASV1565, ASV1601, ASV1602, ASV1615, ASV1617, ASV1624, ASV1627, ASV1647, ASV1669, ASV1672, ASV1723, ASV1724, ASV1732, ASV1734, ASV1767, ASV1769, ASV1774, ASV1778, ASV1783, ASV1812, ASV1857, ASV1859, ASV1862, ASV1869, ASV1887, ASV1926, ASV1935, ASV1980, ASV2008, ASV2029, ASV2033, ASV2047, ASV2055, ASV2071, ASV2087, ASV2159, ASV2214, ASV2267, ASV2269, ASV2301, ASV2322, ASV2323, ASV2328, ASV2329, ASV2331, ASV2358, ASV2365, ASV2387, ASV2395, ASV2396, ASV2414, ASV2433, ASV2480, ASV2559
2024-03-06 21:58:58.905373 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:58:58.905639 INFO::Filtered feature names from variance filtering:
2024-03-06 21:58:58.905866 INFO::Running selected normalization method: TSS
2024-03-06 21:58:58.906252 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:58:58.908598 INFO::Running selected transform method: LOG
2024-03-06 21:58:58.908976 INFO::Running selected analysis method: LM
2024-03-06 21:58:58.909244 INFO::Fitting model to feature number 1, ASV2
2024-03-06 21:58:58.910339 INFO::Fitting model to feature number 2, ASV3
2024-03-06 21:58:58.911304 INFO::Fitting model to feature number 3, ASV4
2024-03-06 21:58:58.91234 INFO::Fitting model to feature number 4, ASV6
2024-03-06 21:58:58.913296 INFO::Fitting model to feature number 5, ASV17
2024-03-06 21:58:58.914193 INFO::Fitting model to feature number 6, ASV31
2024-03-06 21:58:58.916881 INFO::Counting total values for each feature
2024-03-06 21:58:58.917576 INFO::Writing filtered data to file results/maaslin2/genus_DurationGNRH/features/filtered_data.tsv
2024-03-06 21:58:58.918444 INFO::Writing filtered, normalized data to file results/maaslin2/genus_DurationGNRH/features/filtered_data_norm.tsv
2024-03-06 21:58:58.919073 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/genus_DurationGNRH/features/filtered_data_norm_transformed.tsv
2024-03-06 21:58:58.919667 WARNING::Deleting existing residuals file: results/maaslin2/genus_DurationGNRH/fits/residuals.rds
2024-03-06 21:58:58.920138 INFO::Writing residuals to file results/maaslin2/genus_DurationGNRH/fits/residuals.rds
2024-03-06 21:58:58.920542 WARNING::Deleting existing fitted file: results/maaslin2/genus_DurationGNRH/fits/fitted.rds
2024-03-06 21:58:58.921015 INFO::Writing fitted values to file results/maaslin2/genus_DurationGNRH/fits/fitted.rds
2024-03-06 21:58:58.921439 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/genus_DurationGNRH/all_results.tsv
2024-03-06 21:58:58.922041 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/genus_DurationGNRH/significant_results.tsv
2024-03-06 21:58:58.922416 INFO::Writing heatmap of significant results to file: results/maaslin2/genus_DurationGNRH/heatmap.pdf
[1] "There are no associations to plot!"
2024-03-06 21:58:58.922888 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/genus_DurationGNRH
[1] "There are no associations to plot!"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># no significant associations</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>No significant associations.</p>
</section>
<section id="cycle-dependency-in-pre-menopausal-group" class="level3">
<h3 class="anchored" data-anchor-id="cycle-dependency-in-pre-menopausal-group">Cycle dependency in pre-menopausal group</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>ps_pre <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_samples, grp <span class="sc">==</span><span class="st">"02_premenopausal"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x) <span class="sc">*</span> <span class="dv">100</span>) <span class="co"># relative abundances are necessary</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>asv_cycleday <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_pre),</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_pre), <span class="st">"data.frame"</span>),</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/asv_cycle_premeno"</span>,</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">3</span>,</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.1</span>,</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"CycleDaySampling"</span>) <span class="co"># no significant effect</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:58:58.955297 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/asv_cycle_premeno/maaslin2.log"
2024-03-06 21:59:00.357078 INFO::Writing function arguments to log file
2024-03-06 21:59:00.359641 INFO::Verifying options selected are valid
2024-03-06 21:59:00.359907 INFO::Determining format of input files
2024-03-06 21:59:00.360143 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:59:00.369977 INFO::Formula for fixed effects: expr ~  CycleDaySampling
2024-03-06 21:59:00.370365 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:59:00.370589 INFO::Total samples in data: 25
2024-03-06 21:59:00.370801 INFO::Min samples required with min abundance for a feature not to be filtered: 2.500000
2024-03-06 21:59:00.380483 INFO::Total filtered features: 2551
2024-03-06 21:59:00.381308 INFO::Filtered feature names from abundance and prevalence filtering: ASV3, ASV5, ASV6, ASV9, ASV10, ASV12, ASV13, ASV15, ASV17, ASV18, ASV19, ASV20, ASV21, ASV22, ASV23, ASV24, ASV25, ASV26, ASV27, ASV28, ASV29, ASV30, ASV31, ASV32, ASV33, ASV34, ASV35, ASV36, ASV37, ASV38, ASV39, ASV40, ASV41, ASV42, ASV43, ASV44, ASV45, ASV46, ASV47, ASV48, ASV49, ASV50, ASV51, ASV52, ASV53, ASV54, ASV55, ASV56, ASV57, ASV58, ASV59, ASV60, ASV61, ASV62, ASV63, ASV64, ASV65, ASV66, ASV67, ASV68, ASV69, ASV70, ASV71, ASV72, ASV73, ASV74, ASV75, ASV76, ASV77, ASV78, ASV80, ASV81, ASV82, ASV83, ASV84, ASV85, ASV86, ASV87, ASV88, ASV89, ASV90, ASV91, ASV92, ASV93, ASV94, ASV95, ASV96, ASV97, ASV98, ASV99, ASV100, ASV101, ASV102, ASV103, ASV104, ASV105, ASV106, ASV107, ASV108, ASV109, ASV110, ASV111, ASV112, ASV113, ASV114, ASV115, ASV116, ASV117, ASV118, ASV119, ASV120, ASV121, ASV122, ASV123, ASV124, ASV125, ASV126, ASV127, ASV128, ASV129, ASV130, ASV131, ASV132, ASV133, ASV134, ASV135, ASV136, ASV137, ASV138, ASV139, ASV140, ASV141, ASV142, ASV143, ASV144, ASV145, ASV146, ASV147, ASV148, ASV149, ASV150, ASV151, ASV152, ASV153, ASV154, ASV155, ASV156, ASV157, ASV158, ASV159, ASV160, ASV161, ASV162, ASV163, ASV164, ASV165, ASV166, ASV167, ASV168, ASV169, ASV170, ASV171, ASV172, ASV173, ASV174, ASV175, ASV176, ASV177, ASV178, ASV179, ASV180, ASV181, ASV182, ASV183, ASV184, ASV185, ASV186, ASV187, ASV188, ASV189, ASV190, ASV191, ASV192, ASV193, ASV194, ASV195, ASV196, ASV197, ASV198, ASV199, ASV200, ASV201, ASV202, ASV203, ASV204, ASV205, ASV206, ASV207, ASV208, ASV209, ASV210, ASV211, ASV212, ASV213, ASV214, ASV215, ASV216, ASV217, ASV218, ASV219, ASV220, ASV221, ASV222, ASV223, ASV224, ASV225, ASV226, ASV227, ASV228, ASV229, ASV230, ASV231, ASV232, ASV233, ASV234, ASV235, ASV236, ASV237, ASV238, ASV239, ASV240, ASV241, ASV242, ASV243, ASV244, ASV245, ASV246, ASV247, ASV248, ASV249, ASV250, ASV251, ASV252, ASV253, ASV254, ASV255, ASV256, ASV257, ASV258, ASV259, ASV260, ASV261, ASV262, ASV263, ASV264, ASV265, ASV266, ASV267, ASV268, ASV269, ASV270, ASV271, ASV272, ASV273, ASV274, ASV275, ASV276, ASV277, ASV278, ASV279, ASV280, ASV281, ASV282, ASV283, ASV284, ASV285, ASV286, ASV287, ASV288, ASV289, ASV290, ASV291, ASV292, ASV293, ASV294, ASV295, ASV296, ASV297, ASV298, ASV299, ASV300, ASV301, ASV302, ASV303, ASV304, ASV305, ASV306, ASV307, ASV308, ASV309, ASV310, ASV311, ASV312, ASV313, ASV314, ASV315, ASV316, ASV317, ASV318, ASV319, ASV320, ASV321, ASV322, ASV323, ASV324, ASV325, ASV326, ASV327, ASV328, ASV329, ASV330, ASV331, ASV332, ASV333, ASV334, ASV335, ASV336, ASV337, ASV338, ASV339, ASV340, ASV341, ASV342, ASV343, ASV344, ASV345, ASV346, ASV347, ASV348, ASV349, ASV350, ASV351, ASV352, ASV353, ASV354, ASV355, ASV356, ASV357, ASV358, ASV359, ASV360, ASV361, ASV362, ASV363, ASV364, ASV365, ASV366, ASV367, ASV368, ASV369, ASV370, ASV371, ASV372, ASV373, ASV374, ASV375, ASV376, ASV377, ASV378, ASV379, ASV380, ASV381, ASV382, ASV383, ASV384, ASV385, ASV386, ASV387, ASV388, ASV389, ASV390, ASV391, ASV392, ASV393, ASV394, ASV395, ASV396, ASV397, ASV398, ASV399, ASV400, ASV401, ASV402, ASV403, ASV404, ASV405, ASV406, ASV407, ASV408, ASV409, ASV410, ASV411, ASV412, ASV413, ASV414, ASV415, ASV416, ASV417, ASV418, ASV419, ASV420, ASV421, ASV422, ASV423, ASV424, ASV425, ASV426, ASV427, ASV428, ASV429, ASV430, ASV431, ASV432, ASV433, ASV434, ASV435, ASV436, ASV437, ASV438, ASV439, ASV440, ASV441, ASV442, ASV443, ASV444, ASV445, ASV446, ASV447, ASV448, ASV449, ASV450, ASV451, ASV452, ASV453, ASV454, ASV455, ASV456, ASV457, ASV458, ASV459, ASV460, ASV461, ASV462, ASV463, ASV464, ASV465, ASV466, ASV467, ASV468, ASV469, ASV470, ASV471, ASV472, ASV473, ASV474, ASV475, ASV476, ASV477, ASV478, ASV479, ASV480, ASV481, ASV482, ASV483, ASV484, ASV485, ASV486, ASV487, ASV488, ASV489, ASV490, ASV491, ASV492, ASV493, ASV494, ASV495, ASV496, ASV497, ASV498, ASV499, ASV500, ASV501, ASV502, ASV503, ASV504, ASV505, ASV506, ASV507, ASV508, ASV509, ASV510, ASV511, ASV512, ASV513, ASV514, ASV515, ASV516, ASV517, ASV518, ASV519, ASV520, ASV521, ASV522, ASV523, ASV524, ASV525, ASV526, ASV527, ASV528, ASV529, ASV530, ASV531, ASV532, ASV533, ASV534, ASV535, ASV536, ASV537, ASV538, ASV539, ASV540, ASV541, ASV542, ASV543, ASV544, ASV545, ASV546, ASV547, ASV548, ASV549, ASV550, ASV551, ASV552, ASV553, ASV554, ASV555, ASV556, ASV557, ASV558, ASV559, ASV560, ASV561, ASV562, ASV563, ASV564, ASV565, ASV566, ASV567, ASV568, ASV569, ASV570, ASV571, ASV572, ASV573, ASV574, ASV575, ASV576, ASV577, ASV578, ASV579, ASV580, ASV581, ASV582, ASV583, ASV584, ASV585, ASV586, ASV587, ASV588, ASV589, ASV590, ASV591, ASV592, ASV593, ASV594, ASV595, ASV596, ASV597, ASV598, ASV599, ASV600, ASV601, ASV602, ASV603, ASV604, ASV605, ASV606, ASV607, ASV608, ASV609, ASV610, ASV611, ASV612, ASV613, ASV614, ASV615, ASV616, ASV617, ASV618, ASV619, ASV620, ASV621, ASV622, ASV623, ASV624, ASV625, ASV626, ASV627, ASV628, ASV629, ASV630, ASV631, ASV632, ASV633, ASV634, ASV635, ASV636, ASV637, ASV638, ASV639, ASV640, ASV641, ASV642, ASV643, ASV644, ASV645, ASV646, ASV647, ASV648, ASV649, ASV650, ASV651, ASV652, ASV653, ASV654, ASV655, ASV656, ASV657, ASV658, ASV659, ASV660, ASV661, ASV662, ASV663, ASV664, ASV665, ASV666, ASV667, ASV668, ASV669, ASV670, ASV671, ASV672, ASV673, ASV674, ASV675, ASV676, ASV677, ASV678, ASV679, ASV680, ASV681, ASV682, ASV683, ASV684, ASV685, ASV686, ASV687, ASV688, ASV689, ASV690, ASV691, ASV692, ASV693, ASV694, ASV695, ASV696, ASV697, ASV698, ASV699, ASV700, ASV701, ASV702, ASV703, ASV704, ASV705, ASV706, ASV707, ASV708, ASV709, ASV710, ASV711, ASV712, ASV713, ASV714, ASV715, ASV716, ASV717, ASV718, ASV719, ASV720, ASV721, ASV722, ASV723, ASV724, ASV725, ASV726, ASV727, ASV728, ASV729, ASV730, ASV731, ASV732, ASV733, ASV734, ASV735, ASV736, ASV737, ASV738, ASV739, ASV740, ASV741, ASV742, ASV743, ASV744, ASV745, ASV746, ASV747, ASV748, ASV749, ASV750, ASV751, ASV752, ASV753, ASV754, ASV755, ASV756, ASV757, ASV758, ASV759, ASV760, ASV761, ASV762, ASV763, ASV764, ASV765, ASV766, ASV767, ASV768, ASV769, ASV770, ASV771, ASV772, ASV773, ASV774, ASV775, ASV776, ASV777, ASV778, ASV779, ASV780, ASV781, ASV782, ASV783, ASV784, ASV785, ASV786, ASV787, ASV788, ASV789, ASV790, ASV791, ASV792, ASV793, ASV794, ASV795, ASV796, ASV797, ASV798, ASV799, ASV800, ASV801, ASV802, ASV803, ASV804, ASV805, ASV806, ASV807, ASV808, ASV809, ASV810, ASV811, ASV812, ASV813, ASV814, ASV815, ASV816, ASV817, ASV818, ASV819, ASV820, ASV821, ASV822, ASV823, ASV824, ASV825, ASV826, ASV827, ASV828, ASV829, ASV830, ASV831, ASV832, ASV833, ASV834, ASV835, ASV836, ASV837, ASV838, ASV839, ASV840, ASV841, ASV842, ASV843, ASV844, ASV845, ASV846, ASV847, ASV848, ASV849, ASV850, ASV851, ASV852, ASV853, ASV854, ASV855, ASV856, ASV857, ASV858, ASV859, ASV860, ASV861, ASV862, ASV863, ASV864, ASV865, ASV866, ASV867, ASV868, ASV869, ASV870, ASV871, ASV872, ASV873, ASV874, ASV875, ASV876, ASV877, ASV878, ASV879, ASV880, ASV881, ASV882, ASV883, ASV884, ASV885, ASV886, ASV887, ASV888, ASV889, ASV890, ASV891, ASV892, ASV893, ASV894, ASV895, ASV896, ASV897, ASV898, ASV899, ASV900, ASV901, ASV902, ASV903, ASV904, ASV905, ASV906, ASV907, ASV908, ASV909, ASV910, ASV911, ASV912, ASV913, ASV914, ASV915, ASV916, ASV917, ASV918, ASV919, ASV920, ASV921, ASV922, ASV923, ASV924, ASV925, ASV926, ASV927, ASV928, ASV929, ASV930, ASV931, ASV932, ASV933, ASV934, ASV935, ASV936, ASV937, ASV938, ASV939, ASV940, ASV941, ASV942, ASV943, ASV944, ASV945, ASV946, ASV947, ASV948, ASV949, ASV950, ASV951, ASV952, ASV953, ASV954, ASV955, ASV956, ASV957, ASV958, ASV959, ASV960, ASV961, ASV962, ASV963, ASV964, ASV965, ASV966, ASV967, ASV968, ASV969, ASV970, ASV971, ASV972, ASV973, ASV974, ASV975, ASV976, ASV977, ASV978, ASV979, ASV980, ASV981, ASV982, ASV983, ASV984, ASV985, ASV986, ASV987, ASV988, ASV989, ASV990, ASV991, ASV992, ASV993, ASV994, ASV995, ASV996, ASV997, ASV998, ASV999, ASV1000, ASV1001, ASV1002, ASV1003, ASV1004, ASV1005, ASV1006, ASV1007, ASV1008, ASV1009, ASV1010, ASV1011, ASV1012, ASV1013, ASV1014, ASV1015, ASV1016, ASV1017, ASV1018, ASV1019, ASV1020, ASV1021, ASV1022, ASV1023, ASV1024, ASV1025, ASV1026, ASV1027, ASV1028, ASV1029, ASV1030, ASV1031, ASV1032, ASV1033, ASV1034, ASV1035, ASV1036, ASV1037, ASV1038, ASV1039, ASV1040, ASV1041, ASV1042, ASV1043, ASV1044, ASV1045, ASV1046, ASV1047, ASV1048, ASV1049, ASV1050, ASV1051, ASV1052, ASV1053, ASV1054, ASV1055, ASV1056, ASV1057, ASV1058, ASV1059, ASV1060, ASV1061, ASV1062, ASV1063, ASV1064, ASV1065, ASV1066, ASV1067, ASV1068, ASV1069, ASV1070, ASV1071, ASV1072, ASV1073, ASV1074, ASV1075, ASV1076, ASV1077, ASV1078, ASV1079, ASV1080, ASV1081, ASV1082, ASV1083, ASV1084, ASV1085, ASV1086, ASV1087, ASV1088, ASV1089, ASV1090, ASV1091, ASV1092, ASV1093, ASV1094, ASV1095, ASV1096, ASV1097, ASV1098, ASV1099, ASV1100, ASV1101, ASV1102, ASV1103, ASV1104, ASV1105, ASV1106, ASV1107, ASV1108, ASV1109, ASV1110, ASV1111, ASV1112, ASV1113, ASV1114, ASV1115, ASV1116, ASV1117, ASV1118, ASV1119, ASV1120, ASV1121, ASV1122, ASV1123, ASV1124, ASV1125, ASV1126, ASV1127, ASV1128, ASV1129, ASV1130, ASV1131, ASV1132, ASV1133, ASV1134, ASV1135, ASV1136, ASV1137, ASV1138, ASV1139, ASV1140, ASV1141, ASV1142, ASV1143, ASV1144, ASV1145, ASV1146, ASV1147, ASV1148, ASV1149, ASV1150, ASV1151, ASV1152, ASV1153, ASV1154, ASV1155, ASV1156, ASV1157, ASV1158, ASV1159, ASV1160, ASV1161, ASV1162, ASV1163, ASV1164, ASV1165, ASV1166, ASV1167, ASV1168, ASV1169, ASV1170, ASV1171, ASV1172, ASV1173, ASV1174, ASV1175, ASV1176, ASV1177, ASV1178, ASV1179, ASV1180, ASV1181, ASV1182, ASV1183, ASV1184, ASV1185, ASV1186, ASV1187, ASV1188, ASV1189, ASV1190, ASV1191, ASV1192, ASV1193, ASV1194, ASV1195, ASV1196, ASV1197, ASV1198, ASV1199, ASV1200, ASV1201, ASV1202, ASV1203, ASV1204, ASV1205, ASV1206, ASV1207, ASV1208, ASV1209, ASV1210, ASV1211, ASV1212, ASV1213, ASV1214, ASV1215, ASV1216, ASV1217, ASV1218, ASV1219, ASV1220, ASV1221, ASV1222, ASV1223, ASV1224, ASV1225, ASV1226, ASV1227, ASV1228, ASV1229, ASV1230, ASV1231, ASV1232, ASV1233, ASV1234, ASV1235, ASV1236, ASV1237, ASV1238, ASV1239, ASV1240, ASV1241, ASV1242, ASV1243, ASV1244, ASV1245, ASV1246, ASV1247, ASV1248, ASV1249, ASV1250, ASV1251, ASV1252, ASV1253, ASV1254, ASV1255, ASV1256, ASV1257, ASV1258, ASV1259, ASV1260, ASV1261, ASV1262, ASV1263, ASV1264, ASV1265, ASV1266, ASV1267, ASV1268, ASV1269, ASV1270, ASV1271, ASV1272, ASV1273, ASV1274, ASV1275, ASV1276, ASV1277, ASV1278, ASV1279, ASV1280, ASV1281, ASV1282, ASV1283, ASV1284, ASV1285, ASV1286, ASV1287, ASV1288, ASV1289, ASV1290, ASV1291, ASV1292, ASV1293, ASV1294, ASV1295, ASV1296, ASV1297, ASV1298, ASV1299, ASV1300, ASV1301, ASV1302, ASV1303, ASV1304, ASV1305, ASV1306, ASV1307, ASV1308, ASV1309, ASV1310, ASV1311, ASV1312, ASV1313, ASV1314, ASV1315, ASV1316, ASV1317, ASV1318, ASV1319, ASV1320, ASV1321, ASV1322, ASV1323, ASV1324, ASV1325, ASV1326, ASV1327, ASV1328, ASV1329, ASV1330, ASV1331, ASV1332, ASV1333, ASV1334, ASV1335, ASV1336, ASV1337, ASV1338, ASV1339, ASV1340, ASV1341, ASV1342, ASV1343, ASV1344, ASV1345, ASV1346, ASV1347, ASV1348, ASV1349, ASV1350, ASV1351, ASV1352, ASV1353, ASV1354, ASV1355, ASV1356, ASV1357, ASV1358, ASV1359, ASV1360, ASV1361, ASV1362, ASV1363, ASV1364, ASV1365, ASV1366, ASV1367, ASV1368, ASV1369, ASV1370, ASV1371, ASV1372, ASV1373, ASV1374, ASV1375, ASV1376, ASV1377, ASV1378, ASV1379, ASV1380, ASV1381, ASV1382, ASV1383, ASV1384, ASV1385, ASV1386, ASV1387, ASV1388, ASV1389, ASV1390, ASV1391, ASV1392, ASV1393, ASV1394, ASV1395, ASV1396, ASV1397, ASV1398, ASV1399, ASV1400, ASV1401, ASV1402, ASV1403, ASV1404, ASV1405, ASV1406, ASV1407, ASV1408, ASV1409, ASV1410, ASV1411, ASV1412, ASV1413, ASV1414, ASV1415, ASV1416, ASV1417, ASV1418, ASV1419, ASV1420, ASV1421, ASV1422, ASV1423, ASV1424, ASV1425, ASV1426, ASV1427, ASV1428, ASV1429, ASV1430, ASV1431, ASV1432, ASV1433, ASV1434, ASV1435, ASV1436, ASV1437, ASV1438, ASV1439, ASV1440, ASV1441, ASV1442, ASV1443, ASV1444, ASV1445, ASV1446, ASV1447, ASV1448, ASV1449, ASV1450, ASV1451, ASV1452, ASV1453, ASV1454, ASV1455, ASV1456, ASV1457, ASV1458, ASV1459, ASV1460, ASV1461, ASV1462, ASV1463, ASV1464, ASV1465, ASV1466, ASV1467, ASV1468, ASV1469, ASV1470, ASV1471, ASV1472, ASV1473, ASV1474, ASV1475, ASV1476, ASV1477, ASV1478, ASV1479, ASV1480, ASV1481, ASV1482, ASV1483, ASV1484, ASV1485, ASV1486, ASV1487, ASV1488, ASV1489, ASV1490, ASV1491, ASV1492, ASV1493, ASV1494, ASV1495, ASV1496, ASV1497, ASV1498, ASV1499, ASV1500, ASV1501, ASV1502, ASV1503, ASV1504, ASV1505, ASV1506, ASV1507, ASV1508, ASV1509, ASV1510, ASV1511, ASV1512, ASV1513, ASV1514, ASV1515, ASV1516, ASV1517, ASV1518, ASV1519, ASV1520, ASV1521, ASV1522, ASV1523, ASV1524, ASV1525, ASV1526, ASV1527, ASV1528, ASV1529, ASV1530, ASV1531, ASV1532, ASV1533, ASV1534, ASV1535, ASV1536, ASV1537, ASV1538, ASV1539, ASV1540, ASV1541, ASV1542, ASV1543, ASV1544, ASV1545, ASV1546, ASV1547, ASV1548, ASV1549, ASV1550, ASV1551, ASV1552, ASV1553, ASV1554, ASV1555, ASV1556, ASV1557, ASV1558, ASV1559, ASV1560, ASV1561, ASV1562, ASV1563, ASV1564, ASV1565, ASV1566, ASV1567, ASV1568, ASV1569, ASV1570, ASV1571, ASV1572, ASV1573, ASV1574, ASV1575, ASV1576, ASV1577, ASV1578, ASV1579, ASV1580, ASV1581, ASV1582, ASV1583, ASV1584, ASV1585, ASV1586, ASV1587, ASV1588, ASV1589, ASV1590, ASV1591, ASV1592, ASV1593, ASV1594, ASV1595, ASV1596, ASV1597, ASV1598, ASV1599, ASV1600, ASV1601, ASV1602, ASV1603, ASV1604, ASV1605, ASV1606, ASV1607, ASV1608, ASV1609, ASV1610, ASV1611, ASV1612, ASV1613, ASV1614, ASV1615, ASV1616, ASV1617, ASV1618, ASV1619, ASV1620, ASV1621, ASV1622, ASV1623, ASV1624, ASV1625, ASV1626, ASV1627, ASV1628, ASV1629, ASV1630, ASV1631, ASV1632, ASV1633, ASV1634, ASV1635, ASV1636, ASV1637, ASV1638, ASV1639, ASV1640, ASV1641, ASV1642, ASV1643, ASV1644, ASV1645, ASV1646, ASV1647, ASV1648, ASV1649, ASV1650, ASV1651, ASV1652, ASV1653, ASV1654, ASV1655, ASV1656, ASV1657, ASV1658, ASV1659, ASV1660, ASV1661, ASV1662, ASV1663, ASV1664, ASV1665, ASV1666, ASV1667, ASV1668, ASV1669, ASV1670, ASV1671, ASV1672, ASV1673, ASV1674, ASV1675, ASV1676, ASV1677, ASV1678, ASV1679, ASV1680, ASV1681, ASV1682, ASV1683, ASV1684, ASV1685, ASV1686, ASV1687, ASV1688, ASV1689, ASV1690, ASV1691, ASV1692, ASV1693, ASV1694, ASV1695, ASV1696, ASV1697, ASV1698, ASV1699, ASV1700, ASV1701, ASV1702, ASV1703, ASV1704, ASV1705, ASV1706, ASV1707, ASV1708, ASV1709, ASV1710, ASV1711, ASV1712, ASV1713, ASV1714, ASV1715, ASV1716, ASV1717, ASV1718, ASV1719, ASV1720, ASV1721, ASV1722, ASV1723, ASV1724, ASV1725, ASV1726, ASV1727, ASV1728, ASV1729, ASV1730, ASV1731, ASV1732, ASV1733, ASV1734, ASV1735, ASV1736, ASV1737, ASV1738, ASV1739, ASV1740, ASV1741, ASV1742, ASV1743, ASV1744, ASV1745, ASV1746, ASV1747, ASV1748, ASV1749, ASV1750, ASV1751, ASV1752, ASV1753, ASV1754, ASV1755, ASV1756, ASV1757, ASV1758, ASV1759, ASV1760, ASV1761, ASV1762, ASV1763, ASV1764, ASV1765, ASV1766, ASV1767, ASV1768, ASV1769, ASV1770, ASV1771, ASV1772, ASV1773, ASV1774, ASV1775, ASV1776, ASV1777, ASV1778, ASV1779, ASV1780, ASV1781, ASV1782, ASV1783, ASV1784, ASV1785, ASV1786, ASV1787, ASV1788, ASV1789, ASV1790, ASV1791, ASV1792, ASV1793, ASV1794, ASV1795, ASV1796, ASV1797, ASV1798, ASV1799, ASV1800, ASV1801, ASV1802, ASV1803, ASV1804, ASV1805, ASV1806, ASV1807, ASV1808, ASV1809, ASV1810, ASV1811, ASV1812, ASV1813, ASV1814, ASV1815, ASV1816, ASV1817, ASV1818, ASV1819, ASV1820, ASV1821, ASV1822, ASV1823, ASV1824, ASV1825, ASV1826, ASV1827, ASV1828, ASV1829, ASV1830, ASV1831, ASV1832, ASV1833, ASV1834, ASV1835, ASV1836, ASV1837, ASV1838, ASV1839, ASV1840, ASV1841, ASV1842, ASV1843, ASV1844, ASV1845, ASV1846, ASV1847, ASV1848, ASV1849, ASV1850, ASV1851, ASV1852, ASV1853, ASV1854, ASV1855, ASV1856, ASV1857, ASV1858, ASV1859, ASV1860, ASV1861, ASV1862, ASV1863, ASV1864, ASV1865, ASV1866, ASV1867, ASV1868, ASV1869, ASV1870, ASV1871, ASV1872, ASV1873, ASV1874, ASV1875, ASV1876, ASV1877, ASV1878, ASV1879, ASV1880, ASV1881, ASV1882, ASV1883, ASV1884, ASV1885, ASV1886, ASV1887, ASV1888, ASV1889, ASV1890, ASV1891, ASV1892, ASV1893, ASV1894, ASV1895, ASV1896, ASV1897, ASV1898, ASV1899, ASV1900, ASV1901, ASV1902, ASV1903, ASV1904, ASV1905, ASV1906, ASV1907, ASV1908, ASV1909, ASV1910, ASV1911, ASV1912, ASV1913, ASV1914, ASV1915, ASV1916, ASV1917, ASV1918, ASV1919, ASV1920, ASV1921, ASV1922, ASV1923, ASV1924, ASV1925, ASV1926, ASV1927, ASV1928, ASV1929, ASV1930, ASV1931, ASV1932, ASV1933, ASV1934, ASV1935, ASV1936, ASV1937, ASV1938, ASV1939, ASV1940, ASV1941, ASV1942, ASV1943, ASV1944, ASV1945, ASV1946, ASV1947, ASV1948, ASV1949, ASV1950, ASV1951, ASV1952, ASV1953, ASV1954, ASV1955, ASV1956, ASV1957, ASV1958, ASV1959, ASV1960, ASV1961, ASV1962, ASV1963, ASV1964, ASV1965, ASV1966, ASV1967, ASV1968, ASV1969, ASV1970, ASV1971, ASV1972, ASV1973, ASV1974, ASV1975, ASV1976, ASV1977, ASV1978, ASV1979, ASV1980, ASV1981, ASV1982, ASV1983, ASV1984, ASV1985, ASV1986, ASV1987, ASV1988, ASV1989, ASV1990, ASV1991, ASV1992, ASV1993, ASV1994, ASV1995, ASV1996, ASV1997, ASV1998, ASV1999, ASV2000, ASV2001, ASV2002, ASV2003, ASV2004, ASV2005, ASV2006, ASV2007, ASV2008, ASV2009, ASV2010, ASV2011, ASV2012, ASV2013, ASV2014, ASV2015, ASV2016, ASV2017, ASV2018, ASV2019, ASV2020, ASV2021, ASV2022, ASV2023, ASV2024, ASV2025, ASV2026, ASV2027, ASV2028, ASV2029, ASV2030, ASV2031, ASV2032, ASV2033, ASV2034, ASV2035, ASV2036, ASV2037, ASV2038, ASV2039, ASV2040, ASV2041, ASV2042, ASV2043, ASV2044, ASV2045, ASV2046, ASV2047, ASV2048, ASV2049, ASV2050, ASV2051, ASV2052, ASV2053, ASV2054, ASV2055, ASV2056, ASV2057, ASV2058, ASV2059, ASV2060, ASV2061, ASV2062, ASV2063, ASV2064, ASV2065, ASV2066, ASV2067, ASV2068, ASV2069, ASV2070, ASV2071, ASV2072, ASV2073, ASV2074, ASV2075, ASV2076, ASV2077, ASV2078, ASV2079, ASV2080, ASV2081, ASV2082, ASV2083, ASV2084, ASV2085, ASV2086, ASV2087, ASV2088, ASV2089, ASV2090, ASV2091, ASV2092, ASV2093, ASV2094, ASV2095, ASV2096, ASV2097, ASV2098, ASV2099, ASV2100, ASV2101, ASV2102, ASV2103, ASV2104, ASV2105, ASV2106, ASV2107, ASV2108, ASV2109, ASV2110, ASV2111, ASV2112, ASV2113, ASV2114, ASV2115, ASV2116, ASV2117, ASV2118, ASV2119, ASV2120, ASV2121, ASV2122, ASV2123, ASV2124, ASV2125, ASV2126, ASV2127, ASV2128, ASV2129, ASV2130, ASV2131, ASV2132, ASV2133, ASV2134, ASV2135, ASV2136, ASV2137, ASV2138, ASV2139, ASV2140, ASV2141, ASV2142, ASV2143, ASV2144, ASV2145, ASV2146, ASV2147, ASV2148, ASV2149, ASV2150, ASV2151, ASV2152, ASV2153, ASV2154, ASV2155, ASV2156, ASV2157, ASV2158, ASV2159, ASV2160, ASV2161, ASV2162, ASV2163, ASV2164, ASV2165, ASV2166, ASV2167, ASV2168, ASV2169, ASV2170, ASV2171, ASV2172, ASV2173, ASV2174, ASV2175, ASV2176, ASV2177, ASV2178, ASV2179, ASV2180, ASV2181, ASV2182, ASV2183, ASV2184, ASV2185, ASV2186, ASV2187, ASV2188, ASV2189, ASV2190, ASV2191, ASV2192, ASV2193, ASV2194, ASV2195, ASV2196, ASV2197, ASV2198, ASV2199, ASV2200, ASV2201, ASV2202, ASV2203, ASV2204, ASV2205, ASV2206, ASV2207, ASV2208, ASV2209, ASV2210, ASV2211, ASV2212, ASV2213, ASV2214, ASV2215, ASV2216, ASV2217, ASV2218, ASV2219, ASV2220, ASV2221, ASV2222, ASV2223, ASV2224, ASV2225, ASV2226, ASV2227, ASV2228, ASV2229, ASV2230, ASV2231, ASV2232, ASV2233, ASV2234, ASV2235, ASV2236, ASV2237, ASV2238, ASV2239, ASV2240, ASV2241, ASV2242, ASV2243, ASV2244, ASV2245, ASV2246, ASV2247, ASV2248, ASV2249, ASV2250, ASV2251, ASV2252, ASV2253, ASV2254, ASV2255, ASV2256, ASV2257, ASV2258, ASV2259, ASV2260, ASV2261, ASV2262, ASV2263, ASV2264, ASV2265, ASV2266, ASV2267, ASV2268, ASV2269, ASV2270, ASV2271, ASV2272, ASV2273, ASV2274, ASV2275, ASV2276, ASV2277, ASV2278, ASV2279, ASV2280, ASV2281, ASV2282, ASV2283, ASV2284, ASV2285, ASV2286, ASV2287, ASV2288, ASV2289, ASV2290, ASV2291, ASV2292, ASV2293, ASV2294, ASV2295, ASV2296, ASV2297, ASV2298, ASV2299, ASV2300, ASV2301, ASV2302, ASV2303, ASV2304, ASV2305, ASV2306, ASV2307, ASV2308, ASV2309, ASV2310, ASV2311, ASV2312, ASV2313, ASV2314, ASV2315, ASV2316, ASV2317, ASV2318, ASV2319, ASV2320, ASV2321, ASV2322, ASV2323, ASV2324, ASV2325, ASV2326, ASV2327, ASV2328, ASV2329, ASV2330, ASV2331, ASV2332, ASV2333, ASV2334, ASV2335, ASV2336, ASV2337, ASV2338, ASV2339, ASV2340, ASV2341, ASV2342, ASV2343, ASV2344, ASV2345, ASV2346, ASV2347, ASV2348, ASV2349, ASV2350, ASV2351, ASV2352, ASV2353, ASV2354, ASV2355, ASV2356, ASV2357, ASV2358, ASV2359, ASV2360, ASV2361, ASV2362, ASV2363, ASV2364, ASV2365, ASV2366, ASV2367, ASV2368, ASV2369, ASV2370, ASV2371, ASV2372, ASV2373, ASV2374, ASV2375, ASV2376, ASV2377, ASV2378, ASV2379, ASV2380, ASV2381, ASV2382, ASV2383, ASV2384, ASV2385, ASV2386, ASV2387, ASV2388, ASV2389, ASV2390, ASV2391, ASV2392, ASV2393, ASV2394, ASV2395, ASV2396, ASV2397, ASV2398, ASV2399, ASV2400, ASV2401, ASV2402, ASV2403, ASV2404, ASV2405, ASV2406, ASV2407, ASV2408, ASV2409, ASV2410, ASV2411, ASV2412, ASV2413, ASV2414, ASV2415, ASV2416, ASV2417, ASV2418, ASV2419, ASV2420, ASV2421, ASV2422, ASV2423, ASV2424, ASV2425, ASV2426, ASV2427, ASV2428, ASV2429, ASV2430, ASV2431, ASV2432, ASV2433, ASV2434, ASV2435, ASV2436, ASV2437, ASV2438, ASV2439, ASV2440, ASV2441, ASV2442, ASV2443, ASV2444, ASV2445, ASV2446, ASV2447, ASV2448, ASV2449, ASV2450, ASV2451, ASV2452, ASV2453, ASV2454, ASV2455, ASV2456, ASV2457, ASV2458, ASV2459, ASV2460, ASV2461, ASV2462, ASV2463, ASV2464, ASV2465, ASV2466, ASV2467, ASV2468, ASV2469, ASV2470, ASV2471, ASV2472, ASV2473, ASV2474, ASV2475, ASV2476, ASV2477, ASV2478, ASV2479, ASV2480, ASV2481, ASV2482, ASV2483, ASV2484, ASV2485, ASV2486, ASV2487, ASV2488, ASV2489, ASV2490, ASV2491, ASV2492, ASV2493, ASV2494, ASV2495, ASV2496, ASV2497, ASV2498, ASV2499, ASV2500, ASV2501, ASV2502, ASV2503, ASV2504, ASV2505, ASV2506, ASV2507, ASV2508, ASV2509, ASV2510, ASV2511, ASV2512, ASV2513, ASV2514, ASV2515, ASV2516, ASV2517, ASV2518, ASV2519, ASV2520, ASV2521, ASV2522, ASV2523, ASV2524, ASV2525, ASV2526, ASV2527, ASV2528, ASV2529, ASV2530, ASV2531, ASV2532, ASV2533, ASV2534, ASV2535, ASV2536, ASV2537, ASV2538, ASV2539, ASV2540, ASV2541, ASV2542, ASV2543, ASV2544, ASV2545, ASV2546, ASV2547, ASV2548, ASV2549, ASV2550, ASV2551, ASV2552, ASV2553, ASV2554, ASV2555, ASV2556, ASV2557, ASV2558, ASV2559, ASV2560
2024-03-06 21:59:00.382173 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:59:00.382401 INFO::Filtered feature names from variance filtering:
2024-03-06 21:59:00.382604 INFO::Running selected normalization method: TSS
2024-03-06 21:59:00.382992 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:59:00.385293 INFO::Running selected transform method: LOG
2024-03-06 21:59:00.385664 INFO::Running selected analysis method: LM
2024-03-06 21:59:00.385926 INFO::Fitting model to feature number 1, ASV1
2024-03-06 21:59:00.387026 INFO::Fitting model to feature number 2, ASV2
2024-03-06 21:59:00.387912 INFO::Fitting model to feature number 3, ASV4
2024-03-06 21:59:00.388775 INFO::Fitting model to feature number 4, ASV7
2024-03-06 21:59:00.3896 INFO::Fitting model to feature number 5, ASV8
2024-03-06 21:59:00.39042 INFO::Fitting model to feature number 6, ASV11
2024-03-06 21:59:00.391244 INFO::Fitting model to feature number 7, ASV14
2024-03-06 21:59:00.392108 INFO::Fitting model to feature number 8, ASV16
2024-03-06 21:59:00.392923 INFO::Fitting model to feature number 9, ASV79
2024-03-06 21:59:00.395558 INFO::Counting total values for each feature
2024-03-06 21:59:00.396221 INFO::Writing filtered data to file results/maaslin2/asv_cycle_premeno/features/filtered_data.tsv
2024-03-06 21:59:00.396875 INFO::Writing filtered, normalized data to file results/maaslin2/asv_cycle_premeno/features/filtered_data_norm.tsv
2024-03-06 21:59:00.397433 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/asv_cycle_premeno/features/filtered_data_norm_transformed.tsv
2024-03-06 21:59:00.398184 WARNING::Deleting existing residuals file: results/maaslin2/asv_cycle_premeno/fits/residuals.rds
2024-03-06 21:59:00.398659 INFO::Writing residuals to file results/maaslin2/asv_cycle_premeno/fits/residuals.rds
2024-03-06 21:59:00.399076 WARNING::Deleting existing fitted file: results/maaslin2/asv_cycle_premeno/fits/fitted.rds
2024-03-06 21:59:00.399412 INFO::Writing fitted values to file results/maaslin2/asv_cycle_premeno/fits/fitted.rds
2024-03-06 21:59:00.399785 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/asv_cycle_premeno/all_results.tsv
2024-03-06 21:59:00.400349 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/asv_cycle_premeno/significant_results.tsv
2024-03-06 21:59:00.400736 INFO::Writing heatmap of significant results to file: results/maaslin2/asv_cycle_premeno/heatmap.pdf
[1] "There is not enough metadata in the associations to create a heatmap plot. Please review the associations in text output file."
2024-03-06 21:59:00.401271 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/asv_cycle_premeno
2024-03-06 21:59:00.401809 INFO::Plotting associations from most to least significant, grouped by metadata
2024-03-06 21:59:00.402086 INFO::Plotting data for metadata number 1, CycleDaySampling
2024-03-06 21:59:00.403363 INFO::Creating scatter plot for continuous data, CycleDaySampling vs ASV4</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:00.474795 INFO::Creating scatter plot for continuous data, CycleDaySampling vs ASV7</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:00.541566 INFO::Creating scatter plot for continuous data, CycleDaySampling vs ASV14</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).
Removed 1 rows containing missing values (`geom_point()`).
Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>ps_pre_genus <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_pre, <span class="st">"Genus"</span>)</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>genus_cycleday <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_pre_genus),</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_pre_genus), <span class="st">"data.frame"</span>),</span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/genus_cycle_premeno"</span>,</span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">5</span>,</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.3</span>,</span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"CycleDaySampling"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:04.656212 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/genus_cycle_premeno/maaslin2.log"
2024-03-06 21:59:04.829481 INFO::Writing function arguments to log file
2024-03-06 21:59:04.831474 INFO::Verifying options selected are valid
2024-03-06 21:59:04.831714 INFO::Determining format of input files
2024-03-06 21:59:04.831931 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:59:04.833684 INFO::Formula for fixed effects: expr ~  CycleDaySampling
2024-03-06 21:59:04.83394 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:59:04.834149 INFO::Total samples in data: 25
2024-03-06 21:59:04.834346 INFO::Min samples required with min abundance for a feature not to be filtered: 7.500000
2024-03-06 21:59:04.835791 INFO::Total filtered features: 318
2024-03-06 21:59:04.836231 INFO::Filtered feature names from abundance and prevalence filtering: ASV3, ASV6, ASV11, ASV12, ASV13, ASV17, ASV21, ASV22, ASV24, ASV30, ASV34, ASV35, ASV38, ASV44, ASV48, ASV53, ASV57, ASV59, ASV63, ASV69, ASV73, ASV75, ASV83, ASV88, ASV95, ASV104, ASV119, ASV125, ASV131, ASV136, ASV148, ASV151, ASV152, ASV154, ASV155, ASV165, ASV170, ASV177, ASV191, ASV196, ASV197, ASV200, ASV201, ASV202, ASV204, ASV207, ASV208, ASV218, ASV224, ASV225, ASV228, ASV249, ASV255, ASV259, ASV264, ASV268, ASV276, ASV277, ASV280, ASV281, ASV284, ASV294, ASV307, ASV312, ASV319, ASV322, ASV331, ASV337, ASV347, ASV349, ASV353, ASV358, ASV362, ASV363, ASV367, ASV371, ASV375, ASV378, ASV384, ASV390, ASV392, ASV395, ASV405, ASV407, ASV410, ASV414, ASV421, ASV431, ASV432, ASV433, ASV440, ASV443, ASV455, ASV459, ASV463, ASV466, ASV470, ASV473, ASV475, ASV482, ASV483, ASV487, ASV494, ASV518, ASV526, ASV532, ASV556, ASV557, ASV558, ASV570, ASV571, ASV573, ASV586, ASV601, ASV605, ASV608, ASV610, ASV612, ASV615, ASV618, ASV622, ASV628, ASV637, ASV641, ASV650, ASV653, ASV671, ASV685, ASV686, ASV690, ASV693, ASV699, ASV709, ASV711, ASV728, ASV734, ASV740, ASV760, ASV764, ASV785, ASV791, ASV793, ASV808, ASV815, ASV821, ASV823, ASV824, ASV829, ASV838, ASV840, ASV849, ASV853, ASV855, ASV860, ASV864, ASV881, ASV892, ASV913, ASV915, ASV918, ASV925, ASV927, ASV936, ASV948, ASV956, ASV967, ASV979, ASV980, ASV984, ASV988, ASV1000, ASV1004, ASV1008, ASV1009, ASV1011, ASV1033, ASV1040, ASV1044, ASV1050, ASV1055, ASV1068, ASV1069, ASV1070, ASV1083, ASV1087, ASV1088, ASV1093, ASV1103, ASV1109, ASV1111, ASV1115, ASV1117, ASV1123, ASV1132, ASV1134, ASV1136, ASV1145, ASV1167, ASV1172, ASV1176, ASV1194, ASV1196, ASV1197, ASV1200, ASV1203, ASV1221, ASV1222, ASV1235, ASV1236, ASV1238, ASV1242, ASV1249, ASV1251, ASV1253, ASV1255, ASV1260, ASV1275, ASV1282, ASV1284, ASV1285, ASV1286, ASV1291, ASV1304, ASV1316, ASV1317, ASV1321, ASV1322, ASV1331, ASV1344, ASV1357, ASV1361, ASV1367, ASV1382, ASV1383, ASV1414, ASV1426, ASV1427, ASV1428, ASV1438, ASV1439, ASV1440, ASV1483, ASV1489, ASV1499, ASV1501, ASV1503, ASV1517, ASV1520, ASV1534, ASV1535, ASV1536, ASV1537, ASV1539, ASV1550, ASV1565, ASV1601, ASV1602, ASV1615, ASV1617, ASV1624, ASV1627, ASV1647, ASV1662, ASV1666, ASV1669, ASV1671, ASV1672, ASV1682, ASV1689, ASV1694, ASV1724, ASV1732, ASV1734, ASV1767, ASV1769, ASV1774, ASV1778, ASV1783, ASV1812, ASV1857, ASV1859, ASV1862, ASV1869, ASV1887, ASV1909, ASV1910, ASV1918, ASV1926, ASV1935, ASV1936, ASV1937, ASV1943, ASV1953, ASV1956, ASV1980, ASV2008, ASV2029, ASV2033, ASV2047, ASV2071, ASV2087, ASV2119, ASV2159, ASV2214, ASV2247, ASV2267, ASV2269, ASV2273, ASV2279, ASV2322, ASV2328, ASV2329, ASV2331, ASV2387, ASV2395, ASV2396, ASV2480, ASV2525
2024-03-06 21:59:04.836714 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:59:04.836972 INFO::Filtered feature names from variance filtering:
2024-03-06 21:59:04.837188 INFO::Running selected normalization method: TSS
2024-03-06 21:59:04.837557 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:59:04.840303 INFO::Running selected transform method: LOG
2024-03-06 21:59:04.840842 INFO::Running selected analysis method: LM
2024-03-06 21:59:04.841159 INFO::Fitting model to feature number 1, ASV1
2024-03-06 21:59:04.842413 INFO::Fitting model to feature number 2, ASV4
2024-03-06 21:59:04.845422 INFO::Counting total values for each feature
2024-03-06 21:59:04.846165 INFO::Writing filtered data to file results/maaslin2/genus_cycle_premeno/features/filtered_data.tsv
2024-03-06 21:59:04.846824 INFO::Writing filtered, normalized data to file results/maaslin2/genus_cycle_premeno/features/filtered_data_norm.tsv
2024-03-06 21:59:04.847355 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/genus_cycle_premeno/features/filtered_data_norm_transformed.tsv
2024-03-06 21:59:04.847901 WARNING::Deleting existing residuals file: results/maaslin2/genus_cycle_premeno/fits/residuals.rds
2024-03-06 21:59:04.848418 INFO::Writing residuals to file results/maaslin2/genus_cycle_premeno/fits/residuals.rds
2024-03-06 21:59:04.849093 WARNING::Deleting existing fitted file: results/maaslin2/genus_cycle_premeno/fits/fitted.rds
2024-03-06 21:59:04.849524 INFO::Writing fitted values to file results/maaslin2/genus_cycle_premeno/fits/fitted.rds
2024-03-06 21:59:04.8499 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/genus_cycle_premeno/all_results.tsv
2024-03-06 21:59:04.850556 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/genus_cycle_premeno/significant_results.tsv
2024-03-06 21:59:04.851071 INFO::Writing heatmap of significant results to file: results/maaslin2/genus_cycle_premeno/heatmap.pdf
[1] "There is not enough metadata in the associations to create a heatmap plot. Please review the associations in text output file."
2024-03-06 21:59:04.851648 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/genus_cycle_premeno
2024-03-06 21:59:04.852249 INFO::Plotting associations from most to least significant, grouped by metadata
2024-03-06 21:59:04.852565 INFO::Plotting data for metadata number 1, CycleDaySampling
2024-03-06 21:59:04.85314 INFO::Creating scatter plot for continuous data, CycleDaySampling vs ASV4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:04.918942 INFO::Creating scatter plot for continuous data, CycleDaySampling vs ASV1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).
Removed 1 rows containing missing values (`geom_point()`).
Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_pre_genus) <span class="sc">%&gt;%</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ASV"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ASV <span class="sc">%in%</span> <span class="fu">filter</span>(genus_cycleday<span class="sc">$</span>results, qval <span class="sc">&lt;</span> <span class="fl">0.2</span>)<span class="sc">$</span>feature) <span class="sc">%&gt;%</span></span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 4%">
<col style="width: 8%">
<col style="width: 16%">
<col style="width: 14%">
<col style="width: 17%">
<col style="width: 18%">
<col style="width: 13%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">ASV</th>
<th style="text-align: left;">Kingdom</th>
<th style="text-align: left;">Phylum</th>
<th style="text-align: left;">Class</th>
<th style="text-align: left;">Order</th>
<th style="text-align: left;">Family</th>
<th style="text-align: left;">Genus</th>
<th style="text-align: left;">Species</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ASV1</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Firmicutes</td>
<td style="text-align: left;">Bacilli</td>
<td style="text-align: left;">Lactobacillales</td>
<td style="text-align: left;">Lactobacillaceae</td>
<td style="text-align: left;">Lactobacillus</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">ASV4</td>
<td style="text-align: left;">Bacteria</td>
<td style="text-align: left;">Actinobacteriota</td>
<td style="text-align: left;">Actinobacteria</td>
<td style="text-align: left;">Bifidobacteriales</td>
<td style="text-align: left;">Bifidobacteriaceae</td>
<td style="text-align: left;">Gardnerella</td>
<td style="text-align: left;">NA</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>psmelt_pre_genus <span class="ot">=</span> <span class="fu">psmelt</span>(ps_pre_genus)</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>maaslin_res_genus_cycleday <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">filter</span>(psmelt_pre_genus, OTU <span class="sc">%in%</span> <span class="fu">filter</span>(genus_cycleday<span class="sc">$</span>results, qval <span class="sc">&lt;</span> <span class="fl">0.2</span>)<span class="sc">$</span>feature),</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">aes</span>(<span class="at">x =</span> CycleDaySampling, <span class="at">y =</span> Abundance)) <span class="sc">+</span></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Genus, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> F, <span class="at">formula =</span> y <span class="sc">~</span><span class="fu">poly</span>(x, <span class="dv">3</span>, <span class="at">raw =</span>T)) <span class="sc">+</span></span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="cf">function</span>(x){<span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fl">0.1</span>, x))}) <span class="sc">+</span></span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">28</span>)) <span class="sc">+</span></span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Day of sampling"</span>,</span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Genus Abundance (%)"</span>,</span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Maaslin2 results"</span>)</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>maaslin_res_genus_cycleday</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/cycle%20pre-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(maaslin_res_genus_cycleday, <span class="at">filename =</span> <span class="st">"results/maaslin2/genus_CycleDay_panelplot.png"</span>,</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><em>Gardnerella</em> is highest in the early days of the menstrual cycle while <em>Lactobacillus</em> during the middle of the cycle. This seems to correspond to <em>Vomstein et al 2022</em>.</p>
</section>
<section id="influence-of-sexual-activity-on-microbiome-composition" class="level3">
<h3 class="anchored" data-anchor-id="influence-of-sexual-activity-on-microbiome-composition">Influence of sexual activity on microbiome composition</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sexual activity</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>ps_rel_genus <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_samples, <span class="at">taxrank =</span> <span class="st">"Genus"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">transform_sample_counts</span>(<span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x) <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>ps_rel_genus_sexuallyactive <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_rel_genus, <span class="fu">sample_data</span>(ps_rel_genus)<span class="sc">$</span>SexuallyActive <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"yes"</span>, <span class="st">"no"</span>)) <span class="co"># for 3 patients, this information is not available</span></span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a>ps_abs_genus <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_samples, <span class="at">taxrank =</span> <span class="st">"Genus"</span>) </span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>ps_abs_genus_sexuallyactive <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_abs_genus, <span class="fu">sample_data</span>(ps_abs_genus)<span class="sc">$</span>SexuallyActive <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"yes"</span>, <span class="st">"no"</span>))</span>
<span id="cb98-8"><a href="#cb98-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-9"><a href="#cb98-9" aria-hidden="true" tabindex="-1"></a>ps_rel_genus_sexuallyactive</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 320 taxa and 72 samples ]
sample_data() Sample Data:       [ 72 samples by 15 sample variables ]
tax_table()   Taxonomy Table:    [ 320 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 320 tips and 319 internal nodes ]
refseq()      DNAStringSet:      [ 320 reference sequences ]</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Via Maaslin2</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>genus_SexuallyActive <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_rel_genus_sexuallyactive),</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_rel_genus_sexuallyactive), <span class="st">"data.frame"</span>),</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/genus_SexuallyActive"</span>,</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">5</span>,</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.2</span>,</span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"SexuallyActive"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:14.319025 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/genus_SexuallyActive/maaslin2.log"
2024-03-06 21:59:14.49309 INFO::Writing function arguments to log file
2024-03-06 21:59:14.495832 INFO::Verifying options selected are valid
2024-03-06 21:59:14.496133 INFO::Determining format of input files
2024-03-06 21:59:14.49638 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:59:14.498326 INFO::Formula for fixed effects: expr ~  SexuallyActive
2024-03-06 21:59:14.498623 INFO::Factor detected for categorial metadata 'SexuallyActive'. Provide a reference argument or manually set factor ordering to change reference level.
2024-03-06 21:59:14.498835 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:59:14.49905 INFO::Total samples in data: 72
2024-03-06 21:59:14.499256 INFO::Min samples required with min abundance for a feature not to be filtered: 14.400000
2024-03-06 21:59:14.500797 INFO::Total filtered features: 316
2024-03-06 21:59:14.501094 INFO::Filtered feature names from abundance and prevalence filtering: ASV5, ASV11, ASV12, ASV13, ASV17, ASV20, ASV21, ASV22, ASV23, ASV24, ASV30, ASV31, ASV34, ASV35, ASV38, ASV47, ASV48, ASV50, ASV51, ASV53, ASV57, ASV59, ASV63, ASV68, ASV69, ASV71, ASV73, ASV88, ASV95, ASV97, ASV104, ASV115, ASV119, ASV125, ASV128, ASV131, ASV136, ASV140, ASV147, ASV148, ASV151, ASV152, ASV154, ASV155, ASV170, ASV172, ASV174, ASV183, ASV191, ASV196, ASV197, ASV200, ASV201, ASV202, ASV204, ASV207, ASV208, ASV209, ASV218, ASV224, ASV225, ASV228, ASV249, ASV254, ASV255, ASV256, ASV259, ASV264, ASV267, ASV276, ASV277, ASV280, ASV284, ASV289, ASV291, ASV306, ASV307, ASV312, ASV315, ASV319, ASV325, ASV331, ASV347, ASV349, ASV353, ASV356, ASV358, ASV362, ASV363, ASV367, ASV371, ASV375, ASV378, ASV384, ASV390, ASV392, ASV395, ASV398, ASV405, ASV407, ASV410, ASV414, ASV417, ASV421, ASV431, ASV432, ASV440, ASV443, ASV455, ASV459, ASV466, ASV470, ASV473, ASV475, ASV483, ASV494, ASV518, ASV526, ASV532, ASV556, ASV557, ASV558, ASV570, ASV571, ASV573, ASV586, ASV601, ASV605, ASV608, ASV610, ASV612, ASV615, ASV618, ASV628, ASV637, ASV641, ASV650, ASV653, ASV660, ASV671, ASV674, ASV683, ASV685, ASV686, ASV690, ASV693, ASV699, ASV709, ASV711, ASV728, ASV734, ASV740, ASV744, ASV760, ASV764, ASV785, ASV791, ASV793, ASV803, ASV808, ASV821, ASV823, ASV824, ASV829, ASV838, ASV840, ASV849, ASV853, ASV855, ASV860, ASV864, ASV881, ASV891, ASV892, ASV913, ASV918, ASV925, ASV927, ASV936, ASV948, ASV956, ASV967, ASV979, ASV980, ASV984, ASV1000, ASV1004, ASV1009, ASV1011, ASV1038, ASV1039, ASV1040, ASV1044, ASV1050, ASV1055, ASV1068, ASV1069, ASV1070, ASV1083, ASV1087, ASV1088, ASV1089, ASV1103, ASV1109, ASV1111, ASV1115, ASV1117, ASV1121, ASV1123, ASV1132, ASV1134, ASV1136, ASV1145, ASV1172, ASV1176, ASV1194, ASV1196, ASV1197, ASV1200, ASV1201, ASV1203, ASV1221, ASV1222, ASV1235, ASV1236, ASV1238, ASV1242, ASV1249, ASV1251, ASV1253, ASV1255, ASV1260, ASV1275, ASV1282, ASV1284, ASV1285, ASV1286, ASV1291, ASV1304, ASV1312, ASV1316, ASV1317, ASV1321, ASV1322, ASV1331, ASV1344, ASV1361, ASV1382, ASV1383, ASV1412, ASV1414, ASV1426, ASV1428, ASV1438, ASV1439, ASV1440, ASV1483, ASV1489, ASV1499, ASV1501, ASV1503, ASV1517, ASV1520, ASV1534, ASV1535, ASV1536, ASV1537, ASV1539, ASV1550, ASV1565, ASV1601, ASV1602, ASV1615, ASV1617, ASV1624, ASV1627, ASV1647, ASV1669, ASV1672, ASV1724, ASV1732, ASV1734, ASV1767, ASV1769, ASV1774, ASV1778, ASV1783, ASV1812, ASV1857, ASV1859, ASV1862, ASV1869, ASV1887, ASV1926, ASV1935, ASV1980, ASV2008, ASV2029, ASV2033, ASV2047, ASV2071, ASV2087, ASV2119, ASV2159, ASV2214, ASV2267, ASV2269, ASV2301, ASV2322, ASV2328, ASV2329, ASV2331, ASV2387, ASV2395, ASV2396, ASV2480
2024-03-06 21:59:14.501549 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:59:14.501786 INFO::Filtered feature names from variance filtering:
2024-03-06 21:59:14.501996 INFO::Running selected normalization method: TSS
2024-03-06 21:59:14.502395 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:59:14.504611 INFO::Running selected transform method: LOG
2024-03-06 21:59:14.505025 INFO::Running selected analysis method: LM
2024-03-06 21:59:14.505303 INFO::Fitting model to feature number 1, ASV1
2024-03-06 21:59:14.506531 INFO::Fitting model to feature number 2, ASV3
2024-03-06 21:59:14.507575 INFO::Fitting model to feature number 3, ASV4
2024-03-06 21:59:14.508584 INFO::Fitting model to feature number 4, ASV6
2024-03-06 21:59:14.51144 INFO::Counting total values for each feature
2024-03-06 21:59:14.512103 INFO::Writing filtered data to file results/maaslin2/genus_SexuallyActive/features/filtered_data.tsv
2024-03-06 21:59:14.513357 INFO::Writing filtered, normalized data to file results/maaslin2/genus_SexuallyActive/features/filtered_data_norm.tsv
2024-03-06 21:59:14.514147 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/genus_SexuallyActive/features/filtered_data_norm_transformed.tsv
2024-03-06 21:59:14.514957 WARNING::Deleting existing residuals file: results/maaslin2/genus_SexuallyActive/fits/residuals.rds
2024-03-06 21:59:14.515586 INFO::Writing residuals to file results/maaslin2/genus_SexuallyActive/fits/residuals.rds
2024-03-06 21:59:14.516148 WARNING::Deleting existing fitted file: results/maaslin2/genus_SexuallyActive/fits/fitted.rds
2024-03-06 21:59:14.516557 INFO::Writing fitted values to file results/maaslin2/genus_SexuallyActive/fits/fitted.rds
2024-03-06 21:59:14.516933 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/genus_SexuallyActive/all_results.tsv
2024-03-06 21:59:14.517547 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/genus_SexuallyActive/significant_results.tsv
2024-03-06 21:59:14.517966 INFO::Writing heatmap of significant results to file: results/maaslin2/genus_SexuallyActive/heatmap.pdf
[1] "There is not enough metadata in the associations to create a heatmap plot. Please review the associations in text output file."
2024-03-06 21:59:14.518514 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/genus_SexuallyActive
2024-03-06 21:59:14.51911 INFO::Plotting associations from most to least significant, grouped by metadata
2024-03-06 21:59:14.519408 INFO::Plotting data for metadata number 1, SexuallyActive
2024-03-06 21:59:14.520303 INFO::Creating boxplot for categorical data, SexuallyActive vs ASV3</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:14.578045 INFO::Creating boxplot for categorical data, SexuallyActive vs ASV6</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_rel_genus_sexuallyactive) <span class="sc">%&gt;%</span></span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ASV"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ASV <span class="sc">%in%</span> <span class="fu">filter</span>(genus_SexuallyActive<span class="sc">$</span>results, qval <span class="sc">&lt;</span> <span class="fl">0.2</span>)<span class="sc">$</span>feature)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   ASV  Kingdom       Phylum       Class           Order           Family
1 ASV3 Bacteria Bacteroidota Bacteroidia   Bacteroidales   Prevotellaceae
2 ASV6 Bacteria   Firmicutes     Bacilli Lactobacillales Streptococcaceae
          Genus Species
1    Prevotella    &lt;NA&gt;
2 Streptococcus    &lt;NA&gt;</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot sexuallyActive vs these 2 genera </span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>psmelt_rel_genus_sexuallyactive <span class="ot">=</span> <span class="fu">psmelt</span>(ps_rel_genus_sexuallyactive)</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>maaslin_res_genus_sexactive <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">filter</span>(psmelt_rel_genus_sexuallyactive, OTU <span class="sc">%in%</span> <span class="fu">filter</span>(genus_SexuallyActive<span class="sc">$</span>results, qval <span class="sc">&lt;</span> <span class="fl">0.2</span>)<span class="sc">$</span>feature),</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">aes</span>(<span class="at">x =</span> SexuallyActive, <span class="at">y =</span> Abundance,<span class="at">fill =</span> SexuallyActive)) <span class="sc">+</span></span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>() <span class="sc">+</span></span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Genus, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="cf">function</span>(x){<span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fl">0.1</span>, x))}) <span class="sc">+</span></span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb105-10"><a href="#cb105-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sexually Active"</span>,</span>
<span id="cb105-11"><a href="#cb105-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Genus Abundance (%)"</span>,</span>
<span id="cb105-12"><a href="#cb105-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Maaslin2 results"</span>)<span class="sc">+</span></span>
<span id="cb105-13"><a href="#cb105-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.2</span>))) <span class="sc">+</span></span>
<span id="cb105-14"><a href="#cb105-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"lightgrey"</span>))</span>
<span id="cb105-15"><a href="#cb105-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-16"><a href="#cb105-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-17"><a href="#cb105-17" aria-hidden="true" tabindex="-1"></a>maaslin_res_genus_sexactive</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/sexual%20activity-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(maaslin_res_genus_sexactive, <span class="at">filename =</span> <span class="st">"results/maaslin2/genus_SexuallyActive_panelplot.pdf"</span>,</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">3</span>, <span class="at">width =</span> <span class="dv">4</span>)</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Via Deseq2</span></span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>sexactive_ds <span class="ot">=</span> <span class="fu">phyloseq_to_deseq2</span>(ps_abs_genus_sexuallyactive, <span class="sc">~</span>SexuallyActive)</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a>cts <span class="ot">=</span> <span class="fu">counts</span>(sexactive_ds)</span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a>geoMeans <span class="ot">=</span> <span class="fu">apply</span>(cts, <span class="dv">1</span>, <span class="cf">function</span>(row) <span class="cf">if</span> (<span class="fu">all</span>(row <span class="sc">==</span> <span class="dv">0</span>)) <span class="dv">0</span> <span class="cf">else</span> <span class="fu">exp</span>(<span class="fu">mean</span>(<span class="fu">log</span>(row[row <span class="sc">!=</span> <span class="dv">0</span>]))))</span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>sexactive_ds <span class="ot">=</span> <span class="fu">estimateSizeFactors</span>(sexactive_ds, <span class="at">geoMeans=</span>geoMeans)</span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a>sexactive_ds <span class="ot">=</span> <span class="fu">DESeq</span>(sexactive_ds, <span class="at">test =</span> <span class="st">"Wald"</span>, <span class="at">fitType =</span> <span class="st">"parametric"</span>)</span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">results</span>(sexactive_ds, <span class="at">cooksCutoff =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>alpha <span class="ot">=</span> <span class="fl">0.1</span></span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a>sigtab <span class="ot">=</span> res[<span class="fu">which</span>(res<span class="sc">$</span>padj <span class="sc">&lt;</span> alpha), ]</span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a>sigtab <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">as</span>(sigtab, <span class="st">"data.frame"</span>), <span class="fu">as</span>(<span class="fu">tax_table</span>(ps_abs_genus_sexuallyactive)[<span class="fu">rownames</span>(sigtab), ], <span class="st">"matrix"</span>))</span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a>sigtab <span class="ot">=</span> <span class="fu">as.data.frame</span>(sigtab) <span class="sc">%&gt;%</span></span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"feature"</span>)</span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a>sigtab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>  feature baseMean log2FoldChange    lfcSE      stat       pvalue      padj
1  ASV119 38.25813      -4.983385 1.344519 -3.706444 0.0002101895 0.0540187
   Kingdom     Phylum   Class           Order           Family Genus Species
1 Bacteria Firmicutes Bacilli Lactobacillales Lactobacillaceae HT002    &lt;NA&gt;</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Wilcoxon</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>psmelt_rel_genus_sexuallyactive <span class="sc">%&gt;%</span></span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(OTU) <span class="sc">%&gt;%</span></span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">t_test</span>(<span class="at">data =</span> ., Abundance<span class="sc">~</span>SexuallyActive) <span class="sc">%&gt;%</span></span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">adjust_pvalue</span>(<span class="at">method =</span> <span class="st">"fdr"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p.adj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 0 × 10
# ℹ 10 variables: OTU &lt;chr&gt;, .y. &lt;chr&gt;, group1 &lt;chr&gt;, group2 &lt;chr&gt;, n1 &lt;int&gt;,
#   n2 &lt;int&gt;, statistic &lt;dbl&gt;, df &lt;dbl&gt;, p &lt;dbl&gt;, p.adj &lt;dbl&gt;</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="co"># using simple parametric statistical test (t-test) no significant diff.</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a><span class="co"># generate a plot of HT002 abundance per group</span></span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>data_rel <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(ps_abs_genus_sexuallyactive, <span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x)<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>data_psmelt <span class="ot">=</span> <span class="fu">psmelt</span>(data_rel)</span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a><span class="fu">filter</span>(data_psmelt, OTU <span class="sc">%in%</span> sigtab<span class="sc">$</span>feature) <span class="sc">%&gt;%</span></span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> grp, <span class="at">y =</span> Abundance, <span class="at">fill =</span> SexuallyActive)) <span class="sc">+</span></span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>()) <span class="sc">+</span></span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 44 rows containing non-finite values (`stat_boxplot()`).</code></pre>
</div>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/sexual%20activity-2.pdf" class="img-fluid"></p>
</div>
</div>
<p>Although Maaslin2 detects significant changes in some taxa, this might not be statistically relevant due to the extremely skewed distribution of data and non-suitability of a linear model for comparison of 2 groups. Likewise, using a simple parametric statistical test such as the t-test is sub-optimal. This approach fails to detect any significantly altered taxa.</p>
<p>However, Genus <em>HT002</em> of family Lactobacillaceae reaches borderline significance (q = 0.054). Some studies use a p-value cutoff of q &lt; 0.01 for these analyses. However, this might be counfounded by all patients in the premenopausal group being sexually active (i.e.&nbsp;association with premenopausal roup per se and not with seuxal activity) Next, we analyse and describe the taxonomic composition within the <em>trans-men</em> and control groups by sexual activity. There are no significant statistical differences and the sample size is small. Therefore, the top 10 abundant taxa are listed for sexually active and non-active subgroups:</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Trans Group taxa selection</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>ps_rel_genus_sexuallyactive_trans <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_rel_genus_sexuallyactive, grp <span class="sc">==</span><span class="st">"03_trans-men"</span>)</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>top <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">taxa_sums</span>(ps_rel_genus_sexuallyactive_trans), <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>psmelt_rel_genus_sexuallyactive_trans.top <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(top, ps_rel_genus_sexuallyactive_trans) <span class="sc">%&gt;%</span></span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">psmelt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">tax =</span> <span class="fu">paste</span>(Phylum, Class, Order, Family, Genus,<span class="at">sep =</span> <span class="st">"|"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(tax, SexuallyActive)</span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>genus_sexactive_trans_top20 <span class="ot">=</span> <span class="fu">ggplot</span>(psmelt_rel_genus_sexuallyactive_trans.top, <span class="fu">aes</span>(<span class="at">x =</span> tax, <span class="at">y =</span> Abundance, <span class="at">fill =</span> SexuallyActive)) <span class="sc">+</span></span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position=</span><span class="st">"dodge"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> SexuallyActive), <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.7</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"lightgrey"</span>)) <span class="sc">+</span></span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"lightgrey"</span>)) <span class="sc">+</span></span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lims</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span> </span>
<span id="cb113-18"><a href="#cb113-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Trans-men group"</span>, <span class="at">y =</span> <span class="st">"Abundance (%)"</span>, <span class="at">x =</span> <span class="st">""</span>, <span class="at">color =</span> <span class="st">"Sexually Active"</span>, <span class="at">fill =</span> <span class="st">"Sexually Active"</span>)</span>
<span id="cb113-19"><a href="#cb113-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb113-20"><a href="#cb113-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize</span></span>
<span id="cb113-21"><a href="#cb113-21" aria-hidden="true" tabindex="-1"></a>Genus_Sexuallyactive_Trans_top20 <span class="ot">=</span> psmelt_rel_genus_sexuallyactive_trans.top <span class="sc">%&gt;%</span> </span>
<span id="cb113-22"><a href="#cb113-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean =</span> <span class="fu">mean</span>(Abundance, <span class="at">na.rm=</span>T),</span>
<span id="cb113-23"><a href="#cb113-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">median =</span> <span class="fu">median</span>(Abundance, <span class="at">na.rm =</span>T),</span>
<span id="cb113-24"><a href="#cb113-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">sd =</span> <span class="fu">sd</span>(Abundance, <span class="at">na.rm=</span>T),</span>
<span id="cb113-25"><a href="#cb113-25" aria-hidden="true" tabindex="-1"></a>           <span class="at">IQR =</span> <span class="fu">IQR</span>(Abundance ,<span class="at">na.rm=</span>T))</span>
<span id="cb113-26"><a href="#cb113-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-27"><a href="#cb113-27" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(Genus_Sexuallyactive_Trans_top20, <span class="at">digits =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 74%">
<col style="width: 9%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 3%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">tax</th>
<th style="text-align: left;">SexuallyActive</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">median</th>
<th style="text-align: right;">sd</th>
<th style="text-align: right;">IQR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Actinomycetales|Actinomycetaceae|Actinomyces</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.89</td>
<td style="text-align: right;">0.22</td>
<td style="text-align: right;">1.58</td>
<td style="text-align: right;">0.78</td>
</tr>
<tr class="even">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Actinomycetales|Actinomycetaceae|Actinomyces</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.66</td>
<td style="text-align: right;">0.24</td>
<td style="text-align: right;">1.09</td>
<td style="text-align: right;">0.60</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Bifidobacteriales|Bifidobacteriaceae|Alloscardovia</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1.78</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">5.92</td>
<td style="text-align: right;">0.12</td>
</tr>
<tr class="even">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Bifidobacteriales|Bifidobacteriaceae|Alloscardovia</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.52</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.77</td>
<td style="text-align: right;">0.82</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Bifidobacteriales|Bifidobacteriaceae|Gardnerella</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">16.16</td>
<td style="text-align: right;">3.80</td>
<td style="text-align: right;">21.54</td>
<td style="text-align: right;">28.99</td>
</tr>
<tr class="even">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Bifidobacteriales|Bifidobacteriaceae|Gardnerella</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">7.75</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">14.65</td>
<td style="text-align: right;">4.95</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Corynebacteriales|Corynebacteriaceae|Corynebacterium</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1.98</td>
<td style="text-align: right;">0.32</td>
<td style="text-align: right;">3.48</td>
<td style="text-align: right;">1.38</td>
</tr>
<tr class="even">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Corynebacteriales|Corynebacteriaceae|Corynebacterium</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">3.01</td>
<td style="text-align: right;">2.06</td>
<td style="text-align: right;">2.84</td>
<td style="text-align: right;">5.17</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Bacteroidales|Porphyromonadaceae|Porphyromonas</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">4.02</td>
<td style="text-align: right;">2.37</td>
<td style="text-align: right;">6.20</td>
<td style="text-align: right;">3.88</td>
</tr>
<tr class="even">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Bacteroidales|Porphyromonadaceae|Porphyromonas</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">5.72</td>
<td style="text-align: right;">2.18</td>
<td style="text-align: right;">11.20</td>
<td style="text-align: right;">2.15</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Bacteroidales|Prevotellaceae|Prevotella</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">25.68</td>
<td style="text-align: right;">24.59</td>
<td style="text-align: right;">19.97</td>
<td style="text-align: right;">23.16</td>
</tr>
<tr class="even">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Bacteroidales|Prevotellaceae|Prevotella</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">23.56</td>
<td style="text-align: right;">24.81</td>
<td style="text-align: right;">13.84</td>
<td style="text-align: right;">21.41</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Flavobacteriales|Flavobacteriaceae|Flavobacterium</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="even">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Flavobacteriales|Flavobacteriaceae|Flavobacterium</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">4.37</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">13.82</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Campylobacterota|Campylobacteria|Campylobacterales|Campylobacteraceae|Campylobacter</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.68</td>
<td style="text-align: right;">0.28</td>
<td style="text-align: right;">0.88</td>
<td style="text-align: right;">1.41</td>
</tr>
<tr class="even">
<td style="text-align: left;">Campylobacterota|Campylobacteria|Campylobacterales|Campylobacteraceae|Campylobacter</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">2.03</td>
<td style="text-align: right;">0.83</td>
<td style="text-align: right;">3.72</td>
<td style="text-align: right;">0.87</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Bacilli|Lactobacillales|Lactobacillaceae|Lactobacillus</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">4.85</td>
<td style="text-align: right;">0.25</td>
<td style="text-align: right;">7.10</td>
<td style="text-align: right;">7.75</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Bacilli|Lactobacillales|Lactobacillaceae|Lactobacillus</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">8.24</td>
<td style="text-align: right;">0.24</td>
<td style="text-align: right;">16.71</td>
<td style="text-align: right;">7.14</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Bacilli|Lactobacillales|Streptococcaceae|Streptococcus</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">9.01</td>
<td style="text-align: right;">1.30</td>
<td style="text-align: right;">19.43</td>
<td style="text-align: right;">6.11</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Bacilli|Lactobacillales|Streptococcaceae|Streptococcus</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">2.50</td>
<td style="text-align: right;">0.80</td>
<td style="text-align: right;">3.36</td>
<td style="text-align: right;">5.32</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Anaerococcus</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">3.35</td>
<td style="text-align: right;">2.51</td>
<td style="text-align: right;">2.84</td>
<td style="text-align: right;">3.61</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Anaerococcus</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">6.36</td>
<td style="text-align: right;">4.88</td>
<td style="text-align: right;">6.74</td>
<td style="text-align: right;">5.60</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Fenollaria</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1.61</td>
<td style="text-align: right;">0.45</td>
<td style="text-align: right;">2.40</td>
<td style="text-align: right;">2.10</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Fenollaria</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">2.65</td>
<td style="text-align: right;">2.64</td>
<td style="text-align: right;">2.24</td>
<td style="text-align: right;">3.37</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Finegoldia</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.91</td>
<td style="text-align: right;">0.52</td>
<td style="text-align: right;">1.13</td>
<td style="text-align: right;">0.63</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Finegoldia</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">1.17</td>
<td style="text-align: right;">0.65</td>
<td style="text-align: right;">1.62</td>
<td style="text-align: right;">0.76</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Peptoniphilus</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">2.38</td>
<td style="text-align: right;">2.26</td>
<td style="text-align: right;">2.26</td>
<td style="text-align: right;">2.56</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Peptoniphilus</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">3.06</td>
<td style="text-align: right;">3.01</td>
<td style="text-align: right;">1.97</td>
<td style="text-align: right;">3.48</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Negativicutes|Veillonellales-Selenomonadales|Veillonellaceae|Dialister</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">4.49</td>
<td style="text-align: right;">2.69</td>
<td style="text-align: right;">4.52</td>
<td style="text-align: right;">5.67</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Negativicutes|Veillonellales-Selenomonadales|Veillonellaceae|Dialister</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">7.70</td>
<td style="text-align: right;">6.00</td>
<td style="text-align: right;">8.07</td>
<td style="text-align: right;">7.75</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Negativicutes|Veillonellales-Selenomonadales|Veillonellaceae|Veillonella</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1.55</td>
<td style="text-align: right;">0.26</td>
<td style="text-align: right;">2.90</td>
<td style="text-align: right;">1.68</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Negativicutes|Veillonellales-Selenomonadales|Veillonellaceae|Veillonella</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.18</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.38</td>
<td style="text-align: right;">0.21</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fusobacteriota|Fusobacteriia|Fusobacteriales|Fusobacteriaceae|Fusobacterium</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1.32</td>
<td style="text-align: right;">0.32</td>
<td style="text-align: right;">1.72</td>
<td style="text-align: right;">2.01</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fusobacteriota|Fusobacteriia|Fusobacteriales|Fusobacteriaceae|Fusobacterium</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">3.07</td>
<td style="text-align: right;">0.11</td>
<td style="text-align: right;">7.55</td>
<td style="text-align: right;">1.56</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fusobacteriota|Fusobacteriia|Fusobacteriales|Leptotrichiaceae|Sneathia</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">4.28</td>
<td style="text-align: right;">0.12</td>
<td style="text-align: right;">9.65</td>
<td style="text-align: right;">3.56</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fusobacteriota|Fusobacteriia|Fusobacteriales|Leptotrichiaceae|Sneathia</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.79</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">1.71</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Proteobacteria|Alphaproteobacteria|Rhizobiales|Beijerinckiaceae|Methylobacterium-Methylorubrum</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="even">
<td style="text-align: left;">Proteobacteria|Alphaproteobacteria|Rhizobiales|Beijerinckiaceae|Methylobacterium-Methylorubrum</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">2.92</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">8.26</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Proteobacteria|Gammaproteobacteria|Burkholderiales|Neisseriaceae|Neisseria</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">3.13</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">11.12</td>
<td style="text-align: right;">0.06</td>
</tr>
<tr class="even">
<td style="text-align: left;">Proteobacteria|Gammaproteobacteria|Burkholderiales|Neisseriaceae|Neisseria</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.07</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.19</td>
<td style="text-align: right;">0.01</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Postmenopausal group</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>ps_rel_genus_sexuallyactive_post <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_rel_genus_sexuallyactive, grp <span class="sc">==</span><span class="st">"01_postmenopausal"</span>)</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>top2 <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">taxa_sums</span>(ps_rel_genus_sexuallyactive_post), <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>psmelt_rel_genus_sexuallyactive_post.top <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(top2, ps_rel_genus_sexuallyactive_post) <span class="sc">%&gt;%</span></span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">psmelt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">tax =</span> <span class="fu">paste</span>(Phylum, Class, Order, Family, Genus,<span class="at">sep =</span> <span class="st">"|"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(tax, SexuallyActive)</span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>genus_sexactive_post_top20 <span class="ot">=</span> <span class="fu">ggplot</span>(psmelt_rel_genus_sexuallyactive_post.top, <span class="fu">aes</span>(<span class="at">x =</span> tax, <span class="at">y =</span> Abundance, <span class="at">fill =</span> SexuallyActive)) <span class="sc">+</span></span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position=</span><span class="st">"dodge"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> SexuallyActive), <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.7</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"lightgrey"</span>)) <span class="sc">+</span></span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"lightgrey"</span>)) <span class="sc">+</span></span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lims</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span> </span>
<span id="cb114-18"><a href="#cb114-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Postmenopausal group"</span>, <span class="at">y =</span> <span class="st">"Abundance (%)"</span>, <span class="at">x =</span> <span class="st">""</span>, <span class="at">color =</span> <span class="st">"Sexually Active"</span>, <span class="at">fill =</span> <span class="st">"Sexually Active"</span>)</span>
<span id="cb114-19"><a href="#cb114-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-20"><a href="#cb114-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"results/Genus_SexuallyActive_Descriptive_Postmenopausal.pdf"</span>, <span class="at">plot =</span> genus_sexactive_post_top20,</span>
<span id="cb114-21"><a href="#cb114-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb114-22"><a href="#cb114-22" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"results/Genus_SexuallyActive_Descriptive_Trans.pdf"</span>, <span class="at">plot =</span> genus_sexactive_trans_top20,</span>
<span id="cb114-23"><a href="#cb114-23" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">10</span>)</span>
<span id="cb114-24"><a href="#cb114-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-25"><a href="#cb114-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize</span></span>
<span id="cb114-26"><a href="#cb114-26" aria-hidden="true" tabindex="-1"></a>Genus_Sexuallyactive_Post_top20 <span class="ot">=</span> psmelt_rel_genus_sexuallyactive_post.top <span class="sc">%&gt;%</span> </span>
<span id="cb114-27"><a href="#cb114-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean =</span> <span class="fu">mean</span>(Abundance, <span class="at">na.rm=</span>T),</span>
<span id="cb114-28"><a href="#cb114-28" aria-hidden="true" tabindex="-1"></a>           <span class="at">median =</span> <span class="fu">median</span>(Abundance, <span class="at">na.rm =</span>T),</span>
<span id="cb114-29"><a href="#cb114-29" aria-hidden="true" tabindex="-1"></a>           <span class="at">sd =</span> <span class="fu">sd</span>(Abundance, <span class="at">na.rm=</span>T),</span>
<span id="cb114-30"><a href="#cb114-30" aria-hidden="true" tabindex="-1"></a>           <span class="at">IQR =</span> <span class="fu">IQR</span>(Abundance ,<span class="at">na.rm=</span>T))</span>
<span id="cb114-31"><a href="#cb114-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-32"><a href="#cb114-32" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(Genus_Sexuallyactive_Post_top20, <span class="at">digits =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 73%">
<col style="width: 9%">
<col style="width: 3%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 3%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">tax</th>
<th style="text-align: left;">SexuallyActive</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">median</th>
<th style="text-align: right;">sd</th>
<th style="text-align: right;">IQR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Bifidobacteriales|Bifidobacteriaceae|Alloscardovia</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1.47</td>
<td style="text-align: right;">0.20</td>
<td style="text-align: right;">3.02</td>
<td style="text-align: right;">1.74</td>
</tr>
<tr class="even">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Bifidobacteriales|Bifidobacteriaceae|Alloscardovia</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">2.19</td>
<td style="text-align: right;">0.06</td>
<td style="text-align: right;">6.33</td>
<td style="text-align: right;">0.32</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Bifidobacteriales|Bifidobacteriaceae|Bifidobacterium</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">6.33</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">15.73</td>
<td style="text-align: right;">0.18</td>
</tr>
<tr class="even">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Bifidobacteriales|Bifidobacteriaceae|Bifidobacterium</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">2.35</td>
<td style="text-align: right;">0.09</td>
<td style="text-align: right;">5.76</td>
<td style="text-align: right;">0.26</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Bifidobacteriales|Bifidobacteriaceae|Gardnerella</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">15.79</td>
<td style="text-align: right;">1.17</td>
<td style="text-align: right;">25.04</td>
<td style="text-align: right;">14.28</td>
</tr>
<tr class="even">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Bifidobacteriales|Bifidobacteriaceae|Gardnerella</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">2.56</td>
<td style="text-align: right;">0.26</td>
<td style="text-align: right;">4.73</td>
<td style="text-align: right;">1.79</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Corynebacteriales|Corynebacteriaceae|Corynebacterium</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">0.53</td>
<td style="text-align: right;">0.93</td>
<td style="text-align: right;">0.90</td>
</tr>
<tr class="even">
<td style="text-align: left;">Actinobacteriota|Actinobacteria|Corynebacteriales|Corynebacteriaceae|Corynebacterium</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">5.10</td>
<td style="text-align: right;">1.21</td>
<td style="text-align: right;">7.84</td>
<td style="text-align: right;">5.04</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Actinobacteriota|Coriobacteriia|Coriobacteriales|Atopobiaceae|Atopobium</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">3.96</td>
<td style="text-align: right;">1.33</td>
<td style="text-align: right;">7.27</td>
<td style="text-align: right;">4.11</td>
</tr>
<tr class="even">
<td style="text-align: left;">Actinobacteriota|Coriobacteriia|Coriobacteriales|Atopobiaceae|Atopobium</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">3.46</td>
<td style="text-align: right;">0.23</td>
<td style="text-align: right;">11.00</td>
<td style="text-align: right;">1.47</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Bacteroidales|Porphyromonadaceae|Porphyromonas</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">3.88</td>
<td style="text-align: right;">0.06</td>
<td style="text-align: right;">6.98</td>
<td style="text-align: right;">4.58</td>
</tr>
<tr class="even">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Bacteroidales|Porphyromonadaceae|Porphyromonas</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.48</td>
<td style="text-align: right;">0.14</td>
<td style="text-align: right;">0.83</td>
<td style="text-align: right;">0.50</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Bacteroidales|Prevotellaceae|Prevotella</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">17.77</td>
<td style="text-align: right;">11.21</td>
<td style="text-align: right;">18.03</td>
<td style="text-align: right;">30.23</td>
</tr>
<tr class="even">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Bacteroidales|Prevotellaceae|Prevotella</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">5.30</td>
<td style="text-align: right;">1.58</td>
<td style="text-align: right;">9.00</td>
<td style="text-align: right;">4.29</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Bacteroidales|Prevotellaceae|Prevotella_9</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="even">
<td style="text-align: left;">Bacteroidota|Bacteroidia|Bacteroidales|Prevotellaceae|Prevotella_9</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.98</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">3.15</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Campylobacterota|Campylobacteria|Campylobacterales|Campylobacteraceae|Campylobacter</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1.51</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">2.44</td>
<td style="text-align: right;">2.32</td>
</tr>
<tr class="even">
<td style="text-align: left;">Campylobacterota|Campylobacteria|Campylobacterales|Campylobacteraceae|Campylobacter</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.27</td>
<td style="text-align: right;">0.13</td>
<td style="text-align: right;">0.38</td>
<td style="text-align: right;">0.40</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Bacilli|Lactobacillales|Lactobacillaceae|Lactobacillus</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">23.50</td>
<td style="text-align: right;">6.49</td>
<td style="text-align: right;">33.92</td>
<td style="text-align: right;">36.76</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Bacilli|Lactobacillales|Lactobacillaceae|Lactobacillus</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">30.61</td>
<td style="text-align: right;">10.26</td>
<td style="text-align: right;">36.18</td>
<td style="text-align: right;">55.70</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Bacilli|Lactobacillales|Streptococcaceae|Streptococcus</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">5.67</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">9.56</td>
<td style="text-align: right;">5.90</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Bacilli|Lactobacillales|Streptococcaceae|Streptococcus</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">12.99</td>
<td style="text-align: right;">1.23</td>
<td style="text-align: right;">22.29</td>
<td style="text-align: right;">14.78</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Bacilli|Staphylococcales|Gemellaceae|Gemella</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.06</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.14</td>
<td style="text-align: right;">0.05</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Bacilli|Staphylococcales|Gemellaceae|Gemella</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">1.90</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">4.82</td>
<td style="text-align: right;">0.34</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Bacilli|Staphylococcales|Staphylococcaceae|Staphylococcus</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.07</td>
<td style="text-align: right;">0.06</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">0.10</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Bacilli|Staphylococcales|Staphylococcaceae|Staphylococcus</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">9.34</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">24.53</td>
<td style="text-align: right;">0.22</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Anaerococcus</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">3.43</td>
<td style="text-align: right;">2.06</td>
<td style="text-align: right;">5.29</td>
<td style="text-align: right;">2.71</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Anaerococcus</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">2.74</td>
<td style="text-align: right;">0.81</td>
<td style="text-align: right;">4.63</td>
<td style="text-align: right;">2.72</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Ezakiella</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1.09</td>
<td style="text-align: right;">0.11</td>
<td style="text-align: right;">1.86</td>
<td style="text-align: right;">1.47</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Ezakiella</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.28</td>
<td style="text-align: right;">0.08</td>
<td style="text-align: right;">0.51</td>
<td style="text-align: right;">0.33</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Finegoldia</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">2.50</td>
<td style="text-align: right;">0.69</td>
<td style="text-align: right;">3.95</td>
<td style="text-align: right;">2.03</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Finegoldia</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">3.65</td>
<td style="text-align: right;">0.58</td>
<td style="text-align: right;">5.73</td>
<td style="text-align: right;">3.52</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Peptoniphilus</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">2.40</td>
<td style="text-align: right;">1.56</td>
<td style="text-align: right;">3.03</td>
<td style="text-align: right;">3.27</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Clostridia|Peptostreptococcales-Tissierellales|Family XI|Peptoniphilus</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">1.50</td>
<td style="text-align: right;">0.61</td>
<td style="text-align: right;">2.08</td>
<td style="text-align: right;">1.35</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Firmicutes|Negativicutes|Veillonellales-Selenomonadales|Veillonellaceae|Dialister</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1.99</td>
<td style="text-align: right;">0.07</td>
<td style="text-align: right;">3.06</td>
<td style="text-align: right;">4.13</td>
</tr>
<tr class="even">
<td style="text-align: left;">Firmicutes|Negativicutes|Veillonellales-Selenomonadales|Veillonellaceae|Dialister</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.55</td>
<td style="text-align: right;">0.36</td>
<td style="text-align: right;">0.59</td>
<td style="text-align: right;">0.84</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Fusobacteriota|Fusobacteriia|Fusobacteriales|Fusobacteriaceae|Fusobacterium</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">1.13</td>
<td style="text-align: right;">0.06</td>
<td style="text-align: right;">2.23</td>
<td style="text-align: right;">0.11</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fusobacteriota|Fusobacteriia|Fusobacteriales|Fusobacteriaceae|Fusobacterium</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">0.29</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.54</td>
<td style="text-align: right;">0.29</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Proteobacteria|Gammaproteobacteria|Enterobacterales|Enterobacteriaceae|Escherichia-Shigella</td>
<td style="text-align: left;">yes</td>
<td style="text-align: right;">0.33</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">0.83</td>
<td style="text-align: right;">0.05</td>
</tr>
<tr class="even">
<td style="text-align: left;">Proteobacteria|Gammaproteobacteria|Enterobacterales|Enterobacteriaceae|Escherichia-Shigella</td>
<td style="text-align: left;">no</td>
<td style="text-align: right;">5.49</td>
<td style="text-align: right;">0.00</td>
<td style="text-align: right;">16.61</td>
<td style="text-align: right;">1.95</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>genus_sexactive_trans_top20</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/sexactive_toptaxa-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>genus_sexactive_post_top20</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/sexactive_toptaxa-2.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(Genus_Sexuallyactive_Post_top20, <span class="at">file =</span> <span class="st">"results/Genus_SexuallyActive_Descriptive_Postmenopausal.csv"</span>)</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(Genus_Sexuallyactive_Trans_top20, <span class="at">file =</span> <span class="st">"results/Genus_SexuallyActive_Descriptive_Trans.csv"</span>)</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Analysis in Trans group with male sexual partner only</span></span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>genus_Sexofpartner <span class="ot">=</span> psmelt_rel_genus_sexuallyactive_trans.top <span class="sc">%&gt;%</span></span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(SexOfPartner)) <span class="sc">%&gt;%</span></span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SexOfPartner2 =</span> <span class="fu">ifelse</span>(SexOfPartner <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"M"</span>, <span class="st">"M+F"</span>), <span class="st">"M"</span>, <span class="st">"F only"</span>))</span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>genus_sexactive_trans_malepartner_top20 <span class="ot">=</span> <span class="fu">ggplot</span>(genus_Sexofpartner, </span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>                                                 <span class="fu">aes</span>(<span class="at">x =</span> tax, <span class="at">y =</span> Abundance, <span class="at">fill =</span> SexuallyActive)) <span class="sc">+</span></span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">position=</span><span class="st">"dodge"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> SexuallyActive), <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.7</span>), <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"lightgrey"</span>)) <span class="sc">+</span></span>
<span id="cb117-13"><a href="#cb117-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"lightgrey"</span>)) <span class="sc">+</span></span>
<span id="cb117-14"><a href="#cb117-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb117-15"><a href="#cb117-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb117-16"><a href="#cb117-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>SexOfPartner2) <span class="sc">+</span></span>
<span id="cb117-17"><a href="#cb117-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lims</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">+</span> </span>
<span id="cb117-18"><a href="#cb117-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Trans-men group"</span>, <span class="at">y =</span> <span class="st">"Abundance (%)"</span>, <span class="at">x =</span> <span class="st">""</span>, <span class="at">color =</span> <span class="st">"Sexually Active"</span>, <span class="at">fill =</span> <span class="st">"Sexually Active"</span>)</span>
<span id="cb117-19"><a href="#cb117-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb117-20"><a href="#cb117-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"results/Genus_SexuallyActive_Descriptive_Trans_SexOfPartner.pdf"</span>,</span>
<span id="cb117-21"><a href="#cb117-21" aria-hidden="true" tabindex="-1"></a>       genus_sexactive_trans_malepartner_top20,</span>
<span id="cb117-22"><a href="#cb117-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">8</span>, <span class="at">width =</span> <span class="dv">11</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="influence-of-the-duration-of-amenorrhea-on-microbiome-composition" class="level3">
<h3 class="anchored" data-anchor-id="influence-of-the-duration-of-amenorrhea-on-microbiome-composition">Influence of the duration of amenorrhea on microbiome composition</h3>
<p>Both postmenopausal women as well as trans men undergoing GAHT experience amenorrhea. This section explores a potential interaction between time since last menstruation and composition of the vaginal microbiome.</p>
<p><strong>This analysis is performed within each group separately and results are compared.</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="co"># generate relevant phyloseq object</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>ps_menopause <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_samples, grp <span class="sc">!=</span><span class="st">"02_premenopausal"</span>)</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(ps_menopause)<span class="sc">$</span>DurationMenopause <span class="co"># this is the duration of menopause in postmenopausal women (years)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] 11  2 10 22 12 22  4 28  7 26 11  2 18 20 17 24 23 19 23 27 13  1 13  4  3
[26] NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(ps_menopause)<span class="sc">$</span>DurationAmenorrhea <span class="co"># duration of amenorrhea in trans group (months)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> [1] NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA
[26] 19 42  2 33 17 35  6  2 37 14  0 60 46  4 26  9 16 27  0 74  9  1 35 16 18</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(ps_menopause)<span class="sc">$</span>DurationMenopauseMonths <span class="ot">=</span> <span class="fu">sample_data</span>(ps_menopause)<span class="sc">$</span>DurationMenopause<span class="sc">*</span><span class="dv">12</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(ps_menopause)<span class="sc">$</span>DurationCombined <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(<span class="fu">sample_data</span>(ps_menopause)<span class="sc">$</span>DurationMenopauseMonths), <span class="fu">sample_data</span>(ps_menopause)<span class="sc">$</span>DurationAmenorrhea, <span class="fu">sample_data</span>(ps_menopause)<span class="sc">$</span>DurationMenopauseMonths)</span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>ps_menopause_rel <span class="ot">=</span>  <span class="fu">transform_sample_counts</span>(ps_menopause,<span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x) <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Diversity analysis</span></span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>adiv_menopause <span class="ot">=</span>  <span class="fu">plot_richness</span>(ps_menopause, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">"Observed"</span>, <span class="st">"Shannon"</span>, <span class="st">"InvSimpson"</span>))</span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a>adiv_menopause <span class="ot">=</span> adiv_menopause<span class="sc">$</span>data <span class="sc">%&gt;%</span></span>
<span id="cb122-9"><a href="#cb122-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(samples, grp, DurationMenopause, DurationAmenorrhea, DurationCombined, variable, value)</span>
<span id="cb122-10"><a href="#cb122-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-11"><a href="#cb122-11" aria-hidden="true" tabindex="-1"></a>corr_trans <span class="ot">=</span> <span class="fu">filter</span>(adiv_menopause, grp <span class="sc">==</span><span class="st">"03_trans-men"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-12"><a href="#cb122-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb122-13"><a href="#cb122-13" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">cor_test</span>(value,DurationAmenorrhea)</span>
<span id="cb122-14"><a href="#cb122-14" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(corr_trans)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 12%">
<col style="width: 6%">
<col style="width: 21%">
<col style="width: 7%">
<col style="width: 11%">
<col style="width: 6%">
<col style="width: 12%">
<col style="width: 11%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">variable</th>
<th style="text-align: left;">var1</th>
<th style="text-align: left;">var2</th>
<th style="text-align: right;">cor</th>
<th style="text-align: right;">statistic</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">conf.low</th>
<th style="text-align: right;">conf.high</th>
<th style="text-align: left;">method</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationAmenorrhea</td>
<td style="text-align: right;">0.0840</td>
<td style="text-align: right;">0.4031351</td>
<td style="text-align: right;">0.691</td>
<td style="text-align: right;">-0.3220252</td>
<td style="text-align: right;">0.4635524</td>
<td style="text-align: left;">Pearson</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationAmenorrhea</td>
<td style="text-align: right;">0.0095</td>
<td style="text-align: right;">0.0457909</td>
<td style="text-align: right;">0.964</td>
<td style="text-align: right;">-0.3870434</td>
<td style="text-align: right;">0.4031576</td>
<td style="text-align: left;">Pearson</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationAmenorrhea</td>
<td style="text-align: right;">0.0570</td>
<td style="text-align: right;">0.2738670</td>
<td style="text-align: right;">0.787</td>
<td style="text-align: right;">-0.3459110</td>
<td style="text-align: right;">0.4421820</td>
<td style="text-align: left;">Pearson</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">filter</span>(adiv_menopause, grp <span class="sc">==</span><span class="st">"03_trans-men"</span>), <span class="fu">aes</span>(<span class="at">x =</span> DurationAmenorrhea, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"darkred"</span>) <span class="sc">+</span></span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>variable, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Duration of Amenorrhea in postmenopausal group"</span>, <span class="at">y =</span> <span class="st">"Alpha diversity value"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/duration-menopause-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">filter</span>(adiv_menopause, grp <span class="sc">==</span><span class="st">"01_postmenopausal"</span>), <span class="fu">aes</span>(<span class="at">x =</span> DurationMenopause, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>) <span class="sc">+</span></span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>variable, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Duration of Amenorrhea in trans-men group"</span>, <span class="at">y =</span> <span class="st">"Alpha diversity value"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/duration-menopause-2.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>corr_postmeno <span class="ot">=</span> <span class="fu">filter</span>(adiv_menopause, grp <span class="sc">==</span><span class="st">"01_postmenopausal"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">cor_test</span>(value,DurationMenopause,<span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(corr_postmeno)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 12%">
<col style="width: 7%">
<col style="width: 21%">
<col style="width: 5%">
<col style="width: 11%">
<col style="width: 7%">
<col style="width: 12%">
<col style="width: 11%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">variable</th>
<th style="text-align: left;">var1</th>
<th style="text-align: left;">var2</th>
<th style="text-align: right;">cor</th>
<th style="text-align: right;">statistic</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">conf.low</th>
<th style="text-align: right;">conf.high</th>
<th style="text-align: left;">method</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationMenopause</td>
<td style="text-align: right;">0.18</td>
<td style="text-align: right;">0.8607750</td>
<td style="text-align: right;">0.398</td>
<td style="text-align: right;">-0.2348644</td>
<td style="text-align: right;">0.5344828</td>
<td style="text-align: left;">Pearson</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationMenopause</td>
<td style="text-align: right;">0.11</td>
<td style="text-align: right;">0.5281901</td>
<td style="text-align: right;">0.602</td>
<td style="text-align: right;">-0.2985727</td>
<td style="text-align: right;">0.4836819</td>
<td style="text-align: left;">Pearson</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationMenopause</td>
<td style="text-align: right;">0.16</td>
<td style="text-align: right;">0.7669201</td>
<td style="text-align: right;">0.451</td>
<td style="text-align: right;">-0.2530096</td>
<td style="text-align: right;">0.5205587</td>
<td style="text-align: left;">Pearson</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="co"># correlations are not signficiant for trans and postmenopausal group separately</span></span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Next, both groups are combined: in months</span></span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a>corr_menopause <span class="ot">=</span> adiv_menopause <span class="sc">%&gt;%</span></span>
<span id="cb126-5"><a href="#cb126-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb126-6"><a href="#cb126-6" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">cor_test</span>(value, DurationCombined, <span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb126-7"><a href="#cb126-7" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(corr_menopause)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 12%">
<col style="width: 6%">
<col style="width: 19%">
<col style="width: 6%">
<col style="width: 11%">
<col style="width: 8%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">variable</th>
<th style="text-align: left;">var1</th>
<th style="text-align: left;">var2</th>
<th style="text-align: right;">cor</th>
<th style="text-align: right;">statistic</th>
<th style="text-align: right;">p</th>
<th style="text-align: right;">conf.low</th>
<th style="text-align: right;">conf.high</th>
<th style="text-align: left;">method</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Observed</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationCombined</td>
<td style="text-align: right;">-0.21</td>
<td style="text-align: right;">-1.497145</td>
<td style="text-align: right;">0.1410</td>
<td style="text-align: right;">-0.4623821</td>
<td style="text-align: right;">0.0713220</td>
<td style="text-align: left;">Pearson</td>
</tr>
<tr class="even">
<td style="text-align: left;">Shannon</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationCombined</td>
<td style="text-align: right;">-0.30</td>
<td style="text-align: right;">-2.168592</td>
<td style="text-align: right;">0.0351</td>
<td style="text-align: right;">-0.5327673</td>
<td style="text-align: right;">-0.0222174</td>
<td style="text-align: left;">Pearson</td>
</tr>
<tr class="odd">
<td style="text-align: left;">InvSimpson</td>
<td style="text-align: left;">value</td>
<td style="text-align: left;">DurationCombined</td>
<td style="text-align: right;">-0.14</td>
<td style="text-align: right;">-1.005403</td>
<td style="text-align: right;">0.3200</td>
<td style="text-align: right;">-0.4057416</td>
<td style="text-align: right;">0.1403448</td>
<td style="text-align: left;">Pearson</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>comb_menopause_plot <span class="ot">=</span> <span class="fu">ggplot</span>(adiv_menopause, <span class="fu">aes</span>(<span class="at">x =</span> DurationCombined, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> grp)) <span class="sc">+</span></span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"darkgrey"</span>) <span class="sc">+</span></span>
<span id="cb127-4"><a href="#cb127-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>variable, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb127-5"><a href="#cb127-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Duration of Amenorrhea in postmenopausal and trans-men (months)"</span>, <span class="at">y =</span> <span class="st">"Alpha diversity value"</span>) <span class="sc">+</span> </span>
<span id="cb127-6"><a href="#cb127-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> custom_pal[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)])</span>
<span id="cb127-7"><a href="#cb127-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-8"><a href="#cb127-8" aria-hidden="true" tabindex="-1"></a>comb_menopause_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/duration-menopause-3.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> comb_menopause_plot,<span class="at">device =</span> <span class="st">"pdf"</span>, </span>
<span id="cb128-2"><a href="#cb128-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">filename =</span> <span class="st">"MbVagTrans_Results_files/figure-html/duration-menopause_comb.pdf"</span>, <span class="at">height =</span> <span class="dv">3</span>)</span>
<span id="cb128-3"><a href="#cb128-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-4"><a href="#cb128-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Maaslin2 analysis of abundance with duration of amenorrhea</span></span>
<span id="cb128-5"><a href="#cb128-5" aria-hidden="true" tabindex="-1"></a>ps_menopause_rel_genus <span class="ot">=</span> <span class="fu">tax_glom</span>(ps_menopause_rel, <span class="at">taxrank =</span> <span class="st">"Genus"</span>)</span>
<span id="cb128-6"><a href="#cb128-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-7"><a href="#cb128-7" aria-hidden="true" tabindex="-1"></a>genus_MenoPauseDurationMonths <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_menopause_rel_genus),</span>
<span id="cb128-8"><a href="#cb128-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_menopause_rel_genus), <span class="st">"data.frame"</span>),</span>
<span id="cb128-9"><a href="#cb128-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/genus_MenopausedurationMonths"</span>,</span>
<span id="cb128-10"><a href="#cb128-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">5</span>,</span>
<span id="cb128-11"><a href="#cb128-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.2</span>,</span>
<span id="cb128-12"><a href="#cb128-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb128-13"><a href="#cb128-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"DurationCombined"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:26.123748 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/genus_MenopausedurationMonths/maaslin2.log"
2024-03-06 21:59:26.295576 INFO::Writing function arguments to log file
2024-03-06 21:59:26.298198 INFO::Verifying options selected are valid
2024-03-06 21:59:26.298471 INFO::Determining format of input files
2024-03-06 21:59:26.298712 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:59:26.300638 INFO::Formula for fixed effects: expr ~  DurationCombined
2024-03-06 21:59:26.300915 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:59:26.301144 INFO::Total samples in data: 50
2024-03-06 21:59:26.301363 INFO::Min samples required with min abundance for a feature not to be filtered: 10.000000
2024-03-06 21:59:26.303003 INFO::Total filtered features: 313
2024-03-06 21:59:26.303295 INFO::Filtered feature names from abundance and prevalence filtering: ASV5, ASV11, ASV12, ASV13, ASV20, ASV21, ASV22, ASV23, ASV24, ASV30, ASV34, ASV38, ASV47, ASV48, ASV51, ASV53, ASV57, ASV59, ASV63, ASV68, ASV69, ASV70, ASV71, ASV73, ASV88, ASV95, ASV97, ASV104, ASV115, ASV119, ASV125, ASV128, ASV131, ASV136, ASV140, ASV147, ASV148, ASV151, ASV152, ASV154, ASV155, ASV170, ASV174, ASV183, ASV191, ASV193, ASV196, ASV197, ASV200, ASV201, ASV207, ASV208, ASV209, ASV215, ASV218, ASV224, ASV228, ASV254, ASV255, ASV256, ASV259, ASV264, ASV266, ASV267, ASV276, ASV277, ASV280, ASV284, ASV289, ASV291, ASV306, ASV307, ASV315, ASV319, ASV325, ASV330, ASV331, ASV347, ASV349, ASV353, ASV356, ASV358, ASV362, ASV363, ASV367, ASV371, ASV375, ASV378, ASV384, ASV390, ASV392, ASV395, ASV398, ASV405, ASV407, ASV410, ASV414, ASV417, ASV421, ASV431, ASV432, ASV440, ASV443, ASV459, ASV466, ASV470, ASV475, ASV479, ASV483, ASV494, ASV518, ASV526, ASV532, ASV556, ASV557, ASV558, ASV570, ASV571, ASV573, ASV586, ASV601, ASV605, ASV608, ASV610, ASV612, ASV615, ASV618, ASV637, ASV641, ASV642, ASV653, ASV658, ASV660, ASV667, ASV671, ASV674, ASV683, ASV685, ASV686, ASV690, ASV693, ASV699, ASV709, ASV711, ASV728, ASV734, ASV740, ASV760, ASV764, ASV785, ASV793, ASV801, ASV803, ASV808, ASV821, ASV823, ASV824, ASV829, ASV840, ASV849, ASV853, ASV855, ASV860, ASV863, ASV864, ASV865, ASV881, ASV891, ASV892, ASV903, ASV913, ASV915, ASV918, ASV925, ASV927, ASV936, ASV948, ASV956, ASV967, ASV979, ASV980, ASV984, ASV1000, ASV1004, ASV1009, ASV1011, ASV1038, ASV1039, ASV1050, ASV1055, ASV1068, ASV1069, ASV1070, ASV1083, ASV1088, ASV1089, ASV1103, ASV1109, ASV1111, ASV1117, ASV1121, ASV1123, ASV1132, ASV1134, ASV1136, ASV1145, ASV1151, ASV1160, ASV1172, ASV1176, ASV1194, ASV1196, ASV1197, ASV1200, ASV1201, ASV1203, ASV1221, ASV1222, ASV1235, ASV1236, ASV1238, ASV1242, ASV1249, ASV1251, ASV1253, ASV1255, ASV1260, ASV1275, ASV1282, ASV1284, ASV1285, ASV1291, ASV1293, ASV1304, ASV1312, ASV1313, ASV1314, ASV1316, ASV1317, ASV1321, ASV1322, ASV1344, ASV1353, ASV1382, ASV1383, ASV1412, ASV1414, ASV1438, ASV1439, ASV1440, ASV1476, ASV1483, ASV1489, ASV1499, ASV1501, ASV1503, ASV1517, ASV1520, ASV1534, ASV1535, ASV1536, ASV1537, ASV1539, ASV1550, ASV1565, ASV1601, ASV1602, ASV1615, ASV1616, ASV1617, ASV1624, ASV1627, ASV1647, ASV1672, ASV1724, ASV1732, ASV1734, ASV1767, ASV1769, ASV1774, ASV1778, ASV1783, ASV1812, ASV1857, ASV1859, ASV1862, ASV1869, ASV1887, ASV1926, ASV1935, ASV1980, ASV2008, ASV2029, ASV2033, ASV2047, ASV2071, ASV2087, ASV2119, ASV2149, ASV2159, ASV2214, ASV2267, ASV2269, ASV2301, ASV2322, ASV2323, ASV2328, ASV2329, ASV2331, ASV2387, ASV2395, ASV2396, ASV2480
2024-03-06 21:59:26.303748 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:59:26.303989 INFO::Filtered feature names from variance filtering:
2024-03-06 21:59:26.304201 INFO::Running selected normalization method: TSS
2024-03-06 21:59:26.304618 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:59:26.307128 INFO::Running selected transform method: LOG
2024-03-06 21:59:26.307632 INFO::Running selected analysis method: LM
2024-03-06 21:59:26.307919 INFO::Fitting model to feature number 1, ASV2
2024-03-06 21:59:26.30904 INFO::Fitting model to feature number 2, ASV3
2024-03-06 21:59:26.309984 INFO::Fitting model to feature number 3, ASV7
2024-03-06 21:59:26.310893 INFO::Fitting model to feature number 4, ASV15
2024-03-06 21:59:26.311766 INFO::Fitting model to feature number 5, ASV17
2024-03-06 21:59:26.312645 INFO::Fitting model to feature number 6, ASV31
2024-03-06 21:59:26.313516 INFO::Fitting model to feature number 7, ASV50
2024-03-06 21:59:26.316186 INFO::Counting total values for each feature
2024-03-06 21:59:26.316866 INFO::Writing filtered data to file results/maaslin2/genus_MenopausedurationMonths/features/filtered_data.tsv
2024-03-06 21:59:26.317735 INFO::Writing filtered, normalized data to file results/maaslin2/genus_MenopausedurationMonths/features/filtered_data_norm.tsv
2024-03-06 21:59:26.318518 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/genus_MenopausedurationMonths/features/filtered_data_norm_transformed.tsv
2024-03-06 21:59:26.319341 WARNING::Deleting existing residuals file: results/maaslin2/genus_MenopausedurationMonths/fits/residuals.rds
2024-03-06 21:59:26.319756 INFO::Writing residuals to file results/maaslin2/genus_MenopausedurationMonths/fits/residuals.rds
2024-03-06 21:59:26.320357 WARNING::Deleting existing fitted file: results/maaslin2/genus_MenopausedurationMonths/fits/fitted.rds
2024-03-06 21:59:26.320749 INFO::Writing fitted values to file results/maaslin2/genus_MenopausedurationMonths/fits/fitted.rds
2024-03-06 21:59:26.32115 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/genus_MenopausedurationMonths/all_results.tsv
2024-03-06 21:59:26.321758 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/genus_MenopausedurationMonths/significant_results.tsv
2024-03-06 21:59:26.32216 INFO::Writing heatmap of significant results to file: results/maaslin2/genus_MenopausedurationMonths/heatmap.pdf
[1] "There is not enough metadata in the associations to create a heatmap plot. Please review the associations in text output file."
2024-03-06 21:59:26.322787 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/genus_MenopausedurationMonths
2024-03-06 21:59:26.323391 INFO::Plotting associations from most to least significant, grouped by metadata
2024-03-06 21:59:26.323677 INFO::Plotting data for metadata number 1, DurationCombined
2024-03-06 21:59:26.324578 INFO::Creating scatter plot for continuous data, DurationCombined vs ASV2</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:26.389303 INFO::Creating scatter plot for continuous data, DurationCombined vs ASV3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 2 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:26.462941 INFO::Creating scatter plot for continuous data, DurationCombined vs ASV17</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:26.522916 INFO::Creating scatter plot for continuous data, DurationCombined vs ASV31</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).
Removed 2 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 4 ASVs </span></span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_menopause_rel_genus) <span class="sc">%&gt;%</span></span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb136-5"><a href="#cb136-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ASV"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb136-6"><a href="#cb136-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ASV <span class="sc">%in%</span> <span class="fu">filter</span>(genus_MenoPauseDurationMonths<span class="sc">$</span>results, qval <span class="sc">&lt;</span> <span class="fl">0.2</span>)<span class="sc">$</span>feature)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>    ASV  Kingdom       Phylum         Class                               Order
1  ASV2 Bacteria   Firmicutes       Bacilli                     Lactobacillales
2  ASV3 Bacteria Bacteroidota   Bacteroidia                       Bacteroidales
3 ASV17 Bacteria   Firmicutes Negativicutes      Veillonellales-Selenomonadales
4 ASV31 Bacteria   Firmicutes    Clostridia Peptostreptococcales-Tissierellales
            Family         Genus Species
1 Lactobacillaceae Lactobacillus    &lt;NA&gt;
2   Prevotellaceae    Prevotella    &lt;NA&gt;
3  Veillonellaceae     Dialister    &lt;NA&gt;
4        Family XI  Anaerococcus    &lt;NA&gt;</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Lactobacillus, dialister, Anaerococcus, Prevotella</span></span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot DurationCombined vs these 4 genera </span></span>
<span id="cb138-4"><a href="#cb138-4" aria-hidden="true" tabindex="-1"></a>psmelt_rel_genus_durationComb <span class="ot">=</span> <span class="fu">psmelt</span>(ps_menopause_rel_genus)</span>
<span id="cb138-5"><a href="#cb138-5" aria-hidden="true" tabindex="-1"></a>maaslin_res_genus_durationComb <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">filter</span>(psmelt_rel_genus_durationComb, OTU <span class="sc">%in%</span> <span class="fu">filter</span>(genus_MenoPauseDurationMonths<span class="sc">$</span>results, qval <span class="sc">&lt;</span> <span class="fl">0.2</span>)<span class="sc">$</span>feature),</span>
<span id="cb138-6"><a href="#cb138-6" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">aes</span>(<span class="at">x =</span> DurationCombined, <span class="at">y =</span> Abundance,<span class="at">color =</span> grp)) <span class="sc">+</span></span>
<span id="cb138-7"><a href="#cb138-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb138-8"><a href="#cb138-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Genus, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb138-9"><a href="#cb138-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="cf">function</span>(x){<span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fl">0.1</span>, x))}) <span class="sc">+</span></span>
<span id="cb138-10"><a href="#cb138-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb138-11"><a href="#cb138-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sexually Active"</span>,</span>
<span id="cb138-12"><a href="#cb138-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Genus Abundance (%)"</span>,</span>
<span id="cb138-13"><a href="#cb138-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Maaslin2 results"</span>)</span>
<span id="cb138-14"><a href="#cb138-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-15"><a href="#cb138-15" aria-hidden="true" tabindex="-1"></a>maaslin_res_genus_durationComb</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 7 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/duration-menopause-4.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(maaslin_res_genus_durationComb, <span class="at">filename =</span> <span class="st">"results/maaslin2/genus_DurationMenopauseCombined_panelplot.png"</span>,</span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 5 rows containing missing values (`geom_point()`).</code></pre>
</div>
</div>
</section>
<section id="influence-of-sex-of-partner" class="level2">
<h2 class="anchored" data-anchor-id="influence-of-sex-of-partner">Influence of sex of partner</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Analysis in complete cohort</span></span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a><span class="do">### alpha diversity</span></span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a>rstatix<span class="sc">::</span><span class="fu">shapiro_test</span>(<span class="fu">filter</span>(adiv<span class="sc">$</span>data, variable<span class="sc">==</span><span class="st">"Observed"</span>), value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 3
  variable statistic          p
  &lt;chr&gt;        &lt;dbl&gt;      &lt;dbl&gt;
1 value        0.875 0.00000226</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a>rstatix<span class="sc">::</span><span class="fu">shapiro_test</span>(<span class="fu">filter</span>(adiv<span class="sc">$</span>data, variable<span class="sc">==</span><span class="st">"Shannon"</span>), value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 3
  variable statistic     p
  &lt;chr&gt;        &lt;dbl&gt; &lt;dbl&gt;
1 value        0.984 0.490</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a>rstatix<span class="sc">::</span><span class="fu">shapiro_test</span>(<span class="fu">filter</span>(adiv<span class="sc">$</span>data, variable<span class="sc">==</span><span class="st">"InvSimpson"</span>), value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 3
  variable statistic        p
  &lt;chr&gt;        &lt;dbl&gt;    &lt;dbl&gt;
1 value        0.659 6.43e-12</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>adiv<span class="sc">$</span>data <span class="sc">%&gt;%</span> </span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(variable <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Observed"</span>, <span class="st">"Shannon"</span>, <span class="st">"InvSimpson"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb148-4"><a href="#cb148-4" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">kruskal_test</span>(value<span class="sc">~</span>SexOfPartner)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 × 7
  variable   .y.       n statistic    df     p method        
* &lt;fct&gt;      &lt;chr&gt; &lt;int&gt;     &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;         
1 Observed   value    75     -41.0     2     1 Kruskal-Wallis
2 Shannon    value    75     -33.9     2     1 Kruskal-Wallis
3 InvSimpson value    75     -31.4     2     1 Kruskal-Wallis</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a>adiv<span class="sc">$</span>data <span class="sc">%&gt;%</span> </span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(variable <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Observed"</span>, <span class="st">"Shannon"</span>, <span class="st">"InvSimpson"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb150-3"><a href="#cb150-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(variable) <span class="sc">%&gt;%</span></span>
<span id="cb150-4"><a href="#cb150-4" aria-hidden="true" tabindex="-1"></a>  rstatix<span class="sc">::</span><span class="fu">anova_test</span>(value<span class="sc">~</span>SexOfPartner)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 × 8
  variable   Effect         DFn   DFd     F        p `p&lt;.05`   ges
* &lt;fct&gt;      &lt;chr&gt;        &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt;
1 Observed   SexOfPartner     3    71  3.62 0.017    "*"     0.133
2 Shannon    SexOfPartner     3    71  6.96 0.000358 "*"     0.227
3 InvSimpson SexOfPartner     3    71  1.34 0.267    ""      0.054</code></pre>
</div>
</div>
<p>Observed is not distributed normally, and no significant effect of sex of partner is detected using Kruskal Wallis test. However, Shannon and InvSimpson indices show a significant effect.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a>adonis_bc_partnersex <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(D_BC <span class="sc">~</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>SexOfPartner) </span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a>adonis_pw_bc_partnersex <span class="ot">=</span> <span class="fu">pairwise.adonis</span>(D_BC, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>SexOfPartner)</span>
<span id="cb152-3"><a href="#cb152-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-4"><a href="#cb152-4" aria-hidden="true" tabindex="-1"></a>adonis_uf_partnersex <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(D_UF <span class="sc">~</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>SexOfPartner)</span>
<span id="cb152-5"><a href="#cb152-5" aria-hidden="true" tabindex="-1"></a>adonis_pw_uf_partnersex <span class="ot">=</span> <span class="fu">pairwise.adonis</span>(D_UF, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>SexOfPartner)</span>
<span id="cb152-6"><a href="#cb152-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-7"><a href="#cb152-7" aria-hidden="true" tabindex="-1"></a>adonis_wUF_partnersex <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(D_wUF <span class="sc">~</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>SexOfPartner)</span>
<span id="cb152-8"><a href="#cb152-8" aria-hidden="true" tabindex="-1"></a>adonis_pw_wUF_partnersex <span class="ot">=</span> <span class="fu">pairwise.adonis</span>(D_wUF, phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps_filtered_rel)<span class="sc">$</span>SexOfPartner)</span>
<span id="cb152-9"><a href="#cb152-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-10"><a href="#cb152-10" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(adonis_pw_bc, <span class="at">digits =</span> <span class="dv">3</span>, <span class="at">caption =</span> <span class="st">"Bray-Curtis"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Bray-Curtis</caption>
<colgroup>
<col style="width: 43%">
<col style="width: 3%">
<col style="width: 11%">
<col style="width: 9%">
<col style="width: 6%">
<col style="width: 9%">
<col style="width: 12%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">pairs</th>
<th style="text-align: right;">Df</th>
<th style="text-align: right;">SumsOfSqs</th>
<th style="text-align: right;">F.Model</th>
<th style="text-align: right;">R2</th>
<th style="text-align: right;">p.value</th>
<th style="text-align: right;">p.adjusted</th>
<th style="text-align: left;">sig</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">01_postmenopausal vs 02_premenopausal</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.839</td>
<td style="text-align: right;">4.972</td>
<td style="text-align: right;">0.094</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="even">
<td style="text-align: left;">01_postmenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.255</td>
<td style="text-align: right;">3.474</td>
<td style="text-align: right;">0.067</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="odd">
<td style="text-align: left;">02_premenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3.351</td>
<td style="text-align: right;">10.303</td>
<td style="text-align: right;">0.177</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(adonis_pw_uf, <span class="at">digits =</span> <span class="dv">3</span>, <span class="at">caption =</span><span class="st">"Unweigthed Unifrac"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Unweigthed Unifrac</caption>
<colgroup>
<col style="width: 43%">
<col style="width: 3%">
<col style="width: 11%">
<col style="width: 9%">
<col style="width: 6%">
<col style="width: 9%">
<col style="width: 12%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">pairs</th>
<th style="text-align: right;">Df</th>
<th style="text-align: right;">SumsOfSqs</th>
<th style="text-align: right;">F.Model</th>
<th style="text-align: right;">R2</th>
<th style="text-align: right;">p.value</th>
<th style="text-align: right;">p.adjusted</th>
<th style="text-align: left;">sig</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">01_postmenopausal vs 02_premenopausal</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.104</td>
<td style="text-align: right;">5.145</td>
<td style="text-align: right;">0.097</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="even">
<td style="text-align: left;">01_postmenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.704</td>
<td style="text-align: right;">4.133</td>
<td style="text-align: right;">0.079</td>
<td style="text-align: right;">0.004</td>
<td style="text-align: right;">0.012</td>
<td style="text-align: left;">.</td>
</tr>
<tr class="odd">
<td style="text-align: left;">02_premenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2.263</td>
<td style="text-align: right;">12.762</td>
<td style="text-align: right;">0.210</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
</tbody>
</table>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(adonis_pw_wUF, <span class="at">digits =</span> <span class="dv">3</span>, <span class="at">caption =</span> <span class="st">"Weighted Unifrac"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Weighted Unifrac</caption>
<colgroup>
<col style="width: 43%">
<col style="width: 3%">
<col style="width: 11%">
<col style="width: 9%">
<col style="width: 6%">
<col style="width: 9%">
<col style="width: 12%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">pairs</th>
<th style="text-align: right;">Df</th>
<th style="text-align: right;">SumsOfSqs</th>
<th style="text-align: right;">F.Model</th>
<th style="text-align: right;">R2</th>
<th style="text-align: right;">p.value</th>
<th style="text-align: right;">p.adjusted</th>
<th style="text-align: left;">sig</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">01_postmenopausal vs 02_premenopausal</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.559</td>
<td style="text-align: right;">8.149</td>
<td style="text-align: right;">0.145</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="even">
<td style="text-align: left;">01_postmenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.413</td>
<td style="text-align: right;">5.341</td>
<td style="text-align: right;">0.100</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
<tr class="odd">
<td style="text-align: left;">02_premenopausal vs 03_trans-men</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.369</td>
<td style="text-align: right;">23.593</td>
<td style="text-align: right;">0.330</td>
<td style="text-align: right;">0.001</td>
<td style="text-align: right;">0.003</td>
<td style="text-align: left;">*</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Adonis identifies significant influence of sex of partner oall indices of beta diversity. Pairwise comparisons remain significant for all groups (p = 0.03)</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a>genus_SexOfPartner <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_rel_genus_sexuallyactive),</span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_rel_genus_sexuallyactive), <span class="st">"data.frame"</span>),</span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/genus_SexOfPartner"</span>,</span>
<span id="cb155-4"><a href="#cb155-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">5</span>,</span>
<span id="cb155-5"><a href="#cb155-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.2</span>,</span>
<span id="cb155-6"><a href="#cb155-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb155-7"><a href="#cb155-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"SexOfPartner"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:28.079637 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/genus_SexOfPartner/maaslin2.log"
2024-03-06 21:59:28.249198 INFO::Writing function arguments to log file
2024-03-06 21:59:28.251286 INFO::Verifying options selected are valid
2024-03-06 21:59:28.251538 INFO::Determining format of input files
2024-03-06 21:59:28.25177 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:59:28.253655 INFO::Formula for fixed effects: expr ~  SexOfPartner
2024-03-06 21:59:28.253934 INFO::Factor detected for categorial metadata 'SexOfPartner'. Provide a reference argument or manually set factor ordering to change reference level.
2024-03-06 21:59:28.25414 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:59:28.254344 INFO::Total samples in data: 72
2024-03-06 21:59:28.254543 INFO::Min samples required with min abundance for a feature not to be filtered: 14.400000
2024-03-06 21:59:28.256113 INFO::Total filtered features: 316
2024-03-06 21:59:28.256392 INFO::Filtered feature names from abundance and prevalence filtering: ASV5, ASV11, ASV12, ASV13, ASV17, ASV20, ASV21, ASV22, ASV23, ASV24, ASV30, ASV31, ASV34, ASV35, ASV38, ASV47, ASV48, ASV50, ASV51, ASV53, ASV57, ASV59, ASV63, ASV68, ASV69, ASV71, ASV73, ASV88, ASV95, ASV97, ASV104, ASV115, ASV119, ASV125, ASV128, ASV131, ASV136, ASV140, ASV147, ASV148, ASV151, ASV152, ASV154, ASV155, ASV170, ASV172, ASV174, ASV183, ASV191, ASV196, ASV197, ASV200, ASV201, ASV202, ASV204, ASV207, ASV208, ASV209, ASV218, ASV224, ASV225, ASV228, ASV249, ASV254, ASV255, ASV256, ASV259, ASV264, ASV267, ASV276, ASV277, ASV280, ASV284, ASV289, ASV291, ASV306, ASV307, ASV312, ASV315, ASV319, ASV325, ASV331, ASV347, ASV349, ASV353, ASV356, ASV358, ASV362, ASV363, ASV367, ASV371, ASV375, ASV378, ASV384, ASV390, ASV392, ASV395, ASV398, ASV405, ASV407, ASV410, ASV414, ASV417, ASV421, ASV431, ASV432, ASV440, ASV443, ASV455, ASV459, ASV466, ASV470, ASV473, ASV475, ASV483, ASV494, ASV518, ASV526, ASV532, ASV556, ASV557, ASV558, ASV570, ASV571, ASV573, ASV586, ASV601, ASV605, ASV608, ASV610, ASV612, ASV615, ASV618, ASV628, ASV637, ASV641, ASV650, ASV653, ASV660, ASV671, ASV674, ASV683, ASV685, ASV686, ASV690, ASV693, ASV699, ASV709, ASV711, ASV728, ASV734, ASV740, ASV744, ASV760, ASV764, ASV785, ASV791, ASV793, ASV803, ASV808, ASV821, ASV823, ASV824, ASV829, ASV838, ASV840, ASV849, ASV853, ASV855, ASV860, ASV864, ASV881, ASV891, ASV892, ASV913, ASV918, ASV925, ASV927, ASV936, ASV948, ASV956, ASV967, ASV979, ASV980, ASV984, ASV1000, ASV1004, ASV1009, ASV1011, ASV1038, ASV1039, ASV1040, ASV1044, ASV1050, ASV1055, ASV1068, ASV1069, ASV1070, ASV1083, ASV1087, ASV1088, ASV1089, ASV1103, ASV1109, ASV1111, ASV1115, ASV1117, ASV1121, ASV1123, ASV1132, ASV1134, ASV1136, ASV1145, ASV1172, ASV1176, ASV1194, ASV1196, ASV1197, ASV1200, ASV1201, ASV1203, ASV1221, ASV1222, ASV1235, ASV1236, ASV1238, ASV1242, ASV1249, ASV1251, ASV1253, ASV1255, ASV1260, ASV1275, ASV1282, ASV1284, ASV1285, ASV1286, ASV1291, ASV1304, ASV1312, ASV1316, ASV1317, ASV1321, ASV1322, ASV1331, ASV1344, ASV1361, ASV1382, ASV1383, ASV1412, ASV1414, ASV1426, ASV1428, ASV1438, ASV1439, ASV1440, ASV1483, ASV1489, ASV1499, ASV1501, ASV1503, ASV1517, ASV1520, ASV1534, ASV1535, ASV1536, ASV1537, ASV1539, ASV1550, ASV1565, ASV1601, ASV1602, ASV1615, ASV1617, ASV1624, ASV1627, ASV1647, ASV1669, ASV1672, ASV1724, ASV1732, ASV1734, ASV1767, ASV1769, ASV1774, ASV1778, ASV1783, ASV1812, ASV1857, ASV1859, ASV1862, ASV1869, ASV1887, ASV1926, ASV1935, ASV1980, ASV2008, ASV2029, ASV2033, ASV2047, ASV2071, ASV2087, ASV2119, ASV2159, ASV2214, ASV2267, ASV2269, ASV2301, ASV2322, ASV2328, ASV2329, ASV2331, ASV2387, ASV2395, ASV2396, ASV2480
2024-03-06 21:59:28.256821 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:59:28.257049 INFO::Filtered feature names from variance filtering:
2024-03-06 21:59:28.257257 INFO::Running selected normalization method: TSS
2024-03-06 21:59:28.257642 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:59:28.259722 INFO::Running selected transform method: LOG
2024-03-06 21:59:28.260085 INFO::Running selected analysis method: LM
2024-03-06 21:59:28.26035 INFO::Fitting model to feature number 1, ASV1
2024-03-06 21:59:28.2615 INFO::Fitting model to feature number 2, ASV3
2024-03-06 21:59:28.262465 INFO::Fitting model to feature number 3, ASV4
2024-03-06 21:59:28.263416 INFO::Fitting model to feature number 4, ASV6
2024-03-06 21:59:28.266137 INFO::Counting total values for each feature
2024-03-06 21:59:28.26685 INFO::Writing filtered data to file results/maaslin2/genus_SexOfPartner/features/filtered_data.tsv
2024-03-06 21:59:28.268142 INFO::Writing filtered, normalized data to file results/maaslin2/genus_SexOfPartner/features/filtered_data_norm.tsv
2024-03-06 21:59:28.26911 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/genus_SexOfPartner/features/filtered_data_norm_transformed.tsv
2024-03-06 21:59:28.269936 WARNING::Deleting existing residuals file: results/maaslin2/genus_SexOfPartner/fits/residuals.rds
2024-03-06 21:59:28.270519 INFO::Writing residuals to file results/maaslin2/genus_SexOfPartner/fits/residuals.rds
2024-03-06 21:59:28.271103 WARNING::Deleting existing fitted file: results/maaslin2/genus_SexOfPartner/fits/fitted.rds
2024-03-06 21:59:28.271499 INFO::Writing fitted values to file results/maaslin2/genus_SexOfPartner/fits/fitted.rds
2024-03-06 21:59:28.271876 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/genus_SexOfPartner/all_results.tsv
2024-03-06 21:59:28.272492 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/genus_SexOfPartner/significant_results.tsv
2024-03-06 21:59:28.272879 INFO::Writing heatmap of significant results to file: results/maaslin2/genus_SexOfPartner/heatmap.pdf
[1] "There are no associations to plot!"
2024-03-06 21:59:28.273356 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/genus_SexOfPartner
2024-03-06 21:59:28.273933 INFO::Plotting associations from most to least significant, grouped by metadata
2024-03-06 21:59:28.274214 INFO::Plotting data for metadata number 1, SexOfPartner
2024-03-06 21:59:28.274823 INFO::Creating boxplot for categorical data, SexOfPartner vs ASV3</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a>genus_SexOfPartner2  <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_rel_genus_sexuallyactive),</span>
<span id="cb157-2"><a href="#cb157-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_rel_genus_sexuallyactive), <span class="st">"data.frame"</span>),</span>
<span id="cb157-3"><a href="#cb157-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/genus_SexOfPartnergrp"</span>,</span>
<span id="cb157-4"><a href="#cb157-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">5</span>,</span>
<span id="cb157-5"><a href="#cb157-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.2</span>,</span>
<span id="cb157-6"><a href="#cb157-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb157-7"><a href="#cb157-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="fu">c</span>(<span class="st">"SexOfPartner"</span>, <span class="st">"grp"</span>, <span class="st">"SexOfPartner*grp"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:28.408676 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/genus_SexOfPartnergrp/maaslin2.log"
2024-03-06 21:59:28.578053 INFO::Writing function arguments to log file
2024-03-06 21:59:28.580346 INFO::Verifying options selected are valid
2024-03-06 21:59:28.580592 INFO::Determining format of input files
2024-03-06 21:59:28.580815 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:59:28.582721 WARNING::Feature name not found in metadata so not applied to formula as fixed effect: SexOfPartner*grp
2024-03-06 21:59:28.583102 INFO::Formula for fixed effects: expr ~  SexOfPartner + grp
2024-03-06 21:59:28.583371 INFO::Factor detected for categorial metadata 'SexOfPartner'. Provide a reference argument or manually set factor ordering to change reference level.
2024-03-06 21:59:28.583591 INFO::Factor detected for categorial metadata 'grp'. Provide a reference argument or manually set factor ordering to change reference level.
2024-03-06 21:59:28.583786 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:59:28.583988 INFO::Total samples in data: 72
2024-03-06 21:59:28.584188 INFO::Min samples required with min abundance for a feature not to be filtered: 14.400000
2024-03-06 21:59:28.58578 INFO::Total filtered features: 316
2024-03-06 21:59:28.58607 INFO::Filtered feature names from abundance and prevalence filtering: ASV5, ASV11, ASV12, ASV13, ASV17, ASV20, ASV21, ASV22, ASV23, ASV24, ASV30, ASV31, ASV34, ASV35, ASV38, ASV47, ASV48, ASV50, ASV51, ASV53, ASV57, ASV59, ASV63, ASV68, ASV69, ASV71, ASV73, ASV88, ASV95, ASV97, ASV104, ASV115, ASV119, ASV125, ASV128, ASV131, ASV136, ASV140, ASV147, ASV148, ASV151, ASV152, ASV154, ASV155, ASV170, ASV172, ASV174, ASV183, ASV191, ASV196, ASV197, ASV200, ASV201, ASV202, ASV204, ASV207, ASV208, ASV209, ASV218, ASV224, ASV225, ASV228, ASV249, ASV254, ASV255, ASV256, ASV259, ASV264, ASV267, ASV276, ASV277, ASV280, ASV284, ASV289, ASV291, ASV306, ASV307, ASV312, ASV315, ASV319, ASV325, ASV331, ASV347, ASV349, ASV353, ASV356, ASV358, ASV362, ASV363, ASV367, ASV371, ASV375, ASV378, ASV384, ASV390, ASV392, ASV395, ASV398, ASV405, ASV407, ASV410, ASV414, ASV417, ASV421, ASV431, ASV432, ASV440, ASV443, ASV455, ASV459, ASV466, ASV470, ASV473, ASV475, ASV483, ASV494, ASV518, ASV526, ASV532, ASV556, ASV557, ASV558, ASV570, ASV571, ASV573, ASV586, ASV601, ASV605, ASV608, ASV610, ASV612, ASV615, ASV618, ASV628, ASV637, ASV641, ASV650, ASV653, ASV660, ASV671, ASV674, ASV683, ASV685, ASV686, ASV690, ASV693, ASV699, ASV709, ASV711, ASV728, ASV734, ASV740, ASV744, ASV760, ASV764, ASV785, ASV791, ASV793, ASV803, ASV808, ASV821, ASV823, ASV824, ASV829, ASV838, ASV840, ASV849, ASV853, ASV855, ASV860, ASV864, ASV881, ASV891, ASV892, ASV913, ASV918, ASV925, ASV927, ASV936, ASV948, ASV956, ASV967, ASV979, ASV980, ASV984, ASV1000, ASV1004, ASV1009, ASV1011, ASV1038, ASV1039, ASV1040, ASV1044, ASV1050, ASV1055, ASV1068, ASV1069, ASV1070, ASV1083, ASV1087, ASV1088, ASV1089, ASV1103, ASV1109, ASV1111, ASV1115, ASV1117, ASV1121, ASV1123, ASV1132, ASV1134, ASV1136, ASV1145, ASV1172, ASV1176, ASV1194, ASV1196, ASV1197, ASV1200, ASV1201, ASV1203, ASV1221, ASV1222, ASV1235, ASV1236, ASV1238, ASV1242, ASV1249, ASV1251, ASV1253, ASV1255, ASV1260, ASV1275, ASV1282, ASV1284, ASV1285, ASV1286, ASV1291, ASV1304, ASV1312, ASV1316, ASV1317, ASV1321, ASV1322, ASV1331, ASV1344, ASV1361, ASV1382, ASV1383, ASV1412, ASV1414, ASV1426, ASV1428, ASV1438, ASV1439, ASV1440, ASV1483, ASV1489, ASV1499, ASV1501, ASV1503, ASV1517, ASV1520, ASV1534, ASV1535, ASV1536, ASV1537, ASV1539, ASV1550, ASV1565, ASV1601, ASV1602, ASV1615, ASV1617, ASV1624, ASV1627, ASV1647, ASV1669, ASV1672, ASV1724, ASV1732, ASV1734, ASV1767, ASV1769, ASV1774, ASV1778, ASV1783, ASV1812, ASV1857, ASV1859, ASV1862, ASV1869, ASV1887, ASV1926, ASV1935, ASV1980, ASV2008, ASV2029, ASV2033, ASV2047, ASV2071, ASV2087, ASV2119, ASV2159, ASV2214, ASV2267, ASV2269, ASV2301, ASV2322, ASV2328, ASV2329, ASV2331, ASV2387, ASV2395, ASV2396, ASV2480
2024-03-06 21:59:28.586477 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:59:28.586711 INFO::Filtered feature names from variance filtering:
2024-03-06 21:59:28.586921 INFO::Running selected normalization method: TSS
2024-03-06 21:59:28.587307 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:59:28.58961 INFO::Running selected transform method: LOG
2024-03-06 21:59:28.590031 INFO::Running selected analysis method: LM
2024-03-06 21:59:28.590302 INFO::Fitting model to feature number 1, ASV1
2024-03-06 21:59:28.591541 INFO::Fitting model to feature number 2, ASV3
2024-03-06 21:59:28.59259 INFO::Fitting model to feature number 3, ASV4
2024-03-06 21:59:28.593622 INFO::Fitting model to feature number 4, ASV6
2024-03-06 21:59:28.596537 INFO::Counting total values for each feature
2024-03-06 21:59:28.597343 INFO::Writing filtered data to file results/maaslin2/genus_SexOfPartnergrp/features/filtered_data.tsv
2024-03-06 21:59:28.598179 INFO::Writing filtered, normalized data to file results/maaslin2/genus_SexOfPartnergrp/features/filtered_data_norm.tsv
2024-03-06 21:59:28.598923 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/genus_SexOfPartnergrp/features/filtered_data_norm_transformed.tsv
2024-03-06 21:59:28.599692 WARNING::Deleting existing residuals file: results/maaslin2/genus_SexOfPartnergrp/fits/residuals.rds
2024-03-06 21:59:28.600093 INFO::Writing residuals to file results/maaslin2/genus_SexOfPartnergrp/fits/residuals.rds
2024-03-06 21:59:28.600514 WARNING::Deleting existing fitted file: results/maaslin2/genus_SexOfPartnergrp/fits/fitted.rds
2024-03-06 21:59:28.600832 INFO::Writing fitted values to file results/maaslin2/genus_SexOfPartnergrp/fits/fitted.rds
2024-03-06 21:59:28.601183 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/genus_SexOfPartnergrp/all_results.tsv
2024-03-06 21:59:28.601771 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/genus_SexOfPartnergrp/significant_results.tsv
2024-03-06 21:59:28.602153 INFO::Writing heatmap of significant results to file: results/maaslin2/genus_SexOfPartnergrp/heatmap.pdf
2024-03-06 21:59:28.627989 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/genus_SexOfPartnergrp
2024-03-06 21:59:28.628815 INFO::Plotting associations from most to least significant, grouped by metadata
2024-03-06 21:59:28.629126 INFO::Plotting data for metadata number 1, grp
2024-03-06 21:59:28.629656 INFO::Creating boxplot for categorical data, grp vs ASV3</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:28.693932 INFO::Creating boxplot for categorical data, grp vs ASV6</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:28.750983 INFO::Creating boxplot for categorical data, grp vs ASV1</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:28.808518 INFO::Creating boxplot for categorical data, grp vs ASV1</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:28.866289 INFO::Creating boxplot for categorical data, grp vs ASV3</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:29.227206 INFO::Plotting data for metadata number 2, SexOfPartner
2024-03-06 21:59:29.228589 INFO::Creating boxplot for categorical data, SexOfPartner vs ASV6</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(ps_rel_genus_sexuallyactive) <span class="sc">%&gt;%</span></span>
<span id="cb164-2"><a href="#cb164-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb164-3"><a href="#cb164-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"ASV"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb164-4"><a href="#cb164-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ASV <span class="sc">%in%</span> <span class="fu">filter</span>(genus_SexOfPartner2<span class="sc">$</span>results, qval <span class="sc">&lt;</span> <span class="fl">0.2</span>)<span class="sc">$</span>feature)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   ASV  Kingdom       Phylum       Class           Order           Family
1 ASV1 Bacteria   Firmicutes     Bacilli Lactobacillales Lactobacillaceae
2 ASV3 Bacteria Bacteroidota Bacteroidia   Bacteroidales   Prevotellaceae
3 ASV6 Bacteria   Firmicutes     Bacilli Lactobacillales Streptococcaceae
          Genus Species
1 Lactobacillus    &lt;NA&gt;
2    Prevotella    &lt;NA&gt;
3 Streptococcus    &lt;NA&gt;</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb166"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a>maaslin_res_genus_sexpartner <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">filter</span>(psmelt_rel_genus_sexuallyactive, OTU <span class="sc">%in%</span> <span class="fu">filter</span>(genus_SexOfPartner2<span class="sc">$</span>results, qval <span class="sc">&lt;</span> <span class="fl">0.2</span>)<span class="sc">$</span>feature),</span>
<span id="cb166-2"><a href="#cb166-2" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">aes</span>(<span class="at">x =</span> SexOfPartner, <span class="at">y =</span> Abundance,<span class="at">fill =</span> SexOfPartner)) <span class="sc">+</span></span>
<span id="cb166-3"><a href="#cb166-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>() <span class="sc">+</span></span>
<span id="cb166-4"><a href="#cb166-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb166-5"><a href="#cb166-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(grp<span class="sc">~</span>Genus, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb166-6"><a href="#cb166-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="cf">function</span>(x){<span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(<span class="fl">0.1</span>, x))}) <span class="sc">+</span></span>
<span id="cb166-7"><a href="#cb166-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb166-8"><a href="#cb166-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sex of Partner"</span>,</span>
<span id="cb166-9"><a href="#cb166-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Genus Abundance (%)"</span>,</span>
<span id="cb166-10"><a href="#cb166-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">caption =</span> <span class="st">"Maaslin2 results"</span>) <span class="sc">+</span></span>
<span id="cb166-11"><a href="#cb166-11" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.2</span>))) </span>
<span id="cb166-12"><a href="#cb166-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb166-13"><a href="#cb166-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb166-14"><a href="#cb166-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb166-15"><a href="#cb166-15" aria-hidden="true" tabindex="-1"></a>maaslin_res_genus_sexpartner</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><embed src="MbVagTrans_Results_files/figure-html/maaslin2-1.pdf" class="img-fluid"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb167"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(maaslin_res_genus_sexpartner, <span class="at">filename =</span> <span class="st">"results/maaslin2/genus_SexOfPartner_panelplot.pdf"</span>,</span>
<span id="cb167-2"><a href="#cb167-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">5</span>, <span class="at">width =</span> <span class="dv">6</span>)</span>
<span id="cb167-3"><a href="#cb167-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-4"><a href="#cb167-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-5"><a href="#cb167-5" aria-hidden="true" tabindex="-1"></a><span class="co"># M/F distinction only in trans - filter for this group</span></span>
<span id="cb167-6"><a href="#cb167-6" aria-hidden="true" tabindex="-1"></a>ps_rel_genus_sexuallyactive_trans <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_rel_genus_sexuallyactive, grp <span class="sc">==</span><span class="st">"03_trans-men"</span>)</span>
<span id="cb167-7"><a href="#cb167-7" aria-hidden="true" tabindex="-1"></a>genus_SexOfPartnerTrans <span class="ot">=</span> <span class="fu">Maaslin2</span>(<span class="at">input_data =</span> <span class="fu">otu_table</span>(ps_rel_genus_sexuallyactive_trans),</span>
<span id="cb167-8"><a href="#cb167-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">input_metadata =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(ps_rel_genus_sexuallyactive_trans), <span class="st">"data.frame"</span>),</span>
<span id="cb167-9"><a href="#cb167-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">output =</span> <span class="st">"results/maaslin2/genus_SexOfPartnerTrans"</span>,</span>
<span id="cb167-10"><a href="#cb167-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_abundance =</span> <span class="dv">5</span>,</span>
<span id="cb167-11"><a href="#cb167-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">min_prevalence =</span> <span class="fl">0.2</span>,</span>
<span id="cb167-12"><a href="#cb167-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">max_significance =</span> <span class="fl">0.2</span>,</span>
<span id="cb167-13"><a href="#cb167-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">fixed_effects =</span> <span class="st">"SexOfPartner"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>2024-03-06 21:59:29.818752 WARNING::Input is a matrix, passing through as.data.frame() .
[1] "Warning: Deleting existing log file: results/maaslin2/genus_SexOfPartnerTrans/maaslin2.log"
2024-03-06 21:59:29.996536 INFO::Writing function arguments to log file
2024-03-06 21:59:29.998686 INFO::Verifying options selected are valid
2024-03-06 21:59:29.998958 INFO::Determining format of input files
2024-03-06 21:59:29.999185 INFO::Input format is data samples as rows and metadata samples as rows
2024-03-06 21:59:30.00096 INFO::Formula for fixed effects: expr ~  SexOfPartner
2024-03-06 21:59:30.001236 INFO::Factor detected for categorial metadata 'SexOfPartner'. Provide a reference argument or manually set factor ordering to change reference level.
2024-03-06 21:59:30.001445 INFO::Filter data based on min abundance and min prevalence
2024-03-06 21:59:30.001657 INFO::Total samples in data: 23
2024-03-06 21:59:30.001864 INFO::Min samples required with min abundance for a feature not to be filtered: 4.600000
2024-03-06 21:59:30.003347 INFO::Total filtered features: 311
2024-03-06 21:59:30.00362 INFO::Filtered feature names from abundance and prevalence filtering: ASV5, ASV11, ASV12, ASV13, ASV21, ASV22, ASV23, ASV30, ASV34, ASV35, ASV38, ASV47, ASV48, ASV51, ASV53, ASV57, ASV59, ASV63, ASV68, ASV69, ASV71, ASV73, ASV88, ASV95, ASV97, ASV104, ASV115, ASV119, ASV125, ASV128, ASV131, ASV136, ASV140, ASV147, ASV148, ASV151, ASV152, ASV154, ASV155, ASV170, ASV172, ASV174, ASV183, ASV191, ASV196, ASV197, ASV200, ASV201, ASV202, ASV204, ASV207, ASV208, ASV209, ASV218, ASV224, ASV225, ASV228, ASV249, ASV254, ASV255, ASV256, ASV259, ASV264, ASV267, ASV276, ASV277, ASV280, ASV284, ASV289, ASV291, ASV306, ASV307, ASV312, ASV315, ASV319, ASV325, ASV331, ASV347, ASV349, ASV353, ASV356, ASV358, ASV362, ASV363, ASV367, ASV371, ASV375, ASV378, ASV384, ASV390, ASV392, ASV395, ASV398, ASV405, ASV407, ASV410, ASV414, ASV417, ASV421, ASV431, ASV432, ASV440, ASV443, ASV455, ASV459, ASV466, ASV470, ASV473, ASV475, ASV483, ASV494, ASV518, ASV526, ASV532, ASV556, ASV557, ASV558, ASV570, ASV571, ASV573, ASV586, ASV601, ASV605, ASV608, ASV610, ASV612, ASV615, ASV618, ASV628, ASV637, ASV641, ASV650, ASV653, ASV660, ASV671, ASV674, ASV683, ASV685, ASV686, ASV690, ASV693, ASV699, ASV709, ASV711, ASV728, ASV734, ASV740, ASV744, ASV760, ASV764, ASV785, ASV791, ASV793, ASV803, ASV808, ASV821, ASV823, ASV824, ASV829, ASV838, ASV840, ASV849, ASV853, ASV855, ASV860, ASV864, ASV881, ASV891, ASV892, ASV913, ASV918, ASV925, ASV927, ASV936, ASV948, ASV956, ASV967, ASV979, ASV980, ASV984, ASV1000, ASV1004, ASV1009, ASV1011, ASV1038, ASV1039, ASV1040, ASV1044, ASV1050, ASV1055, ASV1068, ASV1069, ASV1070, ASV1083, ASV1087, ASV1088, ASV1089, ASV1103, ASV1109, ASV1111, ASV1115, ASV1117, ASV1121, ASV1123, ASV1132, ASV1134, ASV1136, ASV1145, ASV1172, ASV1176, ASV1194, ASV1196, ASV1197, ASV1200, ASV1201, ASV1203, ASV1221, ASV1222, ASV1235, ASV1236, ASV1238, ASV1242, ASV1249, ASV1251, ASV1253, ASV1255, ASV1260, ASV1275, ASV1282, ASV1284, ASV1285, ASV1286, ASV1291, ASV1304, ASV1312, ASV1316, ASV1317, ASV1321, ASV1322, ASV1331, ASV1344, ASV1361, ASV1382, ASV1383, ASV1412, ASV1414, ASV1426, ASV1428, ASV1438, ASV1439, ASV1440, ASV1483, ASV1489, ASV1499, ASV1501, ASV1503, ASV1517, ASV1520, ASV1534, ASV1535, ASV1536, ASV1537, ASV1539, ASV1550, ASV1565, ASV1601, ASV1602, ASV1615, ASV1617, ASV1624, ASV1627, ASV1647, ASV1669, ASV1672, ASV1724, ASV1732, ASV1734, ASV1767, ASV1769, ASV1774, ASV1778, ASV1783, ASV1812, ASV1857, ASV1859, ASV1862, ASV1869, ASV1887, ASV1926, ASV1935, ASV1980, ASV2008, ASV2029, ASV2033, ASV2047, ASV2071, ASV2087, ASV2119, ASV2159, ASV2214, ASV2267, ASV2269, ASV2301, ASV2322, ASV2328, ASV2329, ASV2331, ASV2387, ASV2395, ASV2396, ASV2480
2024-03-06 21:59:30.004069 INFO::Total filtered features with variance filtering: 0
2024-03-06 21:59:30.004307 INFO::Filtered feature names from variance filtering:
2024-03-06 21:59:30.004518 INFO::Running selected normalization method: TSS
2024-03-06 21:59:30.004901 INFO::Applying z-score to standardize continuous metadata
2024-03-06 21:59:30.007117 INFO::Running selected transform method: LOG
2024-03-06 21:59:30.007598 INFO::Running selected analysis method: LM
2024-03-06 21:59:30.007872 INFO::Fitting model to feature number 1, ASV1
2024-03-06 21:59:30.009119 INFO::Fitting model to feature number 2, ASV3
2024-03-06 21:59:30.010264 INFO::Fitting model to feature number 3, ASV4
2024-03-06 21:59:30.011359 INFO::Fitting model to feature number 4, ASV6
2024-03-06 21:59:30.01257 INFO::Fitting model to feature number 5, ASV17
2024-03-06 21:59:30.013615 INFO::Fitting model to feature number 6, ASV20
2024-03-06 21:59:30.014545 INFO::Fitting model to feature number 7, ASV24
2024-03-06 21:59:30.015509 INFO::Fitting model to feature number 8, ASV31
2024-03-06 21:59:30.016437 INFO::Fitting model to feature number 9, ASV50
2024-03-06 21:59:30.019777 INFO::Counting total values for each feature
2024-03-06 21:59:30.020876 INFO::Writing filtered data to file results/maaslin2/genus_SexOfPartnerTrans/features/filtered_data.tsv
2024-03-06 21:59:30.02191 INFO::Writing filtered, normalized data to file results/maaslin2/genus_SexOfPartnerTrans/features/filtered_data_norm.tsv
2024-03-06 21:59:30.022793 INFO::Writing filtered, normalized, transformed data to file results/maaslin2/genus_SexOfPartnerTrans/features/filtered_data_norm_transformed.tsv
2024-03-06 21:59:30.0235 WARNING::Deleting existing residuals file: results/maaslin2/genus_SexOfPartnerTrans/fits/residuals.rds
2024-03-06 21:59:30.023957 INFO::Writing residuals to file results/maaslin2/genus_SexOfPartnerTrans/fits/residuals.rds
2024-03-06 21:59:30.024399 WARNING::Deleting existing fitted file: results/maaslin2/genus_SexOfPartnerTrans/fits/fitted.rds
2024-03-06 21:59:30.024758 INFO::Writing fitted values to file results/maaslin2/genus_SexOfPartnerTrans/fits/fitted.rds
2024-03-06 21:59:30.025121 INFO::Writing all results to file (ordered by increasing q-values): results/maaslin2/genus_SexOfPartnerTrans/all_results.tsv
2024-03-06 21:59:30.025786 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.200000 ) to file (ordered by increasing q-values): results/maaslin2/genus_SexOfPartnerTrans/significant_results.tsv
2024-03-06 21:59:30.026124 INFO::Writing heatmap of significant results to file: results/maaslin2/genus_SexOfPartnerTrans/heatmap.pdf
[1] "There are no associations to plot!"
2024-03-06 21:59:30.02658 INFO::Writing association plots (one for each significant association) to output folder: results/maaslin2/genus_SexOfPartnerTrans
[1] "There are no associations to plot!"</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="co"># no significant results in trans group only</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>